--- a/backends/platform/libretro/scripts/configure_submodules.sh
+++ b/backends/platform/libretro/scripts/configure_submodules.sh
@@ -72,5 +72,11 @@
 cd $SUBMODULE_FOLDER
 [ $reset -eq 1 ] && git reset --hard > /dev/null 2>&1
 
+# dirty hack to fix gcc15 build issue
+[ -f libfaad/libfaad/common.h ] && sed -i -e '/#  define strchr index/d' \
+       -e '/\#  define strrchr rindex/d' \
+       -e '/char \*strchr(), \*strrchr();/d' \
+       libfaad/libfaad/common.h
+
 # Success
 echo 0
