--- a/ucm2/Rockchip/rk817-sound/HiFi.conf
+++ b/ucm2/Rockchip/rk817-sound/HiFi.conf
@@ -30,7 +30,7 @@
 	]
 
 	Value {
-		PlaybackMixerElem "Master Playback Volume"
+		PlaybackMixerElem "Master"
 		PlaybackPriority 100
 		PlaybackPCM "hw:${CardId}"
 	}
@@ -59,7 +59,7 @@
 	]
 
 	Value {
-		PlaybackMixerElem "Master Playback Volume"
+		PlaybackMixerElem "Master"
 		PlaybackPriority 200
 		PlaybackPCM "hw:${CardId}"
 		JackControl "Headphones Jack"
