diff --git a/src/paths.rs b/src/paths.rs
index d0ab20b8..ac14b0de 100644
--- a/src/paths.rs
+++ b/src/paths.rs
@@ -70,9 +70,15 @@ impl ResourceFile {
             // On other OSes, resources are accessed as ordinary files.
             #[cfg(not(target_os = "android"))]
             file: {
-                let base_path = get_macos_bundled_resources_path();
-                // When not in a bundle, look in the current directory.
-                let path = base_path.as_deref().unwrap_or(Path::new(".")).join(path);
+                #[cfg(target_os = "linux")]
+                let base_path = Path::new(LINUX_RESOURCE_BASE);
+
+                #[cfg(not(target_os = "linux"))]
+                let base_path = get_macos_bundled_resources_path()
+                    .as_deref()
+                    .unwrap_or(Path::new("."));
+
+                let path = base_path.join(path);
                 std::fs::File::open(path).map_err(|e| e.to_string())?
             },
         })
@@ -110,6 +116,12 @@ pub const WALLPAPER_FILES: &[&str] = &[
 /// the `Documents` directory.
 pub const SANDBOX_DIR: &str = "touchHLE_sandbox";
 
+#[cfg(target_os = "linux")]
+const LINUX_RESOURCE_BASE: &str = "/usr/lib/touchHLE";
+
+#[cfg(target_os = "linux")]
+const LINUX_USER_BASE: &str = "/storage/.config/touchHLE";
+
 /// Get a platform-specific base path needed for accessing touchHLE's
 /// user-modifiable files. This is empty on platforms other than Android.
 pub fn user_data_base_path() -> Cow<'static, Path> {
@@ -136,6 +148,11 @@ pub fn user_data_base_path() -> Cow<'static, Path> {
     }
     #[cfg(not(target_os = "android"))]
     {
+        #[cfg(target_os = "linux")]
+        {
+            return Cow::from(PathBuf::from(LINUX_USER_BASE));
+        }
+
         // When touchHLE is run from a .app bundle on macOS, the user might not
         // be able to control the current directory, so user data needs to go in
         // a standard location.
@@ -190,6 +207,10 @@ pub fn prepopulate_user_data_dir() {
         return;
     }
 
+    #[cfg(target_os = "linux")]
+    let apps_dir = PathBuf::from("/storage/roms/ios");
+
+    #[cfg(not(target_os = "linux"))]
     let apps_dir = base_path.join(APPS_DIR);
     if !apps_dir.is_dir() {
         match std::fs::create_dir(&apps_dir) {
