diff --git a/src/app_picker.rs b/src/app_picker.rs
index 0b6fd38b..9ff58508 100644
--- a/src/environment/app_picker.rs
+++ b/src/environment/app_picker.rs
@@ -500,7 +500,6 @@ fn show_app_picker_gui(
         app_frame.size,
         buttons_row_center,
         &[
-            ("File manager", "openFileManager"),
             ("Quick options", "quickOptionsShow"),
         ],
         None,
@@ -513,7 +513,6 @@ fn show_app_picker_gui(
         buttons_row2_center,
         &[
             ("Copyright info", "copyrightInfoShow"),
-            ("touchHLE.org", "visitWebsite"),
         ],
         None,
     );
