diff --speed-large-files --no-dereference --minimal -Naur linux-6.16-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi linux-6.16-rc5/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
--- a/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
@@ -19,6 +19,14 @@
 	#address-cells = <2>;
 	#size-cells = <2>;
 
+	h616_vqmmc_regulator: mmc-regulator {
+		compatible = "allwinner,h616-mmc-regulator";
+		syscon = <&pio>;
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-boot-on;
+	};
+
 	cpus {
 		#address-cells = <1>;
 		#size-cells = <0>;
@@ -340,7 +348,7 @@
 		};
 
 		pio: pinctrl@300b000 {
-			compatible = "allwinner,sun50i-h616-pinctrl";
+			compatible = "allwinner,sun50i-h616-pinctrl", "syscon";
 			reg = <0x0300b000 0x400>;
 			interrupts = <GIC_SPI 51 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 52 IRQ_TYPE_LEVEL_HIGH>,
@@ -525,6 +533,12 @@
 			cap-sdio-irq;
 			#address-cells = <1>;
 			#size-cells = <0>;
+			vqmmc-supply = <&h616_vqmmc_regulator>;
+			sd-uhs-sdr104;
+			sd-uhs-sdr50;
+			sd-uhs-sdr25;
+			sd-uhs-sdr12;
+			mmc-ddr-1_8v;
 		};
 
 		mmc1: mmc@4021000 {
