--- a/Makefile.am.orig	2024-04-28 12:51:38.866216202 +0300
+++ b/Makefile.am	2024-04-28 12:51:54.426096874 +0300
@@ -1,7 +1,7 @@
 ## Makefile.am for GTK+
 include $(top_srcdir)/Makefile.decl
 
-SRC_SUBDIRS = gdk gtk modules demos tests perf
+SRC_SUBDIRS = gdk gtk modules perf
 SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros build
 
 # require automake 1.4
--- a/Makefile.in.orig	2024-04-28 12:59:09.599008260 +0300
+++ b/Makefile.in	2024-04-28 12:59:26.447165380 +0300
@@ -565,7 +565,7 @@
 	|| { echo "Gtk+Tests:ERROR: Failed to start Xvfb environment for X11 target tests."; exit 1; } \
 	&& DISPLAY=:$$XID && export DISPLAY
 
-SRC_SUBDIRS = gdk gtk modules demos tests perf
+SRC_SUBDIRS = gdk gtk modules perf
 SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros build
 
 # require automake 1.4
--- a/gtk/Makefile.in.orig	2024-04-28 12:59:09.599008260 +0300
+++ b/gtk/Makefile.in	2024-04-28 12:59:26.447165380 +0300
@@ -2010,8 +2010,8 @@
 
 @USE_QUARTZ_FALSE@GTK_PRINT_PREVIEW_COMMAND = "evince --unlink-tempfile --preview --print-settings %s %f"
 @USE_QUARTZ_TRUE@GTK_PRINT_PREVIEW_COMMAND = "open -b com.apple.Preview %f"
-SUBDIRS = theme-bits . tests
-DIST_SUBDIRS = theme-bits tests
+SUBDIRS = theme-bits .
+DIST_SUBDIRS = theme-bits
 @HAVE_CUPS_FALSE@@HAVE_PAPI_CUPS_FALSE@@HAVE_PAPI_FALSE@GTK_PRINT_BACKENDS = file,lpr
 @HAVE_CUPS_FALSE@@HAVE_PAPI_CUPS_FALSE@@HAVE_PAPI_TRUE@GTK_PRINT_BACKENDS = file,papi
 @HAVE_CUPS_TRUE@@HAVE_PAPI_CUPS_FALSE@GTK_PRINT_BACKENDS = file,cups
