--- a/src/shared/machine-id-setup.c
+++ b/src/shared/machine-id-setup.c
@@ -129,7 +129,7 @@
                                                   "2) /etc/machine-id exists and is empty.\n"
                                                   "3) /etc/machine-id is missing and /etc is writable.\n");
                                 else
-                                        return log_error_errno(errno, "Cannot open %s: %m", etc_machine_id);
+                                        return 0;
                         }
 
                         writable = false;
