--- a/tmpfiles.d/provision.conf
+++ b/tmpfiles.d/provision.conf
@@ -15,8 +15,3 @@
 
 # Provision a /etc/hosts file from credentials.
 f^ /etc/hosts - - - - network.hosts
-
-# Provision SSH key for root
-d- /root :0700 root :root -
-d- /root/.ssh :0700 root :root -
-f^ /root/.ssh/authorized_keys :0600 root :root - ssh.authorized_keys.root
--- a/tmpfiles.d/credstore.conf
+++ b/tmpfiles.d/credstore.conf
@@ -6,8 +6,3 @@
 #  (at your option) any later version.
 
 # See tmpfiles.d(5) for details
-
-d /etc/credstore 0700 root root
-d /etc/credstore.encrypted 0700 root root
-z /run/credstore 0700 root root
-z /run/credstore.encrypted 0700 root root
