From 0c7198aa324727f4b62ddd8e003b61e50449ce9a Mon Sep 17 00:00:00 2001
From: spycat88 <spycat88@users.noreply.github.com>
Date: Thu, 9 May 2024 19:55:56 +0100
Subject: [PATCH] udev: set keyboard flag, if device advertises btn_mode

---
 src/udev/udev-builtin-input_id.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/udev/udev-builtin-input_id.c b/src/udev/udev-builtin-input_id.c
index 540f390530..62ec41a4c1 100644
--- a/src/udev/udev-builtin-input_id.c
+++ b/src/udev/udev-builtin-input_id.c
@@ -340,6 +340,10 @@ static bool test_key(sd_device *dev,
                 return false;
         }
 
+        /* if BTN_MODE is found, set device as keyboard (needed for adc-keys/input_sense */
+        if(test_bit(BTN_MODE, bitmask_key))
+                found = true;
+
         /* only consider KEY_* here, not BTN_* */
         for (size_t i = 0; i < BTN_MISC/BITS_PER_LONG && !found; i++) {
                 if (bitmask_key[i])
-- 
2.34.1

