From 8c575d1015cfce1f66ce7d67132d43f643090485 Mon Sep 17 00:00:00 2001
From: sydarn <sydarn@proton.me>
Date: Mon, 30 Sep 2024 18:59:27 +0200
Subject: [PATCH] singleadc: port to linux 6.12

---
 Makefile                   | 2 +-
 rocknix-singleadc-joypad.c | 8 +++-----
 2 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/Makefile b/Makefile
index f111bc6..c68a137 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 ifeq ($(DEVICE),$(filter $(DEVICE), S922X RK3588))
 	obj-m := rocknix-joypad.o
-else ifeq ($(DEVICE),$(filter $(DEVICE), RK3399))
+else ifeq ($(DEVICE),$(filter $(DEVICE), RK3399 RK3566))
 	obj-m := rocknix-singleadc-joypad.o
 else
 	obj-m := rocknix-joypad.o rocknix-singleadc-joypad.o
diff --git a/rocknix-singleadc-joypad.c b/rocknix-singleadc-joypad.c
index 97f18c6..39d998b 100644
--- a/rocknix-singleadc-joypad.c
+++ b/rocknix-singleadc-joypad.c
@@ -140,7 +140,7 @@ static int pwm_vibrator_start(struct joypad *joypad)
 	pwm_set_relative_duty_cycle(&state, joypad->level, 0xffff);
 	state.enabled = true;
 
-	err = pwm_apply_state(joypad->pwm, &state);
+	err = pwm_apply_might_sleep(joypad->pwm, &state);
 	if (err) {
 		 dev_err(pdev, "failed to apply pwm state: %d", err);
 		 return err;
@@ -1078,7 +1078,7 @@ static int joypad_rumble_setup(struct device *dev, struct joypad *joypad)
 	pwm_init_state(joypad->pwm, &state);
 	state.enabled = false;
 
-	error = pwm_apply_state(joypad->pwm, &state);
+	error = pwm_apply_might_sleep(joypad->pwm, &state);
 	if (error) {
 		 dev_err(dev, "failed to apply initial PWM state: %d",
 			 error);
@@ -1331,14 +1331,12 @@ static int joypad_probe(struct platform_device *pdev)
 	return 0;
 }
 
-static int joypad_remove(struct platform_device *pdev)
+static void joypad_remove(struct platform_device *pdev)
 {
 	struct joypad *joypad = platform_get_drvdata(pdev);
 	sysfs_remove_group(&pdev->dev.kobj, &joypad_attr_group);
 	if (joypad->has_rumble)
 		sysfs_remove_group(&pdev->dev.kobj, &joypad_rumble_attr_group);
-
-	return 0;
 }
 /*----------------------------------------------------------------------------*/
 static const struct of_device_id joypad_of_match[] = {
-- 
2.46.0

