From 43eebbed29509c09452f5730fd6b57b4728e16ef Mon Sep 17 00:00:00 2001
From: Philippe Simons <simons.philippe@gmail.com>
Date: Sat, 5 Oct 2024 19:14:30 +0200
Subject: [PATCH] compile only rocknix-singleadc-joypad.o for H700

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index c68a137..536c681 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 ifeq ($(DEVICE),$(filter $(DEVICE), S922X RK3588))
 	obj-m := rocknix-joypad.o
-else ifeq ($(DEVICE),$(filter $(DEVICE), RK3399 RK3566))
+else ifeq ($(DEVICE),$(filter $(DEVICE), RK3399 RK3566 H700))
 	obj-m := rocknix-singleadc-joypad.o
 else
 	obj-m := rocknix-joypad.o rocknix-singleadc-joypad.o
-- 
2.46.0

