From 85366ef1d616ea2867cfd7cfe50994548bbb5b91 Mon Sep 17 00:00:00 2001
From: Philippe Simons <simons.philippe@gmail.com>
Date: Sun, 6 Oct 2024 13:06:41 +0200
Subject: [PATCH] allow to remap amux channels

---
 rocknix-singleadc-joypad.c | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/rocknix-singleadc-joypad.c b/rocknix-singleadc-joypad.c
index 39d998b..c2f2efd 100644
--- a/rocknix-singleadc-joypad.c
+++ b/rocknix-singleadc-joypad.c
@@ -893,6 +893,17 @@ err_out:
 static int joypad_adc_setup(struct device *dev, struct joypad *joypad)
 {
 	int nbtn;
+	u32 channel_mapping[] = {0, 1, 2, 3};
+
+	if (device_property_present(dev, "amux-channel-mapping")) {
+		int ret;
+		ret = of_property_read_u32_array(dev->of_node,
+				"amux-channel-mapping", channel_mapping, 4);
+		if (ret < 0) {
+			dev_err(dev, "invalid channel mapping\n");
+			return -EINVAL;
+		}
+	}
 
 	/* adc button struct init */
 	joypad->adcs = devm_kzalloc(dev, joypad->amux_count *
@@ -913,7 +924,6 @@ static int joypad_adc_setup(struct device *dev, struct joypad *joypad)
 			adc->max *= adc->scale;
 			adc->min *= adc->scale;
 		}
-		adc->amux_ch = nbtn;
 		adc->invert = false;
 
 		switch (nbtn) {
@@ -974,6 +984,7 @@ static int joypad_adc_setup(struct device *dev, struct joypad *joypad)
 					__func__, nbtn);
 				return -EINVAL;
 		}
+		adc->amux_ch = channel_mapping[nbtn];
 	}
 	return	0;
 }
-- 
2.46.0

