--- a/timezone.c	2004-11-13 08:35:12.000000000 +0000
+++ b/timezone.c	2024-05-01 09:22:13.780112533 +0000
@@ -39,6 +39,7 @@
 
 #include "zip.h"
 #include "timezone.h"
+#include <time.h>
 #include <ctype.h>
 #include <errno.h>
 
--- a/unix/configure	2008-06-20 03:32:20.000000000 +0000
+++ b/unix/configure	2024-05-01 09:27:12.149292690 +0000
@@ -544,7 +595,7 @@
 echo Check for errno declaration
 cat > conftest.c << _EOF_
 #include <errno.h>
-main()
+int main()
 {
   errno = 0;
   return 0;
@@ -556,6 +607,7 @@
 
 echo Check for directory libraries
 cat > conftest.c << _EOF_
+#include <dirent.h>
 int main() { return closedir(opendir(".")); }
 _EOF_
 
@@ -621,11 +673,13 @@
 
 echo Check for valloc
 cat > conftest.c << _EOF_
-main()
+#include <stdlib.h>
+int main()
 {
 #ifdef MMAP
     valloc();
 #endif
+    return 0;
 }
 _EOF_
 $CC ${CFLAGS} -c conftest.c > /dev/null 2>/dev/null

