diff --git a/src/common/common_funcs.h b/src/common/common_funcs.h
index 639c2d0e0d..aac253e1be 100644
--- a/src/common/common_funcs.h
+++ b/src/common/common_funcs.h
@@ -32,6 +32,8 @@
 #define INSERT_PADDING_WORDS_NOINIT(num_words)                                                     \
     [[maybe_unused]] std::array<u32, num_words> CONCAT2(pad, __LINE__)
 
+#define LTO_NOINLINE __attribute__((noinline))
+
 #ifdef _MSC_VER
 #   define locale_t _locale_t // Locale Cross-Compatibility
 #endif // _MSC_VER
diff --git a/src/core/hle/kernel/kernel.cpp b/src/core/hle/kernel/kernel.cpp
index 6986a98e35..113daad46e 100644
--- a/src/core/hle/kernel/kernel.cpp
+++ b/src/core/hle/kernel/kernel.cpp
@@ -356,7 +356,7 @@ struct KernelCore::Impl {
     }
 
     /// Sets the host thread ID for the caller.
-    u32 SetHostThreadId(std::size_t core_id) {
+    LTO_NOINLINE u32 SetHostThreadId(std::size_t core_id) {
         // This should only be called during core init.
         ASSERT(tls_data.host_thread_id == UINT8_MAX);
 
@@ -367,14 +367,14 @@ struct KernelCore::Impl {
     }
 
     /// Gets the host thread ID for the caller
-    u32 GetHostThreadId() const {
+    LTO_NOINLINE u32 GetHostThreadId() const {
         return tls_data.host_thread_id;
     }
 
     // Gets the dummy KThread for the caller, allocating a new one if this is the first time
     KThread* GetHostDummyThread(ThreadLocalData& t, KThread* existing_thread) {
         if (t.thread == nullptr) {
-            auto const initialize{[](KThread* thread) {
+            auto const initialize{[](KThread* thread) LTO_NOINLINE {
                 ASSERT(KThread::InitializeDummyThread(thread, nullptr).IsSuccess());
                 return thread;
             }};
@@ -406,10 +406,10 @@ struct KernelCore::Impl {
     }
 
     // Forces singlecore
-    bool IsPhantomModeForSingleCore() const {
+    LTO_NOINLINE bool IsPhantomModeForSingleCore() const {
         return tls_data.is_phantom_mode_for_singlecore;
     }
-    void SetIsPhantomModeForSingleCore(bool value) {
+    LTO_NOINLINE void SetIsPhantomModeForSingleCore(bool value) {
         ASSERT(!is_multicore);
         tls_data.is_phantom_mode_for_singlecore = value;
     }
@@ -418,12 +418,12 @@ struct KernelCore::Impl {
         return is_shutting_down.load(std::memory_order_relaxed);
     }
 
-    KThread* GetCurrentEmuThread() {
+    LTO_NOINLINE KThread* GetCurrentEmuThread() {
         auto& t = tls_data;
         return t.current_thread ? t.current_thread : (t.current_thread = GetHostDummyThread(t, nullptr));
     }
 
-    void SetCurrentEmuThread(KThread* thread) {
+    LTO_NOINLINE void SetCurrentEmuThread(KThread* thread) {
         tls_data.current_thread = thread;
     }
 
