diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 780b802916..2a66ab7549 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -52,15 +52,16 @@ include(GNUInstallDirs)
 include(GetGitRevisionDescription)
 include(InstallRequiredSystemLibraries)
 include(FindPkgConfig)
-include(ExternalProject)
-include(BuildLibMPEG2)
+#include(ExternalProject)
+#include(BuildLibMPEG2)
 
 use_cxx11( )
 
 PKG_SEARCH_MODULE(SDL2 REQUIRED sdl2)
 PKG_SEARCH_MODULE(SDL2_TTF REQUIRED SDL2_ttf)
 PKG_SEARCH_MODULE(SDL2_MIXER REQUIRED SDL2_mixer)
-build_libmpeg2( )
+PKG_SEARCH_MODULE(MPEG2 REQUIRED libmpeg2)
+#build_libmpeg2( )
 
 message(STATUS "Target: ${CMAKE_SYSTEM_NAME} ${CMAKE_TARGET_ARCHITECTURES}")
 
@@ -114,7 +115,6 @@ add_subdirectory(timer)
 add_subdirectory(video)
 add_subdirectory(vldp)
 
-add_dependencies( vldp libmpeg2 )
 add_dependencies( ldp-out vldp )
 add_dependencies( game vldp )
 add_dependencies( sound vldp )
@@ -162,3 +162,5 @@ set(CPACK_PACKAGE_VERSION_MINOR "${VER_MINOR}")
 set(CPACK_PACKAGE_VERSION_PATCH "${VER_PATCH}")
 
 include(CPack)
+install(TARGETS hypseus DESTINATION bin)
+install(DIRECTORY ../pics ../fonts ../sound DESTINATION /usr/share/daphne)

diff --git a/src/vldp/vldp.h b/src/vldp/vldp.h
index a59c723..3c986dc 100644
--- a/src/vldp/vldp.h
+++ b/src/vldp/vldp.h
@@ -34,6 +34,8 @@ extern "C" {
 // Ubuntu Linux complains with plain <SDL.h>
 #include <SDL2/SDL.h> // only used for threading
 
+#include <mpeg2dec/mpeg2.h>
+
 struct yuv_buf {
     uint8_t *Y;     // Y channel
     uint8_t *U;     // U channel

diff --git a/src/vldp/vldp_internal.h b/src/vldp/vldp_internal.h
index cd046fe..67cea2d 100644
--- a/src/vldp/vldp_internal.h
+++ b/src/vldp/vldp_internal.h
@@ -27,7 +27,6 @@
 
 #include "vldp.h" // for the VLDP_BOOL definition and SDL.h
 
-#include <mpeg2.h>
 
 // this is which version of the .dat file format we are using
 #define DAT_VERSION 2

diff --git a/src/manymouse/manymouse.c b/src/manymouse/manymouse.c
index c2889c2e2b..992d184f09 100644
--- a/src/manymouse/manymouse.c
+++ b/src/manymouse/manymouse.c
@@ -15,7 +15,6 @@ static const char *manymouse_copyright =
 
 extern const ManyMouseDriver *ManyMouseDriver_windows;
 extern const ManyMouseDriver *ManyMouseDriver_evdev;
-extern const ManyMouseDriver *ManyMouseDriver_xinput2;
 extern const ManyMouseDriver *ManyMouseDriver_hidmanager;
 extern const ManyMouseDriver *ManyMouseDriver_hidutilities;
 
@@ -33,7 +32,6 @@ extern const ManyMouseDriver *ManyMouseDriver_hidutilities;
 static const ManyMouseDriver **mice_drivers[] =
 {
     &ManyMouseDriver_evdev,
-    &ManyMouseDriver_xinput2,
     &ManyMouseDriver_windows,
     &ManyMouseDriver_hidmanager,
     &ManyMouseDriver_hidutilities

--- a/src/manymouse/CMakeLists.txt
+++ b/src/manymouse/CMakeLists.txt
@@ -1,7 +1,8 @@
+
+
 set( LIB_SOURCES
     manymouse.c linux_evdev.c macosx_hidmanager.c
     macosx_hidutilities.c windows_wminput.c
-    x11_xinput2.c
 )
 
 set( LIB_HEADERS