--- a/Makefile
+++ b/Makefile
@@ -14,6 +14,7 @@
 #EXTRA_CFLAGS += -Wno-unused-function
 EXTRA_CFLAGS += -Wno-unused
 #EXTRA_CFLAGS += -Wno-uninitialized
+EXTRA_CFLAGS += -w
 
 ############ ANDROID COMMON KERNEL ############
 # clang
--- a/os_dep/linux/ioctl_cfg80211.h
+++ b/os_dep/linux/ioctl_cfg80211.h
@@ -90,7 +90,7 @@
 
 #if defined(CONFIG_DFS_MASTER) && (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 17, 0))
 #ifndef CONFIG_RTW_CFG80211_CAC_EVENT
-#define CONFIG_RTW_CFG80211_CAC_EVENT 1
+#define CONFIG_RTW_CFG80211_CAC_EVENT 0
 #endif
 #else
 #ifdef CONFIG_RTW_CFG80211_CAC_EVENT
