diff --git a/Source/Core/Core/BootManager.cpp b/Source/Core/Core/BootManager.cpp
index fc93d01a1e..915e77da77 100644
--- a/Source/Core/Core/BootManager.cpp
+++ b/Source/Core/Core/BootManager.cpp
@@ -66,6 +66,11 @@ bool BootCore(Core::System& system, std::unique_ptr<BootParameters> boot,
   if (!StartUp.SetPathsAndGameMetadata(system, *boot))
     return false;
 
+  if (std::holds_alternative<BootParameters::Disc>(boot->parameters))
+  {
+    StartUp.SetBaseSaveState(PathToFileName(std::get<BootParameters::Disc>(boot->parameters).path));
+  }
+
   // Movie settings
   auto& movie = system.GetMovie();
   if (movie.IsPlayingInput() && movie.IsConfigSaved())
diff --git a/Source/Core/Core/ConfigManager.cpp b/Source/Core/Core/ConfigManager.cpp
index 5a123c87cd..06890be2dd 100644
--- a/Source/Core/Core/ConfigManager.cpp
+++ b/Source/Core/Core/ConfigManager.cpp
@@ -183,6 +183,10 @@ void SConfig::SetRunningGameMetadata(const std::string& game_id)
   SetRunningGameMetadata(game_id, "", 0, 0, DiscIO::Region::Unknown);
 }
 
+void SConfig::SetBaseSaveState(const std::string& base_save_state) {
+  m_base_save_state = base_save_state;
+}
+
 void SConfig::SetRunningGameMetadata(const std::string& game_id, const std::string& gametdb_id,
                                      u64 title_id, u16 revision, DiscIO::Region region)
 {
diff --git a/Source/Core/Core/ConfigManager.h b/Source/Core/Core/ConfigManager.h
index d21b087a51..f0ce0b7e71 100644
--- a/Source/Core/Core/ConfigManager.h
+++ b/Source/Core/Core/ConfigManager.h
@@ -63,6 +63,8 @@ struct SConfig
   // TODO: remove this as soon as the ticket view hack in IOS/ES/Views is dropped.
   bool m_disc_booted_from_game_list = false;
 
+  const std::string& GetBaseSaveState() const { return m_base_save_state; }
+  void SetBaseSaveState(const std::string& base_save_state);
   const std::string GetGameID() const;
   const std::string GetGameTDBID() const;
   const std::string GetTitleName() const;
@@ -118,6 +120,7 @@ private:
   static SConfig* m_Instance;
   mutable std::recursive_mutex m_metadata_lock;
 
+  std::string m_base_save_state;
   std::string m_game_id;
   std::string m_gametdb_id;
   std::string m_title_name;
diff --git a/Source/Core/Core/State.cpp b/Source/Core/Core/State.cpp
index a244fda1b6..13d5baa453 100644
--- a/Source/Core/Core/State.cpp
+++ b/Source/Core/Core/State.cpp
@@ -248,11 +248,10 @@ static std::string SystemTimeAsDoubleToString(double time)
   return fmt::format(std::locale{""}, "{:%x %X}", *local_time);
 }
 
-static std::string MakeStateFilename(int number)
-{
-  return fmt::format("{}{}.s{:02d}", File::GetUserPath(D_STATESAVES_IDX),
-                     SConfig::GetInstance().GetGameID(), number);
-}
+static std::string MakeStateFilename(bool useId, int number);
+static std::string MakeStateFilename_partdirectory(int number);
+static std::string MakeStateFilename_partfileName(int number);
+static std::string MakeStateFilename_partfileID(int number);
 
 static std::vector<SlotWithTimestamp> GetUsedSlotsWithTimestamp()
 {
@@ -256,7 +259,7 @@ static std::vector<SlotWithTimestamp> GetUsedSlotsWithTimestamp()
   StateHeader header;
   for (int i = 1; i <= int(NUM_STATES); ++i)
   {
-    std::string filename = MakeStateFilename(i);
+    std::string filename = MakeStateFilename(true, i);
     if (!File::Exists(filename) || !ReadHeader(filename, header))
       continue;
 
@@ -525,7 +528,7 @@ std::string GetInfoStringOfSlot(int slot, bool translate)
 {
   std::lock_guard lk{s_state_saves_in_progress};
 
-  std::string filename = MakeStateFilename(slot);
+  std::string filename = MakeStateFilename(true, slot);
   if (!File::Exists(filename))
     return translate ? Common::GetStringT("Empty") : "Empty";
 
@@ -538,6 +541,6 @@ u64 GetUnixTimeOfSlot(int slot)
 
   State::StateHeader header;
-  if (!ReadHeader(MakeStateFilename(slot), header))
+  if (!ReadHeader(MakeStateFilename(true, slot), header))
     return 0;
 
   constexpr u64 MS_PER_SEC = 1000;
@@ -703,52 +706,88 @@ void Shutdown()
   s_flush_unsaved_data_hook.reset();
 }
 
+static std::string MakeStateFilename(bool useId, int number)
+{
+  // Priority: file named by disc filename over file named by game ID.
+  // If the game was not loaded from a disc image, fall back to game ID.
+  std::string filePartName = MakeStateFilename_partfileName(number);
+  std::string fileWithName =
+      fmt::format("{}{}", MakeStateFilename_partdirectory(number), filePartName);
+  if (File::Exists(fileWithName) && filePartName != "")
+    return fileWithName;
+  std::string fileWithId =
+      fmt::format("{}{}", MakeStateFilename_partdirectory(number), MakeStateFilename_partfileID(number));
+  if (File::Exists(fileWithId) && useId)
+    return fileWithId;
+  if (filePartName == "")
+    return fileWithId;
+  return fileWithName;
+}
+
+static std::string MakeStateFilename_partdirectory(int number)
+{
+  return File::GetUserPath(D_STATESAVES_IDX);
+}
+
+static std::string MakeStateFilename_partfileID(int number)
+{
+  return fmt::format("{}.s{:02d}", SConfig::GetInstance().GetGameID(), number);
+}
+
+static std::string MakeStateFilename_partfileName(int number)
+{
+  std::string basesavestate = SConfig::GetInstance().GetBaseSaveState();
+  if (basesavestate == "")
+    return "";
+  return fmt::format("{}.s{:02d}", basesavestate, number);
+}
+
 void Save(Core::System& system, int slot)
 {
-  SaveAs(system, MakeStateFilename(slot));
+  SaveAs(system, MakeStateFilename(false, slot));
 }
 
 void Load(Core::System& system, int slot)
 {
-  LoadAs(system, MakeStateFilename(slot));
+  LoadAs(system, MakeStateFilename(true, slot));
 }
 
 void LoadLastSaved(Core::System& system, int i)
 {
   if (!CheckIfStateLoadIsAllowed(system))
     return;
 
   Core::RunOnCPUThread(system, [&system, i] {
     // Data must reach the filesystem for up to date "UsedSlots".
     s_compress_and_dump_thread.WaitForCompletion();
 
     std::vector<SlotWithTimestamp> used_slots = GetUsedSlotsWithTimestamp();
     if (std::size_t(i) > used_slots.size())
     {
       Core::DisplayMessage("State doesn't exist", 2000);
       return;
     }
 
     std::ranges::stable_sort(used_slots, std::ranges::greater{}, &SlotWithTimestamp::timestamp);
-    LoadAsFromCore(system, MakeStateFilename(used_slots[i].slot));
+    LoadAsFromCore(system, MakeStateFilename(true, used_slots[i].slot));
   });
 }
 
 void SaveFirstSaved(Core::System& system)
 {
   Core::RunOnCPUThread(system, [&system, lock = GetStateSaveTaskLock()] {
     // Data must reach the filesystem for up to date "UsedSlots".
     s_compress_and_dump_thread.WaitForCompletion();
 
     std::vector<SlotWithTimestamp> used_slots = GetUsedSlotsWithTimestamp();
     auto slot = GetEmptySlot(used_slots);
     if (!slot.has_value())
     {
       // overwrite the oldest state
       std::ranges::stable_sort(used_slots, {}, &SlotWithTimestamp::timestamp);
       slot = used_slots.front().slot;
     }
 
-    SaveAsFromCore(system, MakeStateFilename(*slot));
+    SaveAsFromCore(system, MakeStateFilename(false, *slot));
   });
 }
