diff --git a/CMakeLists.txt b/CMakeLists.txt
index c277e2a10a..c67c2addb4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -735,7 +735,7 @@ endif()
 
 dolphin_find_optional_system_library(MBEDTLS Externals/mbedtls 2.28)
 
-dolphin_find_optional_system_library(CURL Externals/curl)
+add_subdirectory(Externals/curl)
 
 if(NOT ANDROID)
   dolphin_find_optional_system_library(Iconv Externals/libiconv-1.14)
