--- a/GLideN64/src/GLideNHQ/TxHiResLoader.h
+++ b/GLideN64/src/GLideNHQ/TxHiResLoader.h
@@ -1,6 +1,8 @@
 #ifndef TXHIRESLOADER_H
 #define TXHIRESLOADER_H
 
+#include <cstdint>
+
 /* support hires textures
  *   0: disable
  *   1: enable
--- a/GLideN64/src/GLideNHQ/TxHiResLoader.cpp
+++ b/GLideN64/src/GLideNHQ/TxHiResLoader.cpp
@@ -1,3 +1,5 @@
+#include <cstdint>
+
 #include "TxHiResLoader.h"
 #include "TxDbg.h"
 #include "TxDbg.h"
