diff --git a/build.rs b/build.rs
index 2cbcdf5..17844e0 100644
--- a/build.rs
+++ b/build.rs
@@ -145,10 +145,10 @@ fn main() {
         println!("cargo:rustc-link-lib=static=clang_rt.osx");
     }
 
-    volk_build.flag("-flto=thin");
-    rdp_build.flag("-flto=thin");
-    simd_build.flag("-flto=thin");
-    retroachievements_build.flag("-flto=thin");
+    volk_build.flag("-flto");
+    rdp_build.flag("-flto");
+    simd_build.flag("-flto");
+    retroachievements_build.flag("-flto");
 
     volk_build.compile("volk");
     rdp_build.compile("parallel-rdp");
