diff --git a/build.rs b/build.rs
index ad87a95..42e85f4 100644
--- a/build.rs
+++ b/build.rs
@@ -96,9 +96,9 @@ fn main() {
             .unwrap();
     }
 
-    volk_build.flag("-flto=thin");
-    rdp_build.flag("-flto=thin");
-    simd_build.flag("-flto=thin");
+    volk_build.flag("-flto");
+    rdp_build.flag("-flto");
+    simd_build.flag("-flto");
 
     volk_build.compile("volk");
     rdp_build.compile("parallel-rdp");
