diff --git a/src/ui/storage.rs b/src/ui/storage.rs
index 470b077..5810ee7 100644
--- a/src/ui/storage.rs
+++ b/src/ui/storage.rs
@@ -325,9 +325,12 @@ pub fn load_saves(ui: &mut ui::Ui, netplay: &mut Option<netplay::Netplay>) {
 pub fn decompress_file(input: &[u8], name: &str) -> Vec<u8> {
     let mut decompressed_file = Vec::new();
     {
-        let mut reader = zip::ZipArchive::new(std::io::Cursor::new(input)).unwrap();
-        let mut file = reader.by_name(name).unwrap();
-        file.read_to_end(&mut decompressed_file).unwrap();
+        let reader = zip::ZipArchive::new(std::io::Cursor::new(input));
+        if let Ok(mut reader) = reader {
+            if let Ok(mut file) = reader.by_name(name) {
+                file.read_to_end(&mut decompressed_file).unwrap();
+            }
+        }
     }
     decompressed_file
 }
