diff --git a/core/stdclass.h b/core/stdclass.h
index 5a3cad09..3612d9d5 100644
--- a/core/stdclass.h
+++ b/core/stdclass.h
@@ -13,17 +13,8 @@
 #include <cassert>
 #include <time.h>
 
-#ifdef __ANDROID__
-#include <sys/mman.h>
-#undef PAGE_MASK
-#elif defined(__APPLE__) && defined(__aarch64__)
 #define PAGE_SIZE 16384
-#elif !defined(PAGE_SIZE)
-#define PAGE_SIZE 4096
-#endif
-#ifndef PAGE_MASK
 #define PAGE_MASK (PAGE_SIZE-1)
-#endif
 
 class cThread
 {
