---
 drivers/power/supply/bq256xx_charger.c | 2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/power/supply/bq256xx_charger.c
+++ b/drivers/power/supply/bq256xx_charger.c
@@ -1775,6 +1775,7 @@ static const struct i2c_device_id bq256xx_i2c_ids[] = {
 	{ "bq25611d", (kernel_ulong_t)&bq256xx_chip_info_tbl[BQ25611D] },
 	{ "bq25618", (kernel_ulong_t)&bq256xx_chip_info_tbl[BQ25618] },
 	{ "bq25619", (kernel_ulong_t)&bq256xx_chip_info_tbl[BQ25619] },
+	{ "sgm41515", (kernel_ulong_t)&bq256xx_chip_info_tbl[BQ25601D] },
 	{}
 };
 MODULE_DEVICE_TABLE(i2c, bq256xx_i2c_ids);
@@ -1787,6 +1788,7 @@ static const struct of_device_id bq256xx_of_match[] = {
 	{ .compatible = "ti,bq25611d", .data = &bq256xx_chip_info_tbl[BQ25611D] },
 	{ .compatible = "ti,bq25618", .data = &bq256xx_chip_info_tbl[BQ25618] },
 	{ .compatible = "ti,bq25619", .data = &bq256xx_chip_info_tbl[BQ25619] },
+	{ .compatible = "sgmicro,sgm41515", .data = &bq256xx_chip_info_tbl[BQ25601D] },
 	{}
 };
 MODULE_DEVICE_TABLE(of, bq256xx_of_match);
