Subject: [PATCH] ASoC: qcom: sm8250: skip jack setup when no codec supports it

---
 sound/soc/qcom/sm8250.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/sound/soc/qcom/sm8250.c b/sound/soc/qcom/sm8250.c
--- a/sound/soc/qcom/sm8250.c
+++ b/sound/soc/qcom/sm8250.c
@@ -31,7 +31,9 @@ static int sm8250_snd_init(struct snd_soc_pcm_runtime *rtd)
 {
 	struct sm8250_snd_data *data = snd_soc_card_get_drvdata(rtd->card);
 	struct snd_soc_dai *cpu_dai = snd_soc_rtd_to_cpu(rtd, 0);
+	struct snd_soc_dai *codec_dai;
+	int i;
 
 	switch (cpu_dai->id) {
 	case DISPLAY_PORT_RX:
 		return qcom_snd_dp_jack_setup(rtd, &data->dp_jack, 0);
@@ -40,7 +42,13 @@ static int sm8250_snd_init(struct snd_soc_pcm_runtime *rtd)
 		return qcom_snd_usb_offload_jack_setup(rtd, &data->usb_offload_jack,
 						       &data->usb_offload_jack_setup);
 	default:
-		return qcom_snd_wcd_jack_setup(rtd, &data->jack, &data->jack_setup);
+		for_each_rtd_codec_dais(rtd, i, codec_dai) {
+			if (codec_dai->component &&
+			    codec_dai->component->driver->set_jack)
+				return qcom_snd_wcd_jack_setup(rtd, &data->jack,
+							       &data->jack_setup);
+		}
+		return 0;
 	}
 }
 
