diff --git a/src/common/common_paths.h b/src/common/common_paths.h
index f12eedb..1390abe 100644
--- a/src/common/common_paths.h
+++ b/src/common/common_paths.h
@@ -29,7 +29,7 @@
 // For compatibility with XDG paths.
 #define EMU_DATA_DIR "citra-emu"
 #else
-#define EMU_DATA_DIR "citra-emu"
+#define EMU_DATA_DIR "lime3ds"
 #endif
 #endif
 
@@ -44,7 +44,7 @@
 #define SDMC_DIR "sdmc"
 #define NAND_DIR "nand"
 #define SYSDATA_DIR "sysdata"
-#define LOG_DIR "log"
+#define LOG_DIR "/var/log"
 #define CHEATS_DIR "cheats"
 #define DLL_DIR "external_dlls"
 #define SHADER_DIR "shaders"
@@ -55,7 +55,7 @@
 
 // Filenames
 // Files in the directory returned by GetUserPath(UserPath::LogDir)
-#define LOG_FILE "citra_log.txt"
+#define LOG_FILE "lime3ds.log"
 
 // Files in the directory returned by GetUserPath(UserPath::ConfigDir)
 #define EMU_CONFIG "emu.ini"
diff --git a/src/common/file_util.cpp b/src/common/file_util.cpp
index cda752e..c645390 100644
--- a/src/common/file_util.cpp
+++ b/src/common/file_util.cpp
@@ -819,7 +819,7 @@ void SetUserPath(const std::string& path) {
     g_paths.emplace(UserPath::NANDDir, user_path + NAND_DIR DIR_SEP);
     g_paths.emplace(UserPath::SysDataDir, user_path + SYSDATA_DIR DIR_SEP);
     // TODO: Put the logs in a better location for each OS
-    g_paths.emplace(UserPath::LogDir, user_path + LOG_DIR DIR_SEP);
+    g_paths.emplace(UserPath::LogDir, LOG_DIR DIR_SEP);
     g_paths.emplace(UserPath::CheatsDir, user_path + CHEATS_DIR DIR_SEP);
     g_paths.emplace(UserPath::DLLDir, user_path + DLL_DIR DIR_SEP);
     g_paths.emplace(UserPath::ShaderDir, user_path + SHADER_DIR DIR_SEP);
