diff --git a/src/common/common_paths.h b/src/common/common_paths.h
index de3db35f..83022e5b 100644
--- a/src/common/common_paths.h
+++ b/src/common/common_paths.h
@@ -25,11 +25,11 @@
 #define APPLE_EMU_DATA_DIR "Library" DIR_SEP "Application Support" DIR_SEP "Lime3DS"
 #define LEGACY_APPLE_EMU_DATA_DIR "Library" DIR_SEP "Application Support" DIR_SEP "Citra"
 // For compatibility with XDG paths.
-#define EMU_DATA_DIR "lime3ds-emu"
-#define LEGACY_EMU_DATA_DIR "citra-emu"
+#define EMU_DATA_DIR "lime3ds"
+#define LEGACY_EMU_DATA_DIR "lime3ds"
 #else
-#define EMU_DATA_DIR "lime3ds-emu"
-#define LEGACY_EMU_DATA_DIR "citra-emu"
+#define EMU_DATA_DIR "lime3ds"
+#define LEGACY_EMU_DATA_DIR "lime3ds"
 #endif
 #endif
 
@@ -44,7 +44,7 @@
 #define SDMC_DIR "sdmc"
 #define NAND_DIR "nand"
 #define SYSDATA_DIR "sysdata"
-#define LOG_DIR "log"
+#define LOG_DIR "/var/log"
 #define CHEATS_DIR "cheats"
 #define DLL_DIR "external_dlls"
 #define SHADER_DIR "shaders"
diff --git a/src/common/file_util.cpp b/src/common/file_util.cpp
index bf70d5cf..f6a191af 100644
--- a/src/common/file_util.cpp
+++ b/src/common/file_util.cpp
@@ -834,7 +834,7 @@ void SetUserPath(const std::string& path) {
     g_paths.emplace(UserPath::NANDDir, user_path + NAND_DIR DIR_SEP);
     g_paths.emplace(UserPath::SysDataDir, user_path + SYSDATA_DIR DIR_SEP);
     // TODO: Put the logs in a better location for each OS
-    g_paths.emplace(UserPath::LogDir, user_path + LOG_DIR DIR_SEP);
+    g_paths.emplace(UserPath::LogDir, LOG_DIR DIR_SEP);
     g_paths.emplace(UserPath::CheatsDir, user_path + CHEATS_DIR DIR_SEP);
     g_paths.emplace(UserPath::DLLDir, user_path + DLL_DIR DIR_SEP);
     g_paths.emplace(UserPath::ShaderDir, user_path + SHADER_DIR DIR_SEP);
