diff --git a/grub-core/normal/menu_text.c b/grub-core/normal/menu_text.c
index 9c383e64a..e99bf75b9 100644
--- a/grub-core/normal/menu_text.c
+++ b/grub-core/normal/menu_text.c
@@ -167,10 +167,10 @@ command-line or ESC to discard edits and return to the GRUB menu."),
     {
       char *msg_translated;
 
-      msg_translated = grub_xasprintf (_("Use the %C and %C keys to select which "
-					 "entry is highlighted."),
-				       GRUB_UNICODE_UPARROW,
-				       GRUB_UNICODE_DOWNARROW);
+	msg_translated = grub_xasprintf (_("Use the Volume Up and Volume Down buttons to select which "
+					  "device is highlighted."));
+					//GRUB_UNICODE_UPARROW,
+					//GRUB_UNICODE_DOWNARROW);
       if (!msg_translated)
 	return 0;
       ret += grub_print_message_indented_real (msg_translated, STANDARD_MARGIN,
@@ -183,17 +183,17 @@ command-line or ESC to discard edits and return to the GRUB menu."),
 	  if (nested)
 	    {
 	      ret += grub_print_message_indented_real
-		(_("Press enter to boot the selected OS, "
-		   "`e' to edit the commands before booting "
-		   "or `c' for a command-line. ESC to return previous menu."),
+		(_("Press the Power button to boot the selected Device, "
+		   "the selected device will be set as the default "
+		   "after the first full boot."),
 		 STANDARD_MARGIN, STANDARD_MARGIN, term, dry_run);
 	    }
 	  else
 	    {
 	      ret += grub_print_message_indented_real
-		(_("Press enter to boot the selected OS, "
-		   "`e' to edit the commands before booting "
-		   "or `c' for a command-line."),
+		(_("Press the Power button to boot the selected Device, "
+		   "the selected device will be set as the default "
+		   "after the first full boot."),
 		 STANDARD_MARGIN, STANDARD_MARGIN, term, dry_run);
 	    }
 	}
