diff --git a/grub-core/normal/menu.c b/grub-core/normal/menu.c
index b946c834d..352f03117 100644
--- a/grub-core/normal/menu.c
+++ b/grub-core/normal/menu.c
@@ -581,7 +581,7 @@ run_menu (grub_menu_t menu, int nested, int *auto_boot, int *notify_boot)
   int timeout;
   enum timeout_style timeout_style;
 
-  *notify_boot = 1;
+  *notify_boot = 0;
 
   default_entry = get_entry_number (menu, "default");
 
