diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1a36617..b716c6a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,10 +25,10 @@ option(USE_SDL2 "Use SDL2 instead of SDL1" ON)
 option(WITH_SYSTEM_SDL_GFX "Use system SDL_gfx library" OFF)
 option(WITH_SYSTEM_SDL_TTF "Use system SDL_ttf library" OFF)
 
-set(RES_DIR \"res/\" CACHE STRING "Resources directory")
+set(RES_DIR \"/usr/share/commander/res/\" CACHE STRING "Resources directory")
 set(FILE_SYSTEM \"/dev/mmcblk1p1\" CACHE STRING "Default filesystem (for Disk info)")
-set(PATH_DEFAULT "" CACHE STRING "Default path for the left pane (default: getenv(\"PWD\")")
-set(PATH_DEFAULT_RIGHT "" CACHE STRING "Default path for the right pane (default: getenv(\"HOME\")")
+set(PATH_DEFAULT \"/storage/.config/\" CACHE STRING "Default path for the left pane (default: getenv(\"HOME\")")
+set(PATH_DEFAULT_RIGHT \"/storage/.config/\" CACHE STRING "Default path for the right pane (default: getenv(\"HOME\")")
 set(AUTOSCALE 1 CACHE STRING "Auto-detect resolution (maximize on startup if there is a windowing system)")
 set(AUTOSCALE_DPI 1 CACHE STRING "Auto-detect DPI scaling factors")
 set(WITH_SYSTEM_SDL_TTF OFF CACHE BOOL "Use system sdl_ttf")
