--- a/CMakeLists.txt	2025-04-28 18:01:01.129568626 +0200
+++ b/CMakeLists.txt	2025-04-28 18:00:27.891870910 +0200
@@ -12,7 +12,9 @@
 
 add_subdirectory(thirdparty)
 
+option(ARES_BUILD_SOURCERY_ONLY "Only build sourcery tool for cross-compiling" OFF)
 add_subdirectory(nall/nall)
+if (NOT ARES_BUILD_SOURCERY_ONLY)
 add_subdirectory(libco)
 add_subdirectory(ruby)
 add_subdirectory(hiro)
@@ -28,6 +30,7 @@
 add_subdirectory(ares)
 add_subdirectory(mia)
 add_subdirectory(desktop-ui)
+endif()
 
 option(ARES_BUILD_OPTIONAL_TARGETS "Include supplemental tools and tests" OFF)
 
