diff -rupN linux.orig/drivers/gpu/drm/drm_panel_orientation_quirks.c linux/drivers/gpu/drm/drm_panel_orientation_quirks.c
--- linux.orig/drivers/gpu/drm/drm_panel_orientation_quirks.c	2024-03-23 01:52:13.930770363 +0000
+++ linux/drivers/gpu/drm/drm_panel_orientation_quirks.c	2024-03-23 01:53:34.905357358 +0000
@@ -461,7 +461,7 @@ EXPORT_SYMBOL(drm_get_panel_orientation_
 /* There are no quirks for non x86 devices yet */
 int drm_get_panel_orientation_quirk(int width, int height)
 {
-	return DRM_MODE_PANEL_ORIENTATION_UNKNOWN;
+	return DRM_MODE_PANEL_ORIENTATION_LEFT_UP;
 }
 EXPORT_SYMBOL(drm_get_panel_orientation_quirk);
 
