diff --git a/Makefile b/Makefile
index ce2617b1bb1b..92d1f4f4e760 100644
--- a/Makefile
+++ b/Makefile
@@ -771,6 +771,7 @@ KBUILD_CFLAGS	+= $(call cc-disable-warning,frame-address,)
 KBUILD_CFLAGS	+= $(call cc-disable-warning, format-truncation)
 KBUILD_CFLAGS	+= $(call cc-disable-warning, format-overflow)
 KBUILD_CFLAGS	+= $(call cc-disable-warning, address-of-packed-member)
+KBUILD_CFLAGS	+= -Wno-enum-int-mismatch -Wno-dangling-pointer
 
 ifdef CONFIG_CC_OPTIMIZE_FOR_PERFORMANCE
 KBUILD_CFLAGS += -O2
