diff -rupN mgba-b2564482c86378581a7a43ef4e254b2a75167bc7.orig/src/platform/libretro/libretro.c mgba-b2564482c86378581a7a43ef4e254b2a75167bc7/src/platform/libretro/libretro.c
--- mgba-b2564482c86378581a7a43ef4e254b2a75167bc7.orig/src/platform/libretro/libretro.c	2024-06-01 00:52:52.719157171 +0000
+++ mgba-b2564482c86378581a7a43ef4e254b2a75167bc7/src/platform/libretro/libretro.c	2024-06-01 09:48:57.732371849 +0000
@@ -1722,12 +1722,23 @@ void retro_run(void) {
 #endif
 
 	if (rumbleCallback) {
+		FILE *file;
 		if (rumbleUp) {
-			rumbleCallback(0, RETRO_RUMBLE_STRONG, rumbleUp * 0xFFFF / (rumbleUp + rumbleDown));
-			rumbleCallback(0, RETRO_RUMBLE_WEAK, rumbleUp * 0xFFFF / (rumbleUp + rumbleDown));
+			if ((file = fopen("/sys/class/pwm/pwmchip1/pwm0/enable", "r+"))) {
+      	fputs("1", file);
+      	fclose(file);
+      } else {
+				rumbleCallback(0, RETRO_RUMBLE_STRONG, rumbleUp * 0xFFFF / (rumbleUp + rumbleDown));
+				rumbleCallback(0, RETRO_RUMBLE_WEAK, rumbleUp * 0xFFFF / (rumbleUp + rumbleDown));
+			}
 		} else {
-			rumbleCallback(0, RETRO_RUMBLE_STRONG, 0);
-			rumbleCallback(0, RETRO_RUMBLE_WEAK, 0);
+			if ((file = fopen("/sys/class/pwm/pwmchip1/pwm0/enable", "r+"))) {
+      	fputs("0", file);
+      	fclose(file);
+      } else {
+				rumbleCallback(0, RETRO_RUMBLE_STRONG, 0);
+				rumbleCallback(0, RETRO_RUMBLE_WEAK, 0);
+			}
 		}
 		rumbleUp = 0;
 		rumbleDown = 0;
