--- retroarch-v1.19.1/gfx/drivers_context/drm_go2_ctx.c.orig	2024-12-09 21:57:50.536644828 +0000
+++ retroarch-v1.19.1/gfx/drivers_context/drm_go2_ctx.c	2024-12-09 22:02:01.614410451 +0000
@@ -51,6 +51,8 @@
 #include "../common/egl_common.h"
 #endif
 
+#include "../common/gl_common.h"
+
 #ifdef HAVE_MENU
 #include "../../menu/menu_driver.h"
 #endif
--- retroarch-v1.19.1/gfx/drivers/oga_gfx.c.orig	2024-12-09 22:02:28.352574596 +0000
+++ retroarch-v1.19.1/gfx/drivers/oga_gfx.c	2024-12-09 22:02:38.135269709 +0000
@@ -21,7 +21,7 @@
 
 #include <fcntl.h>
 #include <rga/RgaApi.h>
-#include <rga/RockchipRgaMacro.h>
+#include <rga/drmrga.h>
 #include <xf86drm.h>
 #include <xf86drmMode.h>
 #include <drm/drm_fourcc.h>
