diff --git a/config.def.h b/config.def.h
index 52f3bcb5fb..5980cac08f 100644
--- a/config.def.h
+++ b/config.def.h
@@ -579,7 +579,7 @@
 
 /* Automatically disable overlays when a
  * controller is connected in port 1 */
-#define DEFAULT_OVERLAY_HIDE_WHEN_GAMEPAD_CONNECTED false
+#define DEFAULT_OVERLAY_HIDE_WHEN_GAMEPAD_CONNECTED true
 
 #define DEFAULT_OVERLAY_SHOW_MOUSE_CURSOR true
 
@@ -931,9 +931,9 @@
 
 #if defined(HAVE_OVERLAY)
 #if defined(RARCH_MOBILE)
-#define DEFAULT_OVERLAY_SHOW_INPUTS OVERLAY_SHOW_INPUT_TOUCHED
+#define DEFAULT_OVERLAY_SHOW_INPUTS OVERLAY_SHOW_INPUT_NONE
 #else
-#define DEFAULT_OVERLAY_SHOW_INPUTS OVERLAY_SHOW_INPUT_PHYSICAL
+#define DEFAULT_OVERLAY_SHOW_INPUTS OVERLAY_SHOW_INPUT_NONE
 #endif
 #endif
 
diff --git a/configuration.c b/configuration.c
index 7eb6c08e1e..da1728f42e 100644
--- a/configuration.c
+++ b/configuration.c
@@ -2106,7 +2106,7 @@ static struct config_bool_setting *populate_settings_bool(
    SETTING_BOOL("input_overlay_enable_autopreferred", &settings->bools.input_overlay_enable_autopreferred, true, DEFAULT_OVERLAY_ENABLE_AUTOPREFERRED, false);
    SETTING_BOOL("input_overlay_behind_menu",     &settings->bools.input_overlay_behind_menu, true, DEFAULT_OVERLAY_BEHIND_MENU, false);
    SETTING_BOOL("input_overlay_hide_in_menu",    &settings->bools.input_overlay_hide_in_menu, true, DEFAULT_OVERLAY_HIDE_IN_MENU, false);
-   SETTING_BOOL("input_overlay_hide_when_gamepad_connected", &settings->bools.input_overlay_hide_when_gamepad_connected, true, DEFAULT_OVERLAY_HIDE_WHEN_GAMEPAD_CONNECTED, false);
+   SETTING_BOOL("input_overlay_hide_when_gamepad_connected", &settings->bools.input_overlay_hide_when_gamepad_connected, true, DEFAULT_OVERLAY_HIDE_WHEN_GAMEPAD_CONNECTED, true);
    SETTING_BOOL("input_overlay_show_mouse_cursor", &settings->bools.input_overlay_show_mouse_cursor, true, DEFAULT_OVERLAY_SHOW_MOUSE_CURSOR, false);
    SETTING_BOOL("input_overlay_auto_rotate",     &settings->bools.input_overlay_auto_rotate, true, DEFAULT_OVERLAY_AUTO_ROTATE, false);
    SETTING_BOOL("input_overlay_auto_scale",      &settings->bools.input_overlay_auto_scale, true, DEFAULT_INPUT_OVERLAY_AUTO_SCALE, false);
diff --git a/menu/menu_setting.c b/menu/menu_setting.c
index 7513be0697..802a95bc60 100644
--- a/menu/menu_setting.c
+++ b/menu/menu_setting.c
@@ -17466,7 +17466,7 @@ static bool setting_append_list(
                SD_FLAG_NONE
                );
          (*list)[list_info->index - 1].change_handler = overlay_enable_toggle_change_handler;
-
+/*
          CONFIG_BOOL(
                list, list_info,
                &settings->bools.input_overlay_hide_when_gamepad_connected,
@@ -17483,7 +17483,7 @@ static bool setting_append_list(
                SD_FLAG_NONE
                );
          (*list)[list_info->index - 1].change_handler = overlay_enable_toggle_change_handler;
-
+*/
          CONFIG_UINT(
                list, list_info,
                &settings->uints.input_overlay_show_inputs,
diff --git a/runloop.c b/runloop.c
index 0f34c0bec2..171c05d3ab 100644
--- a/runloop.c
+++ b/runloop.c
@@ -5586,7 +5586,7 @@ static enum runloop_state_enum runloop_check_state(
       unsigned video_driver_width                    = video_st->width;
       unsigned video_driver_height                   = video_st->height;
       bool check_next_rotation                       = true;
-      bool input_overlay_hide_when_gamepad_connected = settings->bools.input_overlay_hide_when_gamepad_connected;
+      bool input_overlay_hide_when_gamepad_connected = true;
       bool input_overlay_auto_rotate                 = settings->bools.input_overlay_auto_rotate;
 
       /* Check whether overlay should be hidden
