diff --git a/config.def.h b/config.def.h
index 52f3bcb5fb..5980cac08f 100644
--- a/config.def.h
+++ b/config.def.h
@@ -579,7 +579,7 @@
 
 /* Automatically disable overlays when a
  * controller is connected in port 1 */
-#define DEFAULT_OVERLAY_HIDE_WHEN_GAMEPAD_CONNECTED false
+#define DEFAULT_OVERLAY_HIDE_WHEN_GAMEPAD_CONNECTED true
 
 #define DEFAULT_OVERLAY_SHOW_MOUSE_CURSOR true
 
@@ -931,9 +931,9 @@
 
 #if defined(HAVE_OVERLAY)
 #if defined(RARCH_MOBILE)
-#define DEFAULT_OVERLAY_SHOW_INPUTS OVERLAY_SHOW_INPUT_TOUCHED
+#define DEFAULT_OVERLAY_SHOW_INPUTS OVERLAY_SHOW_INPUT_NONE
 #else
-#define DEFAULT_OVERLAY_SHOW_INPUTS OVERLAY_SHOW_INPUT_PHYSICAL
+#define DEFAULT_OVERLAY_SHOW_INPUTS OVERLAY_SHOW_INPUT_NONE
 #endif
 #endif
 
