--- a/board/anbernic/rgxx3_rk3566/rgxx3-rk3566.c
+++ b/board/anbernic/rgxx3_rk3566/rgxx3-rk3566.c
@@ -60,6 +60,7 @@
 	RK2023,
 	RGARCD,
 	RGB10MAX3,
+	RGB20PRO,
 	/* Devices with duplicate ADC value */
 	RG353PS,
 	RG353VS,
@@ -132,6 +133,14 @@
 		.detect_panel = 0,
 		.detect_regulator = 1,
 	},
+	[RGB20PRO] = {
+		.adc_value = 245, /* Observed average from device */
+		.board = "rk3566-powkiddy-rgb20-pro",
+		.board_name = "Powkiddy RGB20 Pro",
+		.fdtfile = DTB_DIR "rk3566-powkiddy-rgb20-pro.dtb",
+		.detect_panel = 0,
+		.detect_regulator = 1,
+	},
 	/* Devices with duplicate ADC value */
 	[RG353PS] = {
 		.adc_value = 860, /* Observed average from device */
