diff --git a/Source/ui_libretro/CMakeLists.txt b/Source/ui_libretro/CMakeLists.txt
index cc195c1a..cf1acb44 100644
--- a/Source/ui_libretro/CMakeLists.txt
+++ b/Source/ui_libretro/CMakeLists.txt
@@ -50,7 +50,7 @@ elseif(TARGET_PLATFORM_IOS)
 	list(APPEND PROJECT_LIBS "-ObjC -lsqlite3 -framework OpenGLES")
 elseif(TARGET_PLATFORM_UNIX)
 	list(APPEND PROJECT_LIBS "-static-libgcc")
-	list(APPEND PROJECT_LIBS "-static-libstdc++")
+#	list(APPEND PROJECT_LIBS "-static-libstdc++")
 endif()
 
 if(CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 8 AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9)
