diff -rupN flycast2021-lr.orig/core/libretro-common/glsm/glsm.c flycast2021-lr/core/libretro-common/glsm/glsm.c
--- flycast2021-lr.orig/core/libretro-common/glsm/glsm.c	2024-07-08 19:56:17.217751724 +0000
+++ flycast2021-lr/core/libretro-common/glsm/glsm.c	2024-07-08 20:00:09.491339516 +0000
@@ -2747,7 +2747,7 @@ static void glsm_state_setup(void)
 #endif
 
 #ifdef CORE
-   glGenVertexArrays(1, &gl_state.vao);
+   rglGenVertexArrays(1, &gl_state.vao);
 #endif
 }
 
@@ -2755,7 +2755,7 @@ static void glsm_state_bind(void)
 {
    unsigned i;
 #ifdef CORE
-   glBindVertexArray(gl_state.vao);
+   rglBindVertexArray(gl_state.vao);
 #endif
    glBindBuffer(GL_ARRAY_BUFFER, gl_state.array_buffer);
 
@@ -2871,7 +2871,7 @@ static void glsm_state_unbind(void)
 {
    unsigned i;
 #ifdef CORE
-   glBindVertexArray(0);
+   rglBindVertexArray(0);
 #endif
    for (i = 0; i < SGL_CAP_MAX; i ++)
    {
