From aed0afd4578c66194b85587f6904ee6457487d74 Mon Sep 17 00:00:00 2001
From: John Williams <porschemad911@gmail.com>
Date: Wed, 14 Jan 2026 22:18:41 +1100
Subject: [PATCH 2/3] Battery name

---
 src/battery.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/battery.cpp b/src/battery.cpp
index e5d311b..9cf8e5b 100644
--- a/src/battery.cpp
+++ b/src/battery.cpp
@@ -13,7 +13,7 @@ void BatteryStats::numBattery() {
     fs::path path("/sys/class/power_supply/");
     for (auto& p : fs::directory_iterator(path)) {
         string fileName = p.path().filename();
-        if (fileName.find("BAT") != std::string::npos) {
+        if (fileName.find("battery") != std::string::npos) {
             battPath[batteryCount] = p.path();
             batteryCount += 1;
         }
-- 
2.47.3

