diff --git a/src/hud_elements.cpp b/src/hud_elements.cpp
index 999ed7d..449a574 100644
--- a/src/hud_elements.cpp
+++ b/src/hud_elements.cpp
@@ -784,7 +784,7 @@ void HudElements::procmem()
         return;
 
     ImguiNextColumnFirstItem();
-    HUDElements.TextColored(HUDElements.colors.ram, "PMEM");
+    HUDElements.TextColored(HUDElements.colors.ram, "RAM");
     ImguiNextColumnOrNewRow();
     right_aligned_text(HUDElements.colors.text, HUDElements.ralign_width, "%.1f", format_units(proc_mem_resident, unit));
     ImGui::SameLine(0, 1.0f);
