diff --git a/cmake/FindInstallConfigPath.cmake b/cmake/FindInstallConfigPath.cmake
index 4ea6552..fbbf5ed 100644
--- a/cmake/FindInstallConfigPath.cmake
+++ b/cmake/FindInstallConfigPath.cmake
@@ -6,7 +6,7 @@
 #
 
 find_package(Qt6 ${LXQTBT_QT_MINIMUM_VERSION} REQUIRED)
-find_package(Qt6CoreTools ${LXQTBT_QT_MINIMUM_VERSION} REQUIRED)
+#find_package(Qt6CoreTools ${LXQTBT_QT_MINIMUM_VERSION} REQUIRED)
 
 macro(print_set_lxqt_etc_xdg_dir_info)
     message(STATUS "You can set it manually with -DLXQT_ETC_XDG_DIR=<value>")
