From a307d812355589d2792b61c7307c461f3c86cf11 Mon Sep 17 00:00:00 2001
From: John Williams <porschemad911@gmail.com>
Date: Mon, 3 Aug 2026 21:55:32 +1000
Subject: [PATCH] batteryplus capacity

---
 src/battery.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/battery.cpp b/src/battery.cpp
index 16b0355..bb20f51 100644
--- a/src/battery.cpp
+++ b/src/battery.cpp
@@ -49,7 +49,7 @@ float BatteryStats::getPercent()
         string charge_full = syspath + "/charge_full";
         string energy_now = syspath + "/energy_now";
         string energy_full = syspath + "/energy_full";
-        string capacity = syspath + "/capacity";
+        string capacity = "/tmp/battery.percent";
 
         if (fs::exists(charge_now)) {
             std::ifstream input(charge_now);
-- 
2.47.3

