From 2e7a38a7521034098000c057fd8dce4558dfbfa9 Mon Sep 17 00:00:00 2001
From: John Williams <porschemad911@gmail.com>
Date: Thu, 31 Oct 2024 22:25:18 +1100
Subject: [PATCH 07/42] arm64: meson: g12-common: add dual panfrost/mali to dts

---
 .../boot/dts/amlogic/meson-g12-common.dtsi    | 25 +++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/meson-g12-common.dtsi b/arch/arm64/boot/dts/amlogic/meson-g12-common.dtsi
index d08c97797010..2506f3314ffe 100644
--- a/arch/arm64/boot/dts/amlogic/meson-g12-common.dtsi
+++ b/arch/arm64/boot/dts/amlogic/meson-g12-common.dtsi
@@ -2477,6 +2477,7 @@ dwc3: usb@ff500000 {
 			};
 		};
 
+		// Mali GPU - Panfrost
 		mali: gpu@ffe40000 {
 			compatible = "amlogic,meson-g12a-mali", "arm,mali-bifrost";
 			reg = <0x0 0xffe40000 0x0 0x40000>;
@@ -2489,6 +2490,30 @@ mali: gpu@ffe40000 {
 			resets = <&reset RESET_DVALIN_CAPB3>, <&reset RESET_DVALIN>;
 			operating-points-v2 = <&gpu_opp_table>;
 			#cooling-cells = <2>;
+			status = "disabled";
+		};
+
+		// Mali GPU - Vulkan
+		mali_vulkan: gpu_mali_vulkan@ffe40000 {
+			compatible = "arm,mali-midgard";
+			reg = <0x0 0xffe40000 0x0 0x40000>,
+                    <0 0xFFD01000 0 0x01000>,
+                    <0 0xFF800000 0 0x01000>,
+                    <0 0xFF63c000 0 0x01000>,
+                    <0 0xFFD01000 0 0x01000>;
+ 			interrupt-parent = <&gic>;
+			interrupts = <GIC_SPI 160 IRQ_TYPE_LEVEL_HIGH>,
+ 				     <GIC_SPI 161 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 162 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "GPU", "MMU", "JOB";
+ 			clocks = <&clkc CLKID_MALI>;
+			clock-names = "clk_mali";
+ 			resets = <&reset RESET_DVALIN_CAPB3>, <&reset RESET_DVALIN>;
+ 			operating-points-v2 = <&gpu_opp_table>;
+ 			#cooling-cells = <2>;
+			system-coherency = <0>;
+			power_policy = "always_on";
+			status = "okay";
 		};
 	};
 
-- 
2.34.1

