--- a/src/lib/util/corestr.h
+++ b/src/lib/util/corestr.h
@@ -17,6 +17,7 @@
 #include <string_view>
 
 #include <cstring>
+#include <cstdint>
 
 
 /***************************************************************************
--- a/3rdparty/sol2/sol/sol.hpp
+++ b/3rdparty/sol2/sol/sol.hpp
@@ -6056,7 +6056,8 @@
 			static_assert(std::is_constructible<T, Args&&...>::value, "T must be constructible with Args");
 
 			*this = nullopt;
-			this->construct(std::forward<Args>(args)...);
+			m_value = std::addressof(std::forward<Args>(args)...);
+			return *m_value;
 		}
 
 		/// Swaps this optional with the other.
