diff --git a/external/CMakeLists.txt b/external/CMakeLists.txt
index 72b4010b..9509832a 100644
--- a/external/CMakeLists.txt
+++ b/external/CMakeLists.txt
@@ -331,9 +331,9 @@ target_include_directories(libatrac9 PUBLIC LibAtrac9/C/src)
 option(BUILD_SHARED_LIBS "Build shared library" OFF)
 option(XXHASH_BUILD_XXHSUM "Build the xxhsum binary" OFF)
 option(XXH_X86DISPATCH_ALLOW_AVX "Allow building XXH3 with AVX even if it crashes on SSE2-Only CPUs" OFF)
-if(NOT APPLE)
-	set(DISPATCH 1)
-endif()
+#if(NOT APPLE)
+#	set(DISPATCH 1)
+#endif()
 add_subdirectory(xxHash/cmake_unofficial EXCLUDE_FROM_ALL)
 if(XXH_X86DISPATCH_ALLOW_AVX)
 	target_compile_definitions(xxhash PRIVATE XXH_X86DISPATCH_ALLOW_AVX)
