diff -rupN libmali.orig/meson.build libmali/meson.build
--- libmali.orig/meson.build	2024-06-26 18:37:20.569979221 +0000
+++ libmali/meson.build	2024-06-26 18:38:04.922988416 +0000
@@ -332,10 +332,10 @@ if get_option('with-overlay')
 endif
 
 # Install firmwares
-if gpu == 'valhall-g610'
-  install_data('firmware/g610/mali_csffw.bin',
-    install_dir : get_option('firmware-dir'))
-endif
+#if gpu == 'valhall-g610'
+#  install_data('firmware/g610/mali_csffw.bin',
+#    install_dir : get_option('firmware-dir'))
+#endif
 
 if vendor_package
   # Install vendor ld config
