diff --git a/src/ws/swapchain_window_system.cpp b/src/ws/swapchain_window_system.cpp
index 9f9fba3..99b583c 100644
--- a/src/ws/swapchain_window_system.cpp
+++ b/src/ws/swapchain_window_system.cpp
@@ -193,7 +193,7 @@ ManagedResource<vk::SwapchainKHR> SwapchainWindowSystem::create_vk_swapchain()
     if (!(surface_caps.supportedCompositeAlpha &
           vk::CompositeAlphaFlagBitsKHR::eOpaque))
     {
-        throw std::runtime_error("Opaque not supported");
+        //throw std::runtime_error("Opaque not supported");
     }

     if (!vulkan->physical_device().getSurfaceSupportKHR(
diff --git a/src/ws/wayland_native_system.cpp b/src/ws/wayland_native_system.cpp
index 816a7b9..680849e 100644
--- a/src/ws/wayland_native_system.cpp
+++ b/src/ws/wayland_native_system.cpp
@@ -156,8 +156,8 @@ uint32_t WaylandNativeSystem::get_presentation_queue_family_index(vk::PhysicalDe

     for (auto i = 0u; i < queue_families.size(); ++i)
     {
-        if (queue_families[i].queueCount > 0 &&
-            pd.getWaylandPresentationSupportKHR(i, display))
+        //if (queue_families[i].queueCount > 0 &&
+        //    pd.getWaylandPresentationSupportKHR(i, display))
         {
             return i;
         }
