diff --git a/hw/xbox/xid.c b/hw/xbox/xid.c
index c37142bc0d..3f2b3d0526 100644
--- a/hw/xbox/xid.c
+++ b/hw/xbox/xid.c
@@ -64,10 +64,10 @@ void update_input(USBXIDGamepadState *s)
     xemu_input_update_controller(state);
 
     const int button_map_analog[6][2] = {
-        { GAMEPAD_A,     CONTROLLER_BUTTON_A     },
-        { GAMEPAD_B,     CONTROLLER_BUTTON_B     },
-        { GAMEPAD_X,     CONTROLLER_BUTTON_X     },
-        { GAMEPAD_Y,     CONTROLLER_BUTTON_Y     },
+        { GAMEPAD_A,     CONTROLLER_BUTTON_B     },
+        { GAMEPAD_B,     CONTROLLER_BUTTON_A     },
+        { GAMEPAD_X,     CONTROLLER_BUTTON_Y     },
+        { GAMEPAD_Y,     CONTROLLER_BUTTON_X     },
         { GAMEPAD_BLACK, CONTROLLER_BUTTON_BLACK },
         { GAMEPAD_WHITE, CONTROLLER_BUTTON_WHITE },
     };
