diff --git a/Core/Config.cpp b/Core/Config.cpp
index fac682fb84..e1db1d288a 100644
--- a/Core/Config.cpp
+++ b/Core/Config.cpp
@@ -1091,7 +1091,7 @@ void Config::Reload() {
 void Config::UpdateIniLocation(const char *iniFileName, const char *controllerIniFilename) {
 	const bool useIniFilename = iniFileName != nullptr && strlen(iniFileName) > 0;
 	const char *ppssppIniFilename = IsVREnabled() ? "ppssppvr.ini" : "ppsspp.ini";
-	iniFilename_ = FindConfigFile(useIniFilename ? iniFileName : ppssppIniFilename);
+	iniFilename_ = Path("/storage/.config/ppsspp/PSP/SYSTEM") / ppssppIniFilename;
 	const bool useControllerIniFilename = controllerIniFilename != nullptr && strlen(controllerIniFilename) > 0;
 	const char *controlsIniFilename = IsVREnabled() ? "controlsvr.ini" : "controls.ini";
 	controllerIniFilename_ = FindConfigFile(useControllerIniFilename ? controllerIniFilename : controlsIniFilename);
diff --git a/Core/SaveState.cpp b/Core/SaveState.cpp
index 49ac73c0af..a2c254c18b 100644
--- a/Core/SaveState.cpp
+++ b/Core/SaveState.cpp
@@ -532,7 +532,7 @@ namespace SaveState
 
 	Path GenerateSaveSlotFilename(const Path &gameFilename, int slot, const char *extension)
 	{
-		std::string filename = StringFromFormat("%s_%d.%s", GenerateFullDiscId(gameFilename).c_str(), slot, extension);
+		std::string filename = StringFromFormat("%s_%d.%s", gameFilename.WithReplacedExtension("").GetFilename().c_str(), slot, extension);
 		return GetSysDirectory(DIRECTORY_SAVESTATE) / filename;
 	}
 
diff --git a/Core/System.cpp b/Core/System.cpp
index 155c3bbba3..8fbb29dd19 100644
--- a/Core/System.cpp
+++ b/Core/System.cpp
@@ -651,14 +651,8 @@ std::string PSP_GetLoading() {
 }
 
 Path GetSysDirectory(PSPDirectories directoryType) {
-	const Path &memStickDirectory = g_Config.memStickDirectory;
-	Path pspDirectory;
-	if (!strcasecmp(memStickDirectory.GetFilename().c_str(), "PSP")) {
-		// Let's strip this off, to easily allow choosing a root directory named "PSP" on Android.
-		pspDirectory = memStickDirectory;
-	} else {
-		pspDirectory = memStickDirectory / "PSP";
-	}
+	Path memStickDirectory = Path("/storage/roms/psp/");
+ 	Path pspDirectory = Path("/storage/.config/ppsspp/PSP/");
 
 	switch (directoryType) {
 	case DIRECTORY_PSP:
@@ -668,9 +662,9 @@ Path GetSysDirectory(PSPDirectories directoryType) {
 	case DIRECTORY_GAME:
 		return pspDirectory / "GAME";
 	case DIRECTORY_SAVEDATA:
-		return pspDirectory / "SAVEDATA";
+		return Path("/storage/roms/psp/");
 	case DIRECTORY_SCREENSHOT:
-		return pspDirectory / "SCREENSHOT";
+		return Path("/storage/roms/screenshots/");
 	case DIRECTORY_SYSTEM:
 		return pspDirectory / "SYSTEM";
 	case DIRECTORY_PAUTH:
@@ -680,7 +674,7 @@ Path GetSysDirectory(PSPDirectories directoryType) {
 	case DIRECTORY_DUMP:
 		return pspDirectory / "SYSTEM/DUMP";
 	case DIRECTORY_SAVESTATE:
-		return pspDirectory / "PPSSPP_STATE";
+		return Path("/storage/roms/savestates/psp/ppsspp-sa");
 	case DIRECTORY_CACHE:
 		return pspDirectory / "SYSTEM/CACHE";
 	case DIRECTORY_TEXTURES:
@@ -702,11 +696,11 @@ Path GetSysDirectory(PSPDirectories directoryType) {
 		return pspDirectory / "themes";
 
 	case DIRECTORY_MEMSTICK_ROOT:
-		return g_Config.memStickDirectory;
+		return memStickDirectory;
 	// Just return the memory stick root if we run into some sort of problem.
 	default:
 		ERROR_LOG(Log::FileSystem, "Unknown directory type.");
-		return g_Config.memStickDirectory;
+		return pspDirectory;
 	}
 }
 
diff --git a/SDL/SDLJoystick.cpp b/SDL/SDLJoystick.cpp
index a0d206ee21..3bdca780b6 100644
--- a/SDL/SDLJoystick.cpp
+++ b/SDL/SDLJoystick.cpp
@@ -25,7 +25,7 @@ SDLJoystick::SDLJoystick(bool init_SDL ) : registeredAsEventHandler(false) {
 		SDL_Init(SDL_INIT_JOYSTICK | SDL_INIT_VIDEO | SDL_INIT_GAMECONTROLLER);
 	}
 
-	const char *dbPath = "gamecontrollerdb.txt";
+	const char *dbPath = "/storage/.config/SDL-GameControllerDB/gamecontrollerdb.txt";
 	cout << "loading control pad mappings from " << dbPath << ": ";
 
 	size_t size;
@@ -34,7 +34,7 @@ SDLJoystick::SDLJoystick(bool init_SDL ) : registeredAsEventHandler(false) {
 		SDL_RWops *rw = SDL_RWFromConstMem(mappingData, size);
 		// 1 to free the rw after use
 		if (SDL_GameControllerAddMappingsFromRW(rw, 1) == -1) {
-			cout << "Failed to read mapping data - corrupt?" << endl;
+			cout << "/storage/.config/SDL-GameControllerDB/gamecontrollerdb.txt missing" << endl;
 		}
 		delete[] mappingData;
 	} else {
diff --git a/UI/NativeApp.cpp b/UI/NativeApp.cpp
index 272ff35e16..91a58b5691 100644
--- a/UI/NativeApp.cpp
+++ b/UI/NativeApp.cpp
@@ -390,7 +390,7 @@ void NativeInit(int argc, const char *argv[], const char *savegame_dir, const ch
 	g_VFS.Register("", new DirectoryReader(File::GetExeDirectory()));
 	g_VFS.Register("", new DirectoryReader(Path("/usr/local/share/ppsspp/assets")));
 	g_VFS.Register("", new DirectoryReader(Path("/usr/local/share/games/ppsspp/assets")));
-	g_VFS.Register("", new DirectoryReader(Path("/usr/share/ppsspp/assets")));
+	g_VFS.Register("", new DirectoryReader(Path("/storage/.config/ppsspp/assets")));
 	g_VFS.Register("", new DirectoryReader(Path("/usr/share/games/ppsspp/assets")));
 #endif
 
@@ -495,7 +495,7 @@ void NativeInit(int argc, const char *argv[], const char *savegame_dir, const ch
 	else // Just in case
 		config = "./config";
 
-	g_Config.memStickDirectory = Path(config) / "ppsspp";
+	g_Config.memStickDirectory = Path("/storage/roms/psp/");
 	g_Config.flash0Directory = File::GetExeDirectory() / "assets/flash0";
 	if (getenv("HOME") != nullptr) {
 		g_Config.defaultCurrentDirectory = Path(getenv("HOME"));
