From 8e268b3e6501eb9dee62ae7a5d3982210ccd4212 Mon Sep 17 00:00:00 2001
From: spycat88 <spycat88@users.noreply.github.com>
Date: Sun, 16 Nov 2025 15:19:10 +0000
Subject: [PATCH] fix building with gcc15

---
 src/config.h | 2 +-
 src/main.c   | 5 ++++-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/config.h b/src/config.h
index 0a65907..caa90d5 100644
--- a/src/config.h
+++ b/src/config.h
@@ -60,7 +60,7 @@ typedef struct config_params_s {
 } config_params_s;
 
 
-config_params_s init_config();
+config_params_s init_config(char *filename);
 void read_config(config_params_s *conf);
 void read_audio_config(const ini_t *ini, config_params_s *conf);
 void read_graphics_config(const ini_t *ini, config_params_s *conf);
diff --git a/src/main.c b/src/main.c
index d7d60db..09cbd38 100644
--- a/src/main.c
+++ b/src/main.c
@@ -24,7 +24,10 @@ enum state run = WAIT_FOR_DEVICE;
 uint8_t need_display_reset = 0;
 
 // Handles CTRL+C / SIGINT
-void intHandler() { run = QUIT; }
+void intHandler(int sig) {
+    (void)sig;
+    run = QUIT;
+}
 
 void close_serial_port() { disconnect(); }
 
-- 
2.34.1

