--- a/product/kernel/drivers/gpu/arm/midgard/mali_kbase_core_linux.c
+++ b/product/kernel/drivers/gpu/arm/midgard/mali_kbase_core_linux.c
@@ -440,9 +440,9 @@ static struct kbase_device *to_kbase_device(struct device *dev)
 static int get_irqs(struct kbase_device *kbdev, struct platform_device *pdev)
 {
 	int i;
-	static const char *const irq_names_caps[] = { "JOB", "MMU", "GPU" };
+	static const char *const irq_names[] = { "job", "mmu", "gpu" };
 
-	for (i = 0; i < ARRAY_SIZE(irq_names_caps); i++) {
+	for (i = 0; i < ARRAY_SIZE(irq_names); i++) {
 		struct irq_data *irqdata;
 		int irq;
 
@@ -452,11 +452,11 @@ static int get_irqs(struct kbase_device *kbdev, struct platform_device *pdev)
 		 * first then try using Lower case names. If both attempts fail then
 		 * we assume there is no IRQ resource specified for the GPU.
 		 */
-		irq = platform_get_irq_byname(pdev, irq_names_caps[i]);
+		irq = platform_get_irq_byname(pdev, irq_names[i]);
 		if (irq < 0) {
-			static const char *const irq_names[] = { "job", "mmu", "gpu" };
+			static const char *const irq_names_caps[] = { "JOB", "MMU", "GPU" };
 
-			irq = platform_get_irq_byname(pdev, irq_names[i]);
+			irq = platform_get_irq_byname(pdev, irq_names_caps[i]);
 		}
 
 		if (irq < 0)
