diff --git a/start_xemu.sh b/start_xemu.sh
old mode 100755
new mode 100644
index 55246296d0..862465adc0
--- a/start_xemu.sh
+++ b/start_xemu.sh
@@ -15,8 +15,17 @@ if [ ! -d "${CONF_DIR}" ]; then
         cp -r "${SOURCE_DIR}" "${CONF_DIR}"
 fi
 
+
 #Copy Xemu config at script launch
-cp -r "${SOURCE_DIR}/${XEMU_INI}" "${CONF_DIR}/${XEMU_INI}"
+if [ -f "${CONF_DIR}/xemu.override.toml" ] || [ -f "${CONF_DIR}/xemu.${name}.override.toml" ]; then
+    if [ -f "${CONF_DIR}/xemu.${name}.override.toml" ]; then
+        cp -rf "${CONF_DIR}/xemu.${name}.override.toml" "${CONF_DIR}/${XEMU_INI}"
+    else
+        cp -rf "${CONF_DIR}/xemu.override.toml" "${CONF_DIR}/${XEMU_INI}"
+     fi
+else
+    cp -r "${SOURCE_DIR}/${XEMU_INI}" "${CONF_DIR}/${XEMU_INI}"
+fi # finish bypass auto gamepad
 
 #Make xemu bios folder
 if [ ! -d "${BIOS_DIR}/bios" ]; then
