From 425462bfd2913ab9e97a7df114cdcea783bcf8fd Mon Sep 17 00:00:00 2001
From: John Williams <porschemad911@gmail.com>
Date: Sun, 11 Jan 2026 21:42:52 +1100
Subject: [PATCH] RK3326 soc_thermal

---
 src/cpu.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/cpu.cpp b/src/cpu.cpp
index 6307766..e476b34 100644
--- a/src/cpu.cpp
+++ b/src/cpu.cpp
@@ -606,6 +606,9 @@ bool CPUStats::GetCpuFile() {
         } else if (name == "apm_xgene") {
             find_input(path, "temp", input, "SoC Temperature");
             break;
+        } else if (name == "soc_thermal") {
+            find_fallback_input(path, "temp1", input);
+            break;
         } else {
             path.clear();
         }
-- 
2.47.3

