From 0e3eb106c43c640ddb3cc9863297153215913c15 Mon Sep 17 00:00:00 2001
From: John Williams <porschemad911@gmail.com>
Date: Sun, 11 Jan 2026 21:42:52 +1100
Subject: [PATCH 1/2] RK3326 soc_thermal

---
 src/cpu.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/cpu.cpp b/src/cpu.cpp
index f46faba..621e03e 100644
--- a/src/cpu.cpp
+++ b/src/cpu.cpp
@@ -615,6 +615,9 @@ bool CPUStats::GetCpuFile() {
         } else if (name == "apm_xgene") {
             find_input(path, "temp", input, "SoC Temperature");
             break;
+        } else if (name == "soc_thermal") {
+            find_fallback_input(path, "temp1", input);
+            break;
         } else {
             path.clear();
         }
-- 
2.47.3

