From 0941b7be6555286d60f3a551d100597d9b9640c9 Mon Sep 17 00:00:00 2001
From: John Williams <porschemad911@gmail.com>
Date: Mon, 19 Jan 2026 21:25:05 +1100
Subject: [PATCH 2/2] RK3326 battery_watt

---
 src/battery.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/battery.cpp b/src/battery.cpp
index 9cf8e5b..8a39d24 100644
--- a/src/battery.cpp
+++ b/src/battery.cpp
@@ -92,8 +92,8 @@ float BatteryStats::getPower() {
     float voltage = 0;
     for(int i = 0; i < batt_count; i++) {
         string syspath = battPath[i];
-        string current_power = syspath + "/current_now";
-        string current_voltage = syspath + "/voltage_now";
+        string current_power = syspath + "/current_avg";
+        string current_voltage = syspath + "/voltage_avg";
         string power_now = syspath + "/power_now";
         string status = syspath + "/status";
 
-- 
2.47.3

