From b33e56887542dbf313a22880e9b29d4f0b110843 Mon Sep 17 00:00:00 2001
From: John Williams <porschemad911@gmail.com>
Date: Fri, 15 May 2026 23:14:52 +1000
Subject: [PATCH 2/2] RK3326 power / timeRemaining

---
 src/battery.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/battery.cpp b/src/battery.cpp
index 947f446..7615c2b 100644
--- a/src/battery.cpp
+++ b/src/battery.cpp
@@ -92,8 +92,8 @@ float BatteryStats::getPower() {
 
     for (int i = 0; i < batt_count; i++) {
         string syspath = battPath[i];
-        string current_now = syspath + "/current_now";
-        string voltage_now = syspath + "/voltage_now";
+        string current_now = syspath + "/current_avg";
+        string voltage_now = syspath + "/voltage_avg";
         string power_now = syspath + "/power_now";
         string status = syspath + "/status";
 
@@ -153,10 +153,10 @@ float BatteryStats::getTimeRemaining() {
 
     for (int i = 0; i < batt_count; i++) {
         string syspath = battPath[i];
-        string current_now = syspath + "/current_now";
+        string current_now = syspath + "/current_avg";
         string charge_now = syspath + "/charge_now";
         string energy_now = syspath + "/energy_now";
-        string voltage_now = syspath + "/voltage_now";
+        string voltage_now = syspath + "/voltage_avg";
         string power_now = syspath + "/power_now";
 
         if (fs::exists(current_now)) {
-- 
2.47.3

