diff --color -rupN gtkmm-orig/Makefile.am gtkmm-3.24.9/Makefile.am
--- gtkmm-orig/Makefile.am	2024-06-21 17:28:12.276072911 -0400
+++ gtkmm-3.24.9/Makefile.am	2024-06-21 17:29:18.561288534 -0400
@@ -28,7 +28,7 @@ doc_subdirs = docs
 else
 doc_subdirs =
 endif
-SUBDIRS = tools $(src_subdirs) gdk/gdkmm gtk/gtkmm tests demos $(doc_subdirs)
+SUBDIRS = tools $(src_subdirs) gdk/gdkmm gtk/gtkmm tests $(doc_subdirs)
 
 gdkmm_includedir = $(includedir)/$(GDKMM_MODULE_NAME)
 gdkmm_include_HEADERS = gdk/gdkmm.h
diff --color -rupN gtkmm-orig/meson.build gtkmm-3.24.9/meson.build
--- gtkmm-orig/meson.build	2024-06-21 17:28:12.324073790 -0400
+++ gtkmm-3.24.9/meson.build	2024-06-21 17:29:11.793164410 -0400
@@ -395,7 +395,6 @@ subdir('gtk')
 subdir('MSVC_NMake/gtkmm')
 subdir('gtk/gtkmm')
 subdir('tests')
-subdir('demos/gtk-demo')
 subdir('docs/reference')
 
 if can_add_dist_script
