diff --git a/Makefile b/Makefile
index 01d99c3..0b80603 100644
--- a/Makefile
+++ b/Makefile
@@ -403,25 +403,27 @@ else ifeq ($(platform), arm64_cortex_a53_gles2)
 
 #######################################
 
-# ARM64 SM1 Odroid C4
-else ifeq ($(platform), odroidc4)
+# ROCKNIX device
+else ifeq ($(platform), @DEVICE@)
 	EXT ?= so
 	TARGET := $(TARGET_NAME)_libretro.$(EXT)
 	SHARED += -shared -Wl,--version-script=link.T
-	LDFLAGS +=  -Wl,--no-undefined
 	fpic = -fPIC
 	LIBS += -lrt
 	ARM_FLOAT_ABI_HARD = 0
-	FORCE_GLES = 1
 	SINGLE_PREC_FLAGS = 1
+ 	HAVE_OPENMP = 0
-	CPUFLAGS += -DHOST_CPU=0x20000006 -DTARGET_LINUX_ARMv8 -frename-registers
-	CFLAGS += -mcpu=cortex-a55 -mtune=cortex-a55 $(CPUFLAGS)
-	CXXFLAGS += -mcpu=cortex-a55 -mtune=cortex-a55 $(CPUFLAGS)
-	ASFLAGS += $(CFLAGS) -c -frename-registers -fno-strict-aliasing -ffast-math -ftree-vectorize
+	CFLAGS += -Ofast \
+		-fuse-linker-plugin \
+		-fno-stack-protector -fno-ident -fomit-frame-pointer \
+		-fmerge-all-constants -ffast-math -funroll-all-loops \
+		-mcpu=@TARGET_CPU@ -mtune=@TARGET_CPU@ -mno-outline-atomics
+	CXXFLAGS += $(CFLAGS)
+	LDFLAGS += -mcpu=@TARGET_CPU@ -mtune=@TARGET_CPU@ -Ofast -flto -fuse-linker-plugin
 	PLATFORM_EXT := unix
+ 	CORE_DEFINES += -DLOW_END -DLOW_RES
 	WITH_DYNAREC=arm64
 	HAVE_GENERIC_JIT = 0
-	HAVE_VULKAN = 0
 	HAVE_LTCG = 0
 
 #######################################
