diff --git a/keyboard.c b/keyboard.c
index d158403..df761fa 100644
--- a/keyboard.c
+++ b/keyboard.c
@@ -576,7 +576,7 @@ kbd_draw_key(struct kbd *kb, struct key *k, enum key_draw_type type)
 
     drw_draw_text(kb->surf, scheme->text, k->x, k->y, k->w, k->h,
                   KBD_KEY_BORDER, label, scheme->font_description);
-
+/*
     if (type == Press || type == Unpress) {
         kbd_clear_last_popup(kb);
 
@@ -592,7 +592,9 @@ kbd_draw_key(struct kbd *kb, struct key *k, enum key_draw_type type)
         drw_draw_text(kb->popup_surf, scheme->text, k->x, kb->last_popup_y,
                       k->w, k->h, KBD_KEY_BORDER, label,
                       scheme->font_description);
+
     }
+*/
 }
 
 void
