diff --git a/src/citron/main.cpp b/src/citron/main.cpp
index 991ff67768..a9c60f6cab 100644
--- a/src/citron/main.cpp
+++ b/src/citron/main.cpp
@@ -2054,6 +2054,8 @@ void GMainWindow::BootGame(const QString& filename, Service::AM::FrontendAppletP
     LOG_INFO(Frontend, "Booting game: {:016X} | {} | {}", title_id, title_name, title_version);
     const auto gpu_vendor = system->GPU().Renderer().GetDeviceVendor();
     UpdateWindowTitle(title_name, title_version, gpu_vendor);
+    const auto gpu_device_name = system->GPU().Renderer().GetDeviceName();
+    LOG_INFO(Frontend, "GPU device: {}", gpu_device_name);

     loading_screen->Prepare(system->GetAppLoader());
     loading_screen->show();
diff --git a/src/video_core/renderer_base.h b/src/video_core/renderer_base.h
index 0eeb689ad8..942f5c4c34 100644
--- a/src/video_core/renderer_base.h
+++ b/src/video_core/renderer_base.h
@@ -45,6 +45,8 @@ public:

     [[nodiscard]] virtual RasterizerInterface* ReadRasterizer() = 0;

+    [[nodiscard]] virtual std::string GetDeviceName() const = 0;
+
     [[nodiscard]] virtual std::string GetDeviceVendor() const = 0;

     // Getter/setter functions:
diff --git a/src/video_core/renderer_null/renderer_null.h b/src/video_core/renderer_null/renderer_null.h
index 34dbe1e4f0..1ca7ae624a 100644
--- a/src/video_core/renderer_null/renderer_null.h
+++ b/src/video_core/renderer_null/renderer_null.h
@@ -25,6 +25,10 @@ public:
         return &m_rasterizer;
     }

+    [[nodiscard]] std::string GetDeviceName() const override{
+        return "NULL";
+    }
+
     [[nodiscard]] std::string GetDeviceVendor() const override {
         return "NULL";
     }
diff --git a/src/video_core/renderer_opengl/renderer_opengl.h b/src/video_core/renderer_opengl/renderer_opengl.h
index 20ed3730f0..32d7bd56ea 100644
--- a/src/video_core/renderer_opengl/renderer_opengl.h
+++ b/src/video_core/renderer_opengl/renderer_opengl.h
@@ -47,6 +47,10 @@ public:
         return &rasterizer;
     }

+    [[nodiscard]] std::string GetDeviceName() const override{
+        return "GetDeviceName() not implemented for OpenGL";
+    }
+
     [[nodiscard]] std::string GetDeviceVendor() const override {
         return device.GetVendorName();
     }
diff --git a/src/video_core/renderer_vulkan/renderer_vulkan.h b/src/video_core/renderer_vulkan/renderer_vulkan.h
index 77a57f4196..126269157f 100644
--- a/src/video_core/renderer_vulkan/renderer_vulkan.h
+++ b/src/video_core/renderer_vulkan/renderer_vulkan.h
@@ -53,6 +53,10 @@ public:
         return &rasterizer;
     }

+    [[nodiscard]] std::string GetDeviceName() const override{
+        return std::string(device.GetModelName());
+    }
+
     [[nodiscard]] std::string GetDeviceVendor() const override {
         return device.GetDriverName();
     }
