diff --git a/build/gmake/Makefile b/build/gmake/Makefile
index 5938f0d..89f1dfd 100755
--- a/build/gmake/Makefile
+++ b/build/gmake/Makefile
@@ -28,7 +28,7 @@ ifeq ($(config),debug)
   CPPFLAGS  += -MMD -MP $(DEFINES) $(INCLUDES)
   CFLAGS    += $(CPPFLAGS) $(ARCH) -g -Wall -D__GBM__
   CXXFLAGS  += $(CFLAGS) -std=c++2a
-  LDFLAGS   += -Wl,-rpath=. -L../../. -ldl -lEGL -lGLESv2 -lrga -pthread -ffast-math -levdev -lgbm -ldrm -lpng -lasound -lopenal
+  LDFLAGS   += -L../../. -ldl -lEGL -lGLESv2 -lrga -pthread -ffast-math -levdev -lgbm -ldrm -lpng -lasound -lopenal
   LIBS      +=
   RESFLAGS  += $(DEFINES) $(INCLUDES)
   LDDEPS    +=
@@ -50,7 +50,7 @@ ifeq ($(config),release)
   CPPFLAGS  += -MMD -MP $(DEFINES) $(INCLUDES)
   CFLAGS    += $(CPPFLAGS) $(ARCH) -Ofast -Wall -D__GBM__
   CXXFLAGS  += $(CFLAGS) -std=c++2a
-  LDFLAGS   += -s -Wl,-rpath=. -L../../. -ldl -lEGL -lGLESv2 -lrga -pthread -ffast-math -levdev -lgbm -ldrm -lpng -lasound -lopenal
+  LDFLAGS   += -s -L../../. -ldl -lEGL -lGLESv2 -lrga -pthread -ffast-math -levdev -lgbm -ldrm -lpng -lasound -lopenal
   LIBS      +=
   RESFLAGS  += $(DEFINES) $(INCLUDES)
   LDDEPS    +=
diff --git a/premake4.lua b/premake4.lua
index 1cd054f..67c0610 100755
--- a/premake4.lua
+++ b/premake4.lua
@@ -12,7 +12,7 @@ project "retrorun"
    includedirs { "src/.", "." }
    files { "src/*.h", "src/*.cpp" }
    buildoptions { "-std=c++11 -Wall -D__GBM__" }
-   linkoptions { "-Wl,-rpath=. -L../../. -lgo2 -ldl -lEGL -lGLESv2 -lrga" }
+   linkoptions { "-L../../. -lgo2 -ldl -lEGL -lGLESv2 -lrga" }

    configuration "Debug"
       flags { "Symbols" }
diff --git a/src/go2/input.cpp b/src/go2/input.cpp
index 2847adb..50c842d 100755
--- a/src/go2/input.cpp
+++ b/src/go2/input.cpp
@@ -36,7 +36,7 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 #include <stdbool.h>
 #include <pthread.h>

-#include <libevdev-1.0/libevdev/libevdev.h>
+#include <libevdev/libevdev.h>
 #include <linux/limits.h>

 #define BATTERY_BUFFER_SIZE (128)
@@ -669,4 +669,4 @@ go2_thumb_t go2_input_state_thumbstick_get(go2_input_state_t *state, go2_input_t
 void go2_input_state_thumbstick_setNul(go2_input_state_t *state, go2_input_thumbstick_t thumbstick)
 {
     state->thumbs[thumbstick]={0};
-}
\ No newline at end of file
+}
