diff --git a/SDL/SDLJoystick.cpp b/SDL/SDLJoystick.cpp
index 94a6f481ba..977617b77d 100644
--- a/SDL/SDLJoystick.cpp
+++ b/SDL/SDLJoystick.cpp
@@ -141,7 +141,7 @@ InputKeyCode SDLJoystick::getKeycodeForButton(SDL_GameControllerButton button) {
 	case SDL_CONTROLLER_BUTTON_BACK:
 		return NKCODE_BUTTON_9; // select button
 	case SDL_CONTROLLER_BUTTON_GUIDE:
-		return NKCODE_BACK; // pause menu
+		return NKCODE_BUTTON_11;
 	case SDL_CONTROLLER_BUTTON_LEFTSTICK:
 		return NKCODE_BUTTON_THUMBL;
 	case SDL_CONTROLLER_BUTTON_RIGHTSTICK:
@@ -150,7 +150,7 @@ InputKeyCode SDLJoystick::getKeycodeForButton(SDL_GameControllerButton button) {
 	// Found these limits by checking out the SDL2 branch of the SDL repo, doing git blame, then `git tag --contains (commit)` etc.
 #if SDL_VERSION_ATLEAST(2, 0, 16)
 	case SDL_CONTROLLER_BUTTON_MISC1:
-		return NKCODE_BUTTON_11;
+                return NKCODE_BACK; // pause menu
 #endif
 #if SDL_VERSION_ATLEAST(2, 0, 28)
 	case SDL_CONTROLLER_BUTTON_PADDLE1:
