diff --git a/render.c b/render.c
index 8fed6d8..b157323 100644
--- a/render.c
+++ b/render.c
@@ -438,7 +438,7 @@ void render(struct mako_surface *surface, struct pool_buffer *buffer, int scale,
 			++visible_count;
 		}
 	}
-
+/*
 	if (hidden_count > 0) {
 		struct mako_notification *hidden_notif = create_notification(state);
 		hidden_notif->surface = surface;
@@ -478,7 +478,7 @@ void render(struct mako_surface *surface, struct pool_buffer *buffer, int scale,
 		}
 		destroy_notification(hidden_notif);
 	}
-
+*/
 	*rendered_width = max_width;
 	*rendered_height = total_height;
 }
