--- a/render.c	2026-08-09 23:04:49
+++ b/render.c	2026-08-09 23:04:49
@@ -355,17 +355,14 @@
 	cairo_restore(cairo);
 
 	size_t visible_count = 0;
-	size_t hidden_count = 0;
 	int total_height = 0;
 	int max_width = 0;
 	int pending_bottom_margin = 0;
 	struct mako_notification *notif;
-	size_t total_notifications = 0;
 	wl_list_for_each(notif, &state->notifications, link) {
 		if (notif->surface != surface) {
 			continue;
 		}
-		++total_notifications;
 
 		// Immediately before rendering we need to re-match all of the criteria
 		// so that matches against the anchor and output work even if the
@@ -390,7 +387,6 @@
 
 		if (style->max_visible >= 0 &&
 				visible_count >= (size_t)style->max_visible) {
-			++hidden_count;
 			continue;
 		}
 
@@ -435,45 +431,7 @@
 			// group are rendered based on criteria, a group is considered a
 			// single entity for this purpose.
 			++visible_count;
-		}
-	}
-
-	if (hidden_count > 0) {
-		struct mako_notification *hidden_notif = create_notification(state);
-		hidden_notif->surface = surface;
-		hidden_notif->hidden = true;
-		apply_each_criteria(&state->config.criteria, hidden_notif);
-
-		struct mako_style *style = &hidden_notif->style;
-
-		if (style->margin.top > pending_bottom_margin) {
-			total_height += style->margin.top;
-		} else {
-			total_height += pending_bottom_margin;
-		}
-
-		struct mako_hidden_format_data data = {
-			.hidden = hidden_count,
-			.count = total_notifications,
-		};
-
-		size_t text_ln =
-			format_text(style->format, NULL, format_hidden_text, &data);
-		char *text = malloc(text_ln + 1);
-		if (text == NULL) {
-			fprintf(stderr, "allocation failed");
-			return;
 		}
-
-		format_text(style->format, text, format_hidden_text, &data);
-
-		int hidden_height = render_notification(
-			cairo, state, surface, style, text, NULL, total_height, scale, NULL, 0);
-		free(text);
-
-		total_height += hidden_height;
-		pending_bottom_margin = style->margin.bottom;
-		destroy_notification(hidden_notif);
 	}
 
 	*rendered_width = max_width;
