--- a/Src/OSD/SDL/Main.cpp	2024-04-04 17:30:55.333806373 -0400
+++ b/Src/OSD/SDL/Main.cpp	2024-04-04 17:30:24.325773043 -0400
@@ -237,11 +237,11 @@
   // Set the context as the current window context
   SDL_GL_MakeCurrent(s_window, context);
 
   // Initialize GLEW, allowing us to use features beyond OpenGL 1.2
   err = glewInit();
-  if (GLEW_OK != err)
+  if (GLEW_OK != err && GLEW_ERROR_NO_GLX_DISPLAY != err)
   {
     ErrorLog("OpenGL initialization failed: %s\n", glewGetErrorString(err));
     return FAIL;
   }
 
