diff --git a/Makefiles/Rules.inc b/Makefiles/Rules.inc
index 01b8958..5e5c394 100644
--- a/Makefiles/Rules.inc
+++ b/Makefiles/Rules.inc
@@ -79,7 +79,7 @@ CXXSTD = -std=c++17
 # Construct the compiler (C and C++) and linker flags
 #
 COMMON_CFLAGS = -c $(ARCH) $(OPT) $(WARN) $(addprefix -I,$(sort $(INCLUDE_DIRS))) -DGLEW_STATIC $(SUPERMODEL_BUILD_FLAGS)
-CFLAGS = $(COMMON_CFLAGS) $(CSTD)
+CFLAGS = $(COMMON_CFLAGS) $(CSTD) -Wno-implicit-function-declaration
 CXXFLAGS = $(PLATFORM_CXXFLAGS) $(COMMON_CFLAGS) $(CXXSTD)
 LDFLAGS = -o $(BIN_DIR)/$(OUTFILE) $(PLATFORM_LDFLAGS) -s
 
