From: sunshineinabox <aqemail@gmail.com>
Subject: [PATCH] remoteproc: qcom: pas: add SM4450 ADSP and WPSS


--- a/drivers/remoteproc/qcom_q6v5_pas.c
+++ b/drivers/remoteproc/qcom_q6v5_pas.c
@@ -1616,6 +1616,8 @@
 	{ .compatible = "qcom,shikra-cdsp-pas", .data = &shikra_cdsp_resource },
 	{ .compatible = "qcom,shikra-lpaicp-pas", .data = &shikra_lpaicp_resource },
 	{ .compatible = "qcom,shikra-mpss-pas", .data = &shikra_mpss_resource },
+	{ .compatible = "qcom,sm4450-adsp-pas", .data = &sm6350_adsp_resource },
+	{ .compatible = "qcom,sm4450-wpss-pas", .data = &sc7280_wpss_resource },
 	{ .compatible = "qcom,sm6115-adsp-pas", .data = &adsp_resource_init },
 	{ .compatible = "qcom,sm6115-cdsp-pas", .data = &cdsp_resource_init },
 	{ .compatible = "qcom,sm6115-mpss-pas", .data = &sc8180x_mpss_resource },
