diff --git a/src/mainwindow.cpp b/src/mainwindow.cpp
index d5163c8..c327bc9 100644
--- a/src/mainwindow.cpp
+++ b/src/mainwindow.cpp
@@ -589,7 +589,7 @@ void MainWindow::toggleBorderless()
 
 void MainWindow::toggleMenu()
 {
-    m_menuBar->setVisible(!m_menuBar->isVisible());
+    m_menuBar->setVisible(false);
     Properties::Instance()->menuVisible = m_menuBar->isVisible();
 }
 
@@ -748,7 +748,7 @@ void MainWindow::propertiesChanged()
         }
     }
 
-    m_menuBar->setVisible(Properties::Instance()->menuVisible);
+    m_menuBar->setVisible(false);
 
     m_bookmarksDock->setVisible(Properties::Instance()->useBookmarks
                                 && Properties::Instance()->bookmarksVisible);
