diff --git a/src/properties.cpp b/src/properties.cpp
index 78cba0b..8201dfc 100644
--- a/src/properties.cpp
+++ b/src/properties.cpp
@@ -69,7 +69,7 @@ QFont Properties::defaultFont()
 {
     QFont default_font = QApplication::font();
     default_font.setFamily(QLatin1String(DEFAULT_FONT));
-    default_font.setPointSize(12);
+    default_font.setPointSize(20);
     default_font.setStyleHint(QFont::TypeWriter);
     return default_font;
 }
@@ -80,7 +80,7 @@ void Properties::loadSettings()
     if (!guiStyle.isNull())
         QApplication::setStyle(guiStyle);
 
-    colorScheme = m_settings->value(QLatin1String("colorScheme"), QLatin1String("Linux")).toString();
+    colorScheme = m_settings->value(QLatin1String("colorScheme"), QLatin1String("WhiteOnBlack")).toString();
 
     highlightCurrentTerminal = m_settings->value(QLatin1String("highlightCurrentTerminal"), true).toBool();
     focusOnMoueOver = m_settings->value(QLatin1String("focusOnMoueOver"), false).toBool();
@@ -126,7 +126,7 @@ void Properties::loadSettings()
     /* default to BlockCursor */
     keyboardCursorShape = m_settings->value(QLatin1String("KeyboardCursorShape"), 0).toInt();
     keyboardCursorBlink = m_settings->value(QLatin1String("KeyboardCursorBlink"), false).toBool();
-    hideTabBarWithOneTab = m_settings->value(QLatin1String("HideTabBarWithOneTab"), false).toBool();
+    hideTabBarWithOneTab = m_settings->value(QLatin1String("HideTabBarWithOneTab"), true).toBool();
     // For "Motion after paste", 2 (scrolling to bottom) makes more sense
     m_motionAfterPaste = m_settings->value(QLatin1String("MotionAfterPaste"), 2).toInt();
     m_disableBracketedPasteMode = m_settings->value(QLatin1String("DisableBracketedPasteMode"), false).toBool();
