diff --git a/hw/xbox/nv2a/pgraph/vk/instance.c b/hw/xbox/nv2a/pgraph/vk/instance.c
index 9df440930c..8f5cc0f99b 100644
--- a/hw/xbox/nv2a/pgraph/vk/instance.c
+++ b/hw/xbox/nv2a/pgraph/vk/instance.c
@@ -222,8 +222,7 @@ static bool create_instance(PGRAPHState *pg, Error **errp)
     VkApplicationInfo app_info = {
         .sType = VK_STRUCTURE_TYPE_APPLICATION_INFO,
         .pApplicationName = "xemu",
-        .applicationVersion = VK_MAKE_VERSION(
-            xemu_version_major, xemu_version_minor, xemu_version_patch),
+        .applicationVersion = VK_MAKE_VERSION(1, 0, 0),
         .pEngineName = "No Engine",
         .engineVersion = VK_MAKE_VERSION(1, 0, 0),
         .apiVersion = VK_API_VERSION_1_3,
diff --git a/xemu-version.c b/xemu-version.c
index f2e7a958e9..d4148603e2 100644
--- a/xemu-version.c
+++ b/xemu-version.c
@@ -1,8 +1,8 @@
 #include "xemu-version-macro.h"
 
-const int xemu_version_major = XEMU_VERSION_MAJOR;
-const int xemu_version_minor = XEMU_VERSION_MINOR;
-const int xemu_version_patch = XEMU_VERSION_PATCH;
+//const int xemu_version_major = XEMU_VERSION_MAJOR;
+//const int xemu_version_minor = XEMU_VERSION_MINOR;
+//const int xemu_version_patch = XEMU_VERSION_PATCH;
 const char *xemu_version = XEMU_VERSION;
 const char *xemu_branch  = XEMU_BRANCH;;
 const char *xemu_commit  = XEMU_COMMIT;
