diff --git a/vita3k/app/src/app_init.cpp b/vita3k/app/src/app_init.cpp
index 5df6321563..8c1b6bbc80 100644
--- a/vita3k/app/src/app_init.cpp
+++ b/vita3k/app/src/app_init.cpp
@@ -158,8 +158,7 @@ void init_paths(Root &root_paths) {
     root_paths.set_cache_path(storage_path / "cache" / "");
     root_paths.set_patch_path(storage_path / "patch" / "");
 #else
-    auto sdl_base_path = SDL_GetBasePath();
-    auto base_path = fs_utils::utf8_to_path(sdl_base_path);
+    auto base_path = fs_utils::utf8_to_path("/storage/.config/vita3k/");
 
     root_paths.set_base_path(base_path);
     root_paths.set_static_assets_path(base_path);
@@ -184,9 +183,7 @@ void init_paths(Root &root_paths) {
     } else {
         // SDL_GetPrefPath is deferred as it creates the directory.
         // When using a portable directory, it is not needed.
-        auto sdl_pref_path = SDL_GetPrefPath(org_name, app_name);
-        auto pref_path = fs_utils::utf8_to_path(sdl_pref_path);
-        SDL_free(sdl_pref_path);
+        auto pref_path = fs_utils::utf8_to_path("/storage/psvita/vita3k/");
 
 #if defined(__APPLE__)
         // Store other data in the user-wide path. Otherwise we may end up dumping
@@ -226,10 +223,10 @@ void init_paths(Root &root_paths) {
         if (XDG_DATA_HOME != NULL)
             root_paths.set_pref_path(fs::path(XDG_DATA_HOME) / app_name / app_name / "");
 
-        if (XDG_CONFIG_HOME != NULL)
-            root_paths.set_config_path(fs::path(XDG_CONFIG_HOME) / app_name / "");
-        else if (env_home != NULL)
-            root_paths.set_config_path(fs::path(env_home) / ".config" / app_name / "");
+        //if (XDG_CONFIG_HOME != NULL)
+        //    root_paths.set_config_path(fs::path(XDG_CONFIG_HOME) / app_name / "");
+        //else if (env_home != NULL)
+        //    root_paths.set_config_path(fs::path(env_home) / ".config" / app_name / "");
 
         if (XDG_CACHE_HOME != NULL) {
             root_paths.set_cache_path(fs::path(XDG_CACHE_HOME) / app_name / "");
