--- a/vita3k/app/src/app_init.cpp
+++ b/vita3k/app/src/app_init.cpp
@@ -158,8 +158,7 @@
     root_paths.set_cache_path(storage_path / "cache" / "");
     root_paths.set_patch_path(storage_path / "patch" / "");
 #else
-    auto sdl_base_path = SDL_GetBasePath();
-    auto base_path = fs_utils::utf8_to_path(sdl_base_path);
+    auto base_path = fs_utils::utf8_to_path("/storage/.config/Vita3K");
 
     root_paths.set_base_path(base_path);
     root_paths.set_static_assets_path(base_path);
@@ -184,9 +183,7 @@
     } else {
         // SDL_GetPrefPath is deferred as it creates the directory.
         // When using a portable directory, it is not needed.
-        auto sdl_pref_path = SDL_GetPrefPath(org_name, app_name);
-        auto pref_path = fs_utils::utf8_to_path(sdl_pref_path);
-        SDL_free(sdl_pref_path);
+        auto pref_path = fs_utils::utf8_to_path("/storage/roms/psvita/vita3k");
 
 #if defined(__APPLE__)
         // Store other data in the user-wide path. Otherwise we may end up dumping
