--- a/vita3k/gui/src/imgui_impl_sdl.cpp
+++ b/vita3k/gui/src/imgui_impl_sdl.cpp
@@ -392,9 +392,16 @@ static void ImGui_ImplSDL3_UpdateGamepads(ImGui_State *state) {

     // Get gamepad
     io.BackendFlags &= ~ImGuiBackendFlags_HasGamepad;
-    SDL_Gamepad *game_controller = SDL_OpenGamepad(0);
-    if (!game_controller)
+    int num_gamepads = 0;
+    SDL_JoystickID *gamepads = SDL_GetGamepads(&num_gamepads);
+    if (!gamepads || num_gamepads == 0) {
+        SDL_free(gamepads);
         return;
+    }
+    SDL_Gamepad *game_controller = SDL_OpenGamepad(gamepads[0]);
+    SDL_free(gamepads);
+    if (!game_controller)
+        return;
     io.BackendFlags |= ImGuiBackendFlags_HasGamepad;
