From a621a74a5099612fbd4a0689aca4dc6ee7b4080f Mon Sep 17 00:00:00 2001
From: spycat88 <spycat88@users.noreply.github.com>
Date: Sun, 16 Nov 2025 15:40:43 +0000
Subject: [PATCH] fix building with gcc15

---
 jstest-sdl.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/jstest-sdl.c b/jstest-sdl.c
index d576029..b76ed08 100644
--- a/jstest-sdl.c
+++ b/jstest-sdl.c
@@ -44,6 +44,7 @@
 #include <curses.h>
 #include <errno.h>
 #include <limits.h>
+#include <stdbool.h>
 #include <stdio.h>
 #include <stdlib.h>
 
-- 
2.34.1

