From 49962ce8988b1c4c2b7f158afe61a073c553c622 Mon Sep 17 00:00:00 2001
From: John Williams <porschemad911@gmail.com>
Date: Sat, 28 Mar 2026 21:41:02 +1100
Subject: [PATCH 2/2] Enable 'use_gles' for OpenGL ES rendering

---
 src/common/settings.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/settings.h b/src/common/settings.h
index 90922bcff..c121901c9 100644
--- a/src/common/settings.h
+++ b/src/common/settings.h
@@ -511,7 +511,7 @@ struct Values {
 #endif
         GraphicsAPI::Software, GraphicsAPI::Vulkan, Keys::graphics_api};
     SwitchableSetting<u32> physical_device{0, Keys::physical_device};
-    Setting<bool> use_gles{false, Keys::use_gles};
+    Setting<bool> use_gles{true, Keys::use_gles};
     Setting<bool> renderer_debug{false, Keys::renderer_debug};
     Setting<bool> dump_command_buffers{false, Keys::dump_command_buffers};
     SwitchableSetting<bool> spirv_shader_gen{true, Keys::spirv_shader_gen};
-- 
2.47.3

