From 35452a058ed9ea08434a159558c0749b03bac07d Mon Sep 17 00:00:00 2001
From: John Williams <porschemad911@gmail.com>
Date: Fri, 10 Oct 2025 16:17:50 +1100
Subject: [PATCH 4/4] Enable 'use_gles' for OpenGL ES rendering

---
 src/common/settings.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/settings.h b/src/common/settings.h
index c0f595d12..f80dab47c 100644
--- a/src/common/settings.h
+++ b/src/common/settings.h
@@ -505,7 +505,7 @@ struct Values {
 #endif
         GraphicsAPI::Software, GraphicsAPI::Vulkan, "graphics_api"};
     SwitchableSetting<u32> physical_device{0, "physical_device"};
-    Setting<bool> use_gles{false, "use_gles"};
+    Setting<bool> use_gles{true, "use_gles"};
     Setting<bool> renderer_debug{false, "renderer_debug"};
     Setting<bool> dump_command_buffers{false, "dump_command_buffers"};
     SwitchableSetting<bool> spirv_shader_gen{true, "spirv_shader_gen"};
-- 
2.47.3

