From c2b17d28cf8800f3011c3c13492fa836d8c773f3 Mon Sep 17 00:00:00 2001
From: Philippe Simons <simons.philippe@gmail.com>
Date: Sat, 28 Dec 2024 10:55:19 +0100
Subject: [PATCH] update defaults config

---
 src/config.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/config.c b/src/config.c
index 50848a9..95a36ba 100644
--- a/src/config.c
+++ b/src/config.c
@@ -23,13 +23,13 @@ config_params_s init_config() {
 
   c.filename = "config.ini"; // default config file to load
 
-  c.init_fullscreen = 0; // default fullscreen state at load
+  c.init_fullscreen = 1; // default fullscreen state at load
   c.init_use_gpu = 1;    // default to use hardware acceleration
   c.idle_ms = 10;        // default to high performance
   c.wait_for_device = 1; // default to exit if device disconnected
   c.wait_packets = 1024; // default zero-byte attempts to disconnect (about 2
   // sec for default idle_ms)
-  c.audio_enabled = 0;        // route M8 audio to default output
+  c.audio_enabled = 1;        // route M8 audio to default output
   c.audio_buffer_size = 1024; // requested audio buffer size in samples
   c.audio_device_name = NULL; // Use this device, leave NULL to use the default output device
 
-- 
2.47.1

