diff -rupbN grub.orig/grub-core/normal/menu_text.c grub/grub-core/normal/menu_text.c
--- grub.orig/grub-core/normal/menu_text.c	2024-10-28 03:36:55.522217953 +0000
+++ grub/grub-core/normal/menu_text.c	2024-10-28 04:26:02.424879849 +0000
@@ -167,10 +167,10 @@ command-line or ESC to discard edits and
     {
       char *msg_translated;
 
-      msg_translated = grub_xasprintf (_("Use the %C and %C keys to select which "
-					 "entry is highlighted."),
-				       GRUB_UNICODE_UPARROW,
-				       GRUB_UNICODE_DOWNARROW);
+      msg_translated = grub_xasprintf (_("Use the Volume Up and Volume Down buttons to select which "
+					 "device is highlighted."));
+				       //GRUB_UNICODE_UPARROW,
+				       //GRUB_UNICODE_DOWNARROW);
       if (!msg_translated)
 	return 0;
       ret += grub_print_message_indented_real (msg_translated, STANDARD_MARGIN,
@@ -181,17 +181,17 @@ command-line or ESC to discard edits and
       if (nested)
 	{
 	  ret += grub_print_message_indented_real
-	    (_("Press enter to boot the selected OS, "
-	       "`e' to edit the commands before booting "
-	       "or `c' for a command-line. ESC to return previous menu."),
+	    (_("Press the Power button to boot the selected Device, "
+	      "the selected device will be set as the default "
+	      "after the first full boot."),
 	     STANDARD_MARGIN, STANDARD_MARGIN, term, dry_run);
 	}
       else
 	{
 	  ret += grub_print_message_indented_real
-	    (_("Press enter to boot the selected OS, "
-	       "`e' to edit the commands before booting "
-	       "or `c' for a command-line."),
+            (_("Press the Power button to boot the selected Device, "
+              "the selected device will be set as the default "
+              "after the first full boot."),
 	     STANDARD_MARGIN, STANDARD_MARGIN, term, dry_run);
 	}
     }
