--- a/Source/ui_libretro/CMakeLists.txt
+++ b/Source/ui_libretro/CMakeLists.txt
@@ -50,7 +50,6 @@
 	list(APPEND UI_LIBRETRO_PROJECT_LIBS "-ObjC -lsqlite3 -framework OpenGLES")
 elseif(TARGET_PLATFORM_UNIX)
 	list(APPEND UI_LIBRETRO_PROJECT_LIBS "-static-libgcc")
-	list(APPEND UI_LIBRETRO_PROJECT_LIBS "-static-libstdc++")
 endif()
 
 if(CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 8 AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9)
