diff --color -rupN sway.orig/sway/desktop/xdg_shell.c sway/sway/desktop/xdg_shell.c
--- sway.orig/sway/desktop/xdg_shell.c	2024-06-30 17:08:57.636569873 -0400
+++ sway/sway/desktop/xdg_shell.c	2024-06-30 17:10:25.749276012 -0400
@@ -188,7 +188,7 @@ static void set_fullscreen(struct sway_v
 	if (xdg_shell_view_from_view(view) == NULL) {
 		return;
 	}
-	wlr_xdg_toplevel_set_fullscreen(view->wlr_xdg_toplevel, fullscreen);
+	wlr_xdg_toplevel_set_maximized(view->wlr_xdg_toplevel, fullscreen);
 }
 
 static void set_resizing(struct sway_view *view, bool resizing) {
diff --color -rupN sway.orig/sway/desktop/xwayland.c sway/sway/desktop/xwayland.c
--- sway.orig/sway/desktop/xwayland.c	2024-06-30 17:08:57.636569873 -0400
+++ sway/sway/desktop/xwayland.c	2024-06-30 17:10:08.741946696 -0400
@@ -318,7 +318,7 @@ static void set_fullscreen(struct sway_v
 		return;
 	}
 	struct wlr_xwayland_surface *surface = view->wlr_xwayland_surface;
-	wlr_xwayland_surface_set_fullscreen(surface, fullscreen);
+	wlr_xwayland_surface_set_maximized(surface, fullscreen);
 }
 
 static bool wants_floating(struct sway_view *view) {
