From: sunshineinabox <aqemail@gmail.com>
Subject: [PATCH] soc: qcom: add SM6450/SM7435/SG6150 SoC IDs and TrimUI TG4070 board

---
diff -Nur a/Documentation/devicetree/bindings/arm/qcom.yaml b/Documentation/devicetree/bindings/arm/qcom.yaml
--- a/Documentation/devicetree/bindings/arm/qcom.yaml	2026-09-26 19:28:31
+++ b/Documentation/devicetree/bindings/arm/qcom.yaml	2026-09-26 19:28:31
@@ -1007,6 +1007,11 @@
 
       - items:
           - enum:
+              - trimui,tg4070
+          - const: qcom,sm6450
+
+      - items:
+          - enum:
               - fxtec,pro1x
           - const: qcom,sm6115
 
diff -Nur a/drivers/soc/qcom/socinfo.c b/drivers/soc/qcom/socinfo.c
--- a/drivers/soc/qcom/socinfo.c	2026-09-26 19:28:31
+++ b/drivers/soc/qcom/socinfo.c	2026-09-26 19:28:31
@@ -511,6 +511,12 @@
 	{ qcom_board_id(X1E80100) },
 	{ qcom_board_id(SM8650) },
 	{ qcom_board_id(SM4450) },
+	{ qcom_board_id(SM6450) },
+	{ qcom_board_id(SM6450P) },
+	{ qcom_board_id(SM7435) },
+	{ qcom_board_id(SG6150) },
+	{ qcom_board_id(SG6150P) },
+	{ qcom_board_id(SM7435P) },
 	{ qcom_board_id(SAR1130P) },
 	{ qcom_board_id(QDU1010) },
 	{ qcom_board_id(QRU1032) },
diff -Nur a/include/dt-bindings/arm/qcom,ids.h b/include/dt-bindings/arm/qcom,ids.h
--- a/include/dt-bindings/arm/qcom,ids.h	2026-09-26 19:28:31
+++ b/include/dt-bindings/arm/qcom,ids.h	2026-09-26 19:28:31
@@ -270,6 +270,12 @@
 #define QCOM_ID_X1E80100		555
 #define QCOM_ID_SM8650			557
 #define QCOM_ID_SM4450			568
+#define QCOM_ID_SM6450			537
+#define QCOM_ID_SM6450P			583
+#define QCOM_ID_SM7435			613
+#define QCOM_ID_SG6150			633
+#define QCOM_ID_SG6150P			634
+#define QCOM_ID_SM7435P			638
 #define QCOM_ID_SAR1130P		579
 #define QCOM_ID_QDU1010			587
 #define QCOM_ID_QRU1032			588
