From: sunshineinabox <aqemail@gmail.com>
Subject: [PATCH] pmdomain: qcom: rpmhpd: add SM6450 power domains

---
diff -Nur a/drivers/pmdomain/qcom/rpmhpd.c b/drivers/pmdomain/qcom/rpmhpd.c
--- a/drivers/pmdomain/qcom/rpmhpd.c	2026-09-26 19:28:31
+++ b/drivers/pmdomain/qcom/rpmhpd.c	2026-09-26 19:28:31
@@ -462,6 +462,24 @@
 	.num_pds = ARRAY_SIZE(sm4450_rpmhpds),
 };
 
+/* SM6450 RPMH powerdomains */
+static struct rpmhpd *sm6450_rpmhpds[] = {
+	[RPMHPD_CX] = &cx,
+	[RPMHPD_CX_AO] = &cx_ao,
+	[RPMHPD_EBI] = &ebi,
+	[RPMHPD_GFX] = &gfx,
+	[RPMHPD_LCX] = &lcx,
+	[RPMHPD_LMX] = &lmx,
+	[RPMHPD_MSS] = &mss,
+	[RPMHPD_MX] = &mx,
+	[RPMHPD_MX_AO] = &mx_ao,
+};
+
+static const struct rpmhpd_desc sm6450_desc = {
+	.rpmhpds = sm6450_rpmhpds,
+	.num_pds = ARRAY_SIZE(sm6450_rpmhpds),
+};
+
 /* SM6350 RPMH powerdomains */
 static struct rpmhpd *sm6350_rpmhpds[] = {
 	[SM6350_CX] = &cx_w_mx_parent,
@@ -912,6 +930,7 @@
 	{ .compatible = "qcom,sdx65-rpmhpd", .data = &sdx65_desc},
 	{ .compatible = "qcom,sdx75-rpmhpd", .data = &sdx75_desc},
 	{ .compatible = "qcom,sm4450-rpmhpd", .data = &sm4450_desc },
+	{ .compatible = "qcom,sm6450-rpmhpd", .data = &sm6450_desc },
 	{ .compatible = "qcom,sm6350-rpmhpd", .data = &sm6350_desc },
 	{ .compatible = "qcom,sm7150-rpmhpd", .data = &sm7150_desc },
 	{ .compatible = "qcom,sm8150-rpmhpd", .data = &sm8150_desc },
