From: sunshineinabox <aqemail@gmail.com>
Subject: [PATCH] soc: qcom: pd-mapper: add SM6450 domains

---
diff -Nur a/drivers/soc/qcom/qcom_pd_mapper.c b/drivers/soc/qcom/qcom_pd_mapper.c
--- a/drivers/soc/qcom/qcom_pd_mapper.c	2026-09-26 19:28:31
+++ b/drivers/soc/qcom/qcom_pd_mapper.c	2026-09-26 19:28:31
@@ -491,6 +491,15 @@
 	NULL,
 };
 
+static const struct qcom_pdm_domain_data *sm6450_domains[] = {
+	&adsp_audio_pd,
+	&adsp_root_pd,
+	&adsp_charger_pd,
+	&adsp_sensor_pd,
+	&cdsp_root_pd,
+	NULL,
+};
+
 static const struct qcom_pdm_domain_data *sdm845_domains[] = {
 	&adsp_audio_pd,
 	&adsp_root_pd,
@@ -615,6 +624,7 @@
 	{ .compatible = "qcom,sm4250", .data = sm6115_domains, },
 	{ .compatible = "qcom,sm6115", .data = sm6115_domains, },
 	{ .compatible = "qcom,sm6350", .data = sm6350_domains, },
+	{ .compatible = "qcom,sm6450", .data = sm6450_domains, },
 	{ .compatible = "qcom,sm7150", .data = sm7150_domains, },
 	{ .compatible = "qcom,sm7225", .data = sm6350_domains, },
 	{ .compatible = "qcom,sm7325", .data = sc7280_domains, },
