From: sunshineinabox <aqemail@gmail.com>
Subject: [PATCH] iommu/arm-smmu-qcom: add SM6450 support

---
diff -Nur a/drivers/iommu/arm/arm-smmu/arm-smmu-qcom.c b/drivers/iommu/arm/arm-smmu/arm-smmu-qcom.c
--- a/drivers/iommu/arm/arm-smmu/arm-smmu-qcom.c	2026-09-26 19:28:31
+++ b/drivers/iommu/arm/arm-smmu/arm-smmu-qcom.c	2026-09-26 19:28:31
@@ -57,6 +57,8 @@
 			.data = (const void *) (PREFETCH_SHALLOW | CPRE | CMTLB) },
 	{ .compatible = "qcom,sm6115-mdss",
 			.data = (const void *) (PREFETCH_SHALLOW | CPRE | CMTLB) },
+	{ .compatible = "qcom,sm6450-mdss",
+			.data = (const void *) (PREFETCH_DEFAULT | CMTLB) },
 	{ .compatible = "qcom,sm6125-mdss",
 			.data = (const void *) (PREFETCH_SHALLOW | CPRE | CMTLB) },
 	{ .compatible = "qcom,sm6350-mdss",
@@ -423,6 +425,7 @@
 	{ .compatible = "qcom,sm6115-mdss" },
 	{ .compatible = "qcom,sm6350-mdss" },
 	{ .compatible = "qcom,sm6375-mdss" },
+	{ .compatible = "qcom,sm6450-mdss" },
 	{ .compatible = "qcom,sm8150-mdss" },
 	{ .compatible = "qcom,sm8250-mdss" },
 	{ .compatible = "qcom,x1e80100-mdss" },
