--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -396,6 +396,7 @@
 	add_compile_definitions(PNG_ARM_NEON_OPT=0)
 
 	add_compile_options(-Ofast -fno-tree-slp-vectorize)
+	add_compile_options(-mno-outline-atomics)
 	add_compile_options(-Wall -Werror=return-type -Wno-unused-function -Wno-sign-compare -Wno-unused-but-set-variable "$<$<COMPILE_LANGUAGE:CXX>:-Wno-reorder>" -Wno-unknown-pragmas -Wno-unused-value -Wno-unused-variable)
 	if(NOT CLANG)
 		# This one is very useful but has many false positives.
