--- a/ffmpeg/linux_arm64.sh
+++ b/ffmpeg/linux_arm64.sh
@@ -86,7 +86,7 @@
 ./configure --target-os=linux \
     --prefix=./linux/aarch64 \
     ${GENERAL} \
-    --extra-cflags=" -O3 -fasm -Wno-psabi -fno-short-enums -fno-strict-aliasing -finline-limit=300 " \
+    --extra-cflags=" -O3 -fasm -Wno-psabi -mno-outline-atomics -fno-short-enums -fno-strict-aliasing -finline-limit=300 " \
     --disable-shared \
     --enable-static \
     --enable-zlib \
