diff --git a/src/drivers/Joystick.cpp b/src/drivers/Joystick.cpp
index 4944723..978d1bc 100644
--- a/src/drivers/Joystick.cpp
+++ b/src/drivers/Joystick.cpp
@@ -175,16 +175,7 @@ void Init(void)
 
  try
  {
-  #ifdef HAVE_LINUX_JOYSTICK
-  main_driver = JoystickDriver_Linux_New();
-  #elif defined(WIN32)
-  {
-   hicp_driver = JoystickDriver_XInput_New();
-   main_driver = JoystickDriver_DX5_New(hicp_driver != NULL && hicp_driver->NumJoysticks() > 0);
-  }
-  #elif defined(HAVE_SDL)
   main_driver = JoystickDriver_SDL_New();
-  #endif
 
   if(hicp_driver != NULL)
   {
