diff -rupN pcsx_rearmed-87a0a6318564fc171d0b576da8ea7e629aa229e7.orig/frontend/libretro.c pcsx_rearmed-87a0a6318564fc171d0b576da8ea7e629aa229e7/frontend/libretro.c
--- pcsx_rearmed-87a0a6318564fc171d0b576da8ea7e629aa229e7.orig/frontend/libretro.c	2024-06-01 22:50:24.814992900 +0000
+++ pcsx_rearmed-87a0a6318564fc171d0b576da8ea7e629aa229e7/frontend/libretro.c	2024-06-03 01:07:11.613563019 +0000
@@ -596,6 +596,29 @@ void plat_trigger_vibrate(int pad, int l
 
    if (in_enable_vibration)
    {
+      int total_strength = 1000000;
+      FILE *duty_cycle_fs;
+
+      if (duty_cycle_fs = fopen("/sys/class/pwm/pwmchip1/pwm0/duty_cycle", "r+")) {
+         if (high > 0) 
+            total_strength = 1000000;
+         else if (low > 0)
+            total_strength = 600000;
+         else
+            total_strength = 1000000;
+
+         fprintf(duty_cycle_fs, "%u", total_strength);
+         fclose(duty_cycle_fs);
+      }
+
+      FILE *enable_fs;
+      if (enable_fs = fopen("/sys/class/pwm/pwmchip1/pwm0/enable", "r+")) {
+         int on = high || low ? 1 : 0;
+         fprintf(enable_fs, "%u", on);
+         fclose(enable_fs);     
+      }
+      
+      
       rumble_cb(pad, RETRO_RUMBLE_STRONG, high << 8);
       rumble_cb(pad, RETRO_RUMBLE_WEAK, low ? 0xffff : 0x0);
    }
