diff -rupN linux.orig/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts linux/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts
--- linux.orig/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts	2024-05-10 12:36:52.097707251 +0000
+++ linux/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts	2024-05-10 12:40:41.366968088 +0000
@@ -204,10 +204,10 @@
 	};
 };
 
-&codec {
-	allwinner,audio-routing = "Line Out", "LINEOUT";
-	status = "okay";
-};
+//&codec {
+//	allwinner,audio-routing = "Line Out", "LINEOUT";
+//	status = "okay";
+//};
 
 &cpu0 {
 	cpu-supply = <&reg_dcdc1>;
@@ -221,6 +221,7 @@
 	mali-supply = <&reg_dcdc2>;
 	power-domains = <&r_ccu 0>;
 	status = "okay";
+};
 	
 &hdmi {
 	hvcc-supply = <&reg_bldo1>;
diff -rupN linux.orig/drivers/iommu/sun50i-iommu.c linux/drivers/iommu/sun50i-iommu.c
--- linux.orig/drivers/iommu/sun50i-iommu.c	2024-05-10 12:36:52.757722451 +0000
+++ linux/drivers/iommu/sun50i-iommu.c	2024-05-10 12:46:20.350459373 +0000
@@ -1049,7 +1049,7 @@ err_unregister:
 err_remove_sysfs:
 	iommu_device_sysfs_remove(&iommu->iommu);
 
-err_free_group:
+err_free_cache:
 	iommu_group_put(iommu->group);
 
 	return ret;
