From 16418f7d23805d055a6d14e189170be6fbd8205f Mon Sep 17 00:00:00 2001
From: Ryan Walklin <ryan@testtoast.com>
Date: Tue, 28 May 2024 14:17:06 +1200
Subject: [PATCH] clk: sunxi-ng: h6-r: don't disable GPU power domain?

---
 drivers/clk/sunxi-ng/ccu-sun50i-h6-r.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/clk/sunxi-ng/ccu-sun50i-h6-r.c b/drivers/clk/sunxi-ng/ccu-sun50i-h6-r.c
index 991658fe8e84..b5af0ba8ae1b 100644
--- a/drivers/clk/sunxi-ng/ccu-sun50i-h6-r.c
+++ b/drivers/clk/sunxi-ng/ccu-sun50i-h6-r.c
@@ -239,8 +239,6 @@ static int sun50i_h616_ppu_pd_set_power(const struct sun50i_h616_ppu_pd *pd,
 {
 	if (power_on)
 		writel(0, pd->base + PD_H616_GPU_REG);
-	else
-		writel(1, pd->base + PD_H616_GPU_REG);
 
 	return 0;
 }
-- 
2.45.2

