From cb4cf770d993d14e87ee20a74f66bc4cec77b528 Mon Sep 17 00:00:00 2001
From: Philippe Simons <simons.philippe@gmail.com>
Date: Wed, 14 Aug 2024 19:01:59 +0200
Subject: [PATCH] rg35xx: enable gpu

---
 .../sun50i-h700-anbernic-rg35xx-2024.dts      | 36 +++++++++++++++++++
 1 file changed, 36 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts
index 99974e28c..37f852f4d 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts
@@ -224,12 +224,48 @@ panel_in_rgb: endpoint {
 			};
 		};
 	};
+
+	gpu_opp_table: opp-table-1 {
+		compatible = "operating-points-v2";
+
+		opp-420000000 {
+			opp-hz = /bits/ 64 <420000000>;
+			opp-microvolt = <940000>;
+		};
+		opp-456000000 {
+			opp-hz = /bits/ 64 <456000000>;
+			opp-microvolt = <940000>;
+		};
+		opp-504000000 {
+			opp-hz = /bits/ 64 <504000000>;
+			opp-microvolt = <940000>;
+		};
+		opp-552000000 {
+			opp-hz = /bits/ 64 <552000000>;
+			opp-microvolt = <940000>;
+		};
+		opp-600000000 {
+			opp-hz = /bits/ 64 <600000000>;
+			opp-microvolt = <940000>;
+		};
+		opp-648000000 {
+			opp-hz = /bits/ 64 <648000000>;
+			opp-microvolt = <940000>;
+		};
+	};
 };
 
 &cpu0 {
 	cpu-supply = <&reg_dcdc1>;
 };
 
+&gpu {
+	mali-supply = <&reg_dcdc2>;
+	power-domains = <&r_ccu 0>;
+	operating-points-v2 = <&gpu_opp_table>;
+	status = "okay";
+};
+
 &de {
 	status = "okay";
 };
-- 
2.34.1

