diff -rupN linux.orig/drivers/clk/sunxi-ng/ccu-sun50i-h6-r.c linux/drivers/clk/sunxi-ng/ccu-sun50i-h6-r.c
--- linux.orig/drivers/clk/sunxi-ng/ccu-sun50i-h6-r.c	2024-05-08 19:56:00.776486511 +0000
+++ linux/drivers/clk/sunxi-ng/ccu-sun50i-h6-r.c	2024-05-09 21:53:32.229470725 +0000
@@ -295,6 +295,7 @@ static int sun50i_h616_register_ppu(stru
 	ret = of_genpd_add_provider_onecell(dev->of_node, ppu);
 	if (ret)
 		dev_warn(dev, "Failed to add provider: %d\n", ret);
+		sun50i_h616_ppu_pd_power_on(&pd->genpd);
 
 	return 0;
 }
