From 3a27caf3d3b452ef98ea01c90d4a46d42a5e2cab Mon Sep 17 00:00:00 2001
From: Philippe Simons <simons.philippe@gmail.com>
Date: Tue, 23 Jul 2024 18:19:55 +0200
Subject: [PATCH 1/1] RG35XX-H: ROCKNIX joypad dts

---
 .../sun50i-h700-anbernic-rg35xx-h.dts         | 180 +++++++++++++-----
 1 file changed, 129 insertions(+), 51 deletions(-)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-h.dts b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-h.dts
index f57357e5ecd1..02e5a4b939da 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-h.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-h.dts
@@ -10,48 +10,142 @@ / {
 	model = "Anbernic RG35XX H";
 	compatible = "anbernic,rg35xx-h", "allwinner,sun50i-h700";
 
-	adc-joystick {
-		compatible = "adc-joystick";
-		io-channels = <&adc_mux 0>,
-		              <&adc_mux 1>,
-		              <&adc_mux 2>,
-		              <&adc_mux 3>;
+	joypad: rocknix-joypad {
+		compatible = "rocknix-joypad";
+		status = "okay";
+
+		/* the following 4 dts entries starting from joypad-name are important to keep
+		   compatibility with existing userspace programs such as ES and retroarch */
+		joypad-name = "retrogame_joypad";
+		joypad-product = <0x1101>;
+		joypad-revision = <0x0100>;
+		joypad-vendor = <0x484B>;
+
+		/* analog mux define */
+		io-channel-names = "key-RY", "key-RX", "key-LY", "key-LX";
+		io-channels = <&adc_mux 3>, <&adc_mux 2>, <&adc_mux 1>, <&adc_mux 0>;	
+
 		pinctrl-0 = <&joy_mux_pin>;
 		pinctrl-names = "default";
-		poll-interval = <60>;
-		#address-cells = <1>;
-		#size-cells = <0>;
 
-		axis@0 {
-			reg = <0>;
-			abs-flat = <32>;
-			abs-fuzz = <32>;
-			abs-range = <4096 0>;
-			linux,code = <ABS_X>;
-		};
+		/* adc calculate scale */
+		button-adc-scale = <2>;
 
-		axis@1 {
-			reg = <1>;
-			abs-flat = <32>;
-			abs-fuzz = <32>;
-			abs-range = <0 4096>;
-			linux,code = <ABS_Y>;
-		};
+		/* adc deadzone range (minimum should be around 12%) */
+		button-adc-deadzone = <216>;
 
-		axis@2 {
-			reg = <2>;
-			abs-flat = <32>;
-			abs-fuzz = <32>;
-			abs-range = <0 4096>;
-			linux,code = <ABS_RX>;
-		};
+		/* specifies fuzz value that is used to filter noise from
+		   the event stream. (around 3%) */
+		button-adc-fuzz = <54>;
+		button-adc-flat = <54>;
+
+		/* analog Stick data tuning value(precent)
+		   p = positive direction, n = negative direction
+		   report value = (real_adc_data * tuning_value) / 100 */
+		abs_x-p-tuning = <180>;
+		abs_x-n-tuning = <180>;
+		invert-absx;
+
+		abs_y-p-tuning = <180>;
+		abs_y-n-tuning = <180>;
+		invert-absy;
 
-		axis@3 {
-			reg = <3>;
-			abs-flat = <32>;
-			abs-fuzz = <32>;
-			abs-range = <4096 0>;
-			linux,code = <ABS_RY>;
+		abs_rx-p-tuning = <180>;
+		abs_rx-n-tuning = <180>;
+
+		abs_ry-p-tuning = <180>;
+		abs_ry-n-tuning = <180>;
+
+		/* poll device interval (ms), adc read interval */
+		poll-interval = <10>;
+
+		/* gpio button auto repeat set value : default disable */
+		/* autorepeat; */
+
+		sw1 {
+			gpios = <&pio 0 6 GPIO_ACTIVE_LOW>; /* PA6 */
+			label = "GPIO DPAD-UP";
+			linux,code = <BTN_DPAD_UP>;
+		};
+		sw2 {
+			gpios = <&pio 4 0 GPIO_ACTIVE_LOW>; /* PE0 */
+			label = "GPIO DPAD-DOWN";
+			linux,code = <BTN_DPAD_DOWN>;
+		};
+		sw3 {
+			gpios = <&pio 0 8 GPIO_ACTIVE_LOW>; /* PA8 */
+			label = "GPIO DPAD-LEFT";
+			linux,code = <BTN_DPAD_LEFT>;
+		};
+		sw4 {
+			gpios = <&pio 0 9 GPIO_ACTIVE_LOW>; /* PA9 */
+			label = "GPIO DPAD-RIGHT";
+			linux,code = <BTN_DPAD_RIGHT>;
+		};
+		sw5 {
+			gpios = <&pio 0 0 GPIO_ACTIVE_LOW>; /* PA0 */
+			label = "GPIO BTN-A";
+			linux,code = <BTN_EAST>;
+		};
+		sw6 {
+			gpios = <&pio 0 1 GPIO_ACTIVE_LOW>; /* PA1 */
+			label = "GPIO BTN-B";
+			linux,code = <BTN_SOUTH>;
+		};
+		sw7 {
+			gpios = <&pio 0 2 GPIO_ACTIVE_LOW>; /* PA2 */
+			label = "GPIO BTN-Y";
+			linux,code = <BTN_WEST>;
+		};
+		sw8 {
+			gpios = <&pio 0 3 GPIO_ACTIVE_LOW>; /* PA3 */
+			label = "GPIO BTN-X";
+			linux,code = <BTN_NORTH>;
+		};
+		sw9 {
+			gpios = <&pio 0 5 GPIO_ACTIVE_LOW>; /* PA5 */
+			label = "BTN_SELECT";
+			linux,code = <BTN_SELECT>;
+		};
+		sw10 {
+			gpios = <&pio 0 4 GPIO_ACTIVE_LOW>; /* PA4 */
+			label = "BTN_START";
+			linux,code = <BTN_START>;
+		};
+		sw11 {
+			gpios = <&pio 4 3 GPIO_ACTIVE_LOW>; /* PE3 */
+			label = "GPIO BTN_F";
+			linux,code = <BTN_MODE>;
+		};
+		sw12 {
+			gpios = <&pio 4 8 GPIO_ACTIVE_LOW>; /* PE8 */
+			label = "BTN_THUMBL";
+			linux,code = <BTN_THUMBL>;
+		};
+		sw13 {
+			gpios = <&pio 4 9 GPIO_ACTIVE_LOW>; /* PE9 */
+			label = "BTN_THUMBR";
+			linux,code = <BTN_THUMBR>;
+		};
+		sw14 {
+			gpios = <&pio 0 12 GPIO_ACTIVE_LOW>; /* PA12 */
+			label = "GPIO BTN_TR";
+			linux,code = <BTN_TR>;
+		};
+		sw15 {
+			gpios = <&pio 0 7 GPIO_ACTIVE_LOW>; /* PA7 */
+			label = "GPIO BTN_TR2";
+			linux,code = <BTN_TR2>;
+		};
+		sw16 {
+			gpios = <&pio 0 10 GPIO_ACTIVE_LOW>; /* PA10 */
+			label = "GPIO BTN_TL";
+			linux,code = <BTN_TL>;
+		};
+		sw17 {
+			gpios = <&pio 0 11 GPIO_ACTIVE_LOW>; /* PA11 */
+			label = "GPIO BTN_TL2";
+			linux,code = <BTN_TL2>;
 		};
 	};
 
@@ -73,21 +165,7 @@ gpio_mux: mux-controller {
 	};
 };
 
-&gpio_keys_gamepad {
-	button-thumbl {
-		label = "GPIO Thumb Left";
-		gpios = <&pio 4 8 GPIO_ACTIVE_LOW>; /* PE8 */
-		linux,input-type = <EV_KEY>;
-		linux,code = <BTN_THUMBL>;
-	};
-
-	button-thumbr {
-		label = "GPIO Thumb Right";
-		gpios = <&pio 4 9 GPIO_ACTIVE_LOW>; /* PE9 */
-		linux,input-type = <EV_KEY>;
-		linux,code = <BTN_THUMBR>;
-	};
-};
+/delete-node/ &gpio_keys_gamepad;
 
 &ehci1 {
 	status = "okay";
-- 
2.34.1

