diff -rupN linux.orig/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi linux/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
--- linux.orig/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2024-07-25 15:30:35.961984664 +0000
+++ linux/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2024-07-25 15:32:15.008302405 +0000
@@ -159,6 +159,35 @@
 		#size-cells = <1>;
 		ranges = <0x0 0x0 0x0 0x40000000>;
 
+        gpu_opp_table: opp-table-1 {
+            compatible = "operating-points-v2";
+
+            opp-420000000 {
+                opp-hz = /bits/ 64 <420000000>;
+                opp-microvolt = <940000>;
+            };
+            opp-456000000 {
+                opp-hz = /bits/ 64 <456000000>;
+                opp-microvolt = <940000>;
+            };
+            opp-504000000 {
+                opp-hz = /bits/ 64 <504000000>;
+                opp-microvolt = <940000>;
+            };
+            opp-552000000 {
+                opp-hz = /bits/ 64 <552000000>;
+                opp-microvolt = <940000>;
+            };
+            opp-600000000 {
+                opp-hz = /bits/ 64 <600000000>;
+                opp-microvolt = <940000>;
+            };
+            opp-648000000 {
+                opp-hz = /bits/ 64 <648000000>;
+                opp-microvolt = <940000>;
+            };
+        };
+
 		gpu: gpu@1800000 {
 			compatible = "allwinner,sun50i-h616-mali",
 				     "arm,mali-bifrost";
@@ -170,6 +199,7 @@
 			clocks = <&ccu CLK_GPU0>, <&ccu CLK_BUS_GPU>;
 			clock-names = "core", "bus";
 			resets = <&ccu RST_BUS_GPU>;
+			operating-points-v2 = <&gpu_opp_table>;
 			status = "disabled";
 		};
 
