diff -rupN linux.orig/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi linux/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
--- linux.orig/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2024-08-20 17:25:30.543982337 +0000
+++ linux/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2024-08-20 17:52:07.649300939 +0000
@@ -204,6 +204,35 @@
 			};
 		};
 
+               gpu_opp_table: opp-table-1 {
+                       compatible = "operating-points-v2";
+
+                       opp-420000000 {
+                               opp-hz = /bits/ 64 <420000000>;
+                               opp-microvolt = <940000>;
+                       };
+                       opp-456000000 {
+                               opp-hz = /bits/ 64 <456000000>;
+                               opp-microvolt = <940000>;
+                       };
+                       opp-504000000 {
+                               opp-hz = /bits/ 64 <504000000>;
+                               opp-microvolt = <940000>;
+                       };
+                       opp-552000000 {
+                               opp-hz = /bits/ 64 <552000000>;
+                               opp-microvolt = <940000>;
+                       };
+                       opp-600000000 {
+                               opp-hz = /bits/ 64 <600000000>;
+                               opp-microvolt = <940000>;
+                       };
+                       opp-648000000 {
+                               opp-hz = /bits/ 64 <648000000>;
+                               opp-microvolt = <940000>;
+                       };
+               };
+
 		gpu: gpu@1800000 {
 			compatible = "allwinner,sun50i-h616-mali",
 				     "arm,mali-bifrost";
@@ -215,6 +244,7 @@
 			clocks = <&ccu CLK_GPU0>, <&ccu CLK_BUS_GPU>;
 			clock-names = "core", "bus";
 			resets = <&ccu RST_BUS_GPU>;
+			operating-points-v2 = <&gpu_opp_table>;
 			status = "disabled";
 		};

