diff -rupN linux.orig/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts linux/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts
--- linux.orig/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts	2024-08-20 17:25:30.543982337 +0000
+++ linux/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts	2024-08-22 03:02:38.815679048 +0000
@@ -21,19 +21,19 @@
 		serial0 = &uart0;
 	};
 
-	backlight: backlight {
-		compatible = "gpio-backlight";
-		gpios = <&pio 3 28 GPIO_ACTIVE_HIGH>; // PD28
-		default-on;
+	//backlight: backlight {
+	//	compatible = "gpio-backlight";
+	//	gpios = <&pio 3 28 GPIO_ACTIVE_HIGH>; // PD28
+	//	default-on;
+	//};
+
+	 backlight: backlight {
+		compatible = "pwm-backlight";
+		pwms = <&pwm 0 50000 0>;
+		pinctrl-0 = <&lcd_backlight_pin>;
+		pinctrl-names = "default";
 	};
 
-	// backlight: backlight {
-	// 	compatible = "pwm-backlight";
-	// 	pwms = <&pwm 0 50000 0>;
-	// 	pinctrl-0 = <&lcd_backlight_pin>;
-	// 	pinctrl-names = "default";
-	// };
-
 	battery: battery {
 		compatible = "simple-battery";
 		constant-charge-current-max-microamp = <1024000>;
@@ -319,10 +319,10 @@
 	vcc-ph-supply = <&reg_cldo3>;
 	vcc-pi-supply = <&reg_cldo3>;
 
-	/* lcd_backlight_pin: pwm0-pin {
+	lcd_backlight_pin: pwm0-pin {
 		pins = "PD28";
 		function = "pwm0";
-	}; */
+	};
 	
 	lvs-gpio {
 		gpio-hog;
@@ -512,8 +512,8 @@
 	status = "okay";
 };
 
-/*&pwm {
+&pwm {
 	allwinner,pwm-paired-channel-clock-sources = "hosc", "hosc", "hosc";
 	allwinner,pwm-paired-channel-clock-prescales = <0>, <0>, <0>;
 	status = "okay";
-};*/
+};
