diff -rupN linux.orig/arch/arm64/boot/dts/allwinner/Makefile linux/arch/arm64/boot/dts/allwinner/Makefile
--- linux.orig/arch/arm64/boot/dts/allwinner/Makefile   2024-07-26 18:22:47.474113725 +0000
+++ linux/arch/arm64/boot/dts/allwinner/Makefile        2024-07-26 22:24:48.531627218 +0000
@@ -49,6 +49,6 @@ dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h618-
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h618-orangepi-zero3.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h618-transpeed-8k618-t.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h700-anbernic-rg35xx-2024.dtb
-dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h700-anbernic-rg35xx-h.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h700-anbernic-rg35xx-plus.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h700-anbernic-rg35xx-sp.dtb
+dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h700-anbernic-rg28xx.dtb
diff -rupN linux.orig/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg28xx.dts linux/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg28xx.dts
--- linux.orig/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg28xx.dts	1970-01-01 00:00:00.000000000 +0000
+++ linux/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg28xx.dts	2024-07-26 23:14:22.651172006 +0000
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/*
+ * Copyright (C) 2024 Ryan Walklin <ryan@testtoast.com>.
+ * Copyright (C) 2024 Chris Morgan <macroalpha82@gmail.com>.
+ */
+
+#include <dt-bindings/input/gpio-keys.h>
+#include "sun50i-h700-anbernic-rg35xx-2024.dts"
+
+/ {
+	model = "Anbernic RG28XX";
+	compatible = "anbernic,rg28xx", "allwinner,sun50i-h700";
+
+	/delete-node/ spi_lcd;
+
+    spi_lcd:    spi {
+        //compatible = "spi-gpio";
+        #address-cells = <1>;
+        #size-cells = <0>;
+
+        panel@0 {
+            compatible = "anbernic,rg28xx-panel", "sitronix,st7701";
+            reg = <0>;
+            spi-max-frequency = <3125000>;
+            VCC-supply = <&reg_lcd>;
+            IOVCC-supply = <&reg_lcd>;
+            reset-gpios = <&pio 8 14 GPIO_ACTIVE_HIGH>; /* LCD-RST: PI14 */
+            //backlight = <&backlight>;
+
+            port {
+                panel_in_rgb: endpoint {
+                    remote-endpoint = <&tcon_lcd0_out_lcd>;
+                };
+            };
+        };
+    };
+
+};
