diff -rupN linux.orig/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts linux/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts
--- linux.orig/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts	2024-08-20 17:25:30.543982337 +0000
+++ linux/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts	2024-08-22 03:04:28.638462602 +0000
@@ -62,7 +62,7 @@
 		clock-output-names = "ext_osc32k";
 	};
 
-	gpio_keys_gamepad: gpio-keys-gamepad {
+	gpio_keys_gamepad: h700-gamepad {
 		compatible = "gpio-keys";
 
 		button-a {
diff -rupN linux.orig/drivers/input/keyboard/gpio_keys.c linux/drivers/input/keyboard/gpio_keys.c
--- linux.orig/drivers/input/keyboard/gpio_keys.c	2024-08-20 17:25:32.144018964 +0000
+++ linux/drivers/input/keyboard/gpio_keys.c	2024-08-22 03:04:47.614943341 +0000
@@ -898,8 +898,8 @@ static int gpio_keys_probe(struct platfo
 
 	input->id.bustype = BUS_HOST;
 	input->id.vendor = 0x0001;
-	input->id.product = 0x0001;
-	input->id.version = 0x0100;
+	input->id.product = 0x7891;
+	input->id.version = 0x0511;
 
 	input->keycode = ddata->keymap;
 	input->keycodesize = sizeof(ddata->keymap[0]);
