diff -rupbN linux.orig/drivers/gpu/drm/panel/panel-sitronix-st7701.c linux/drivers/gpu/drm/panel/panel-sitronix-st7701.c
--- linux.orig/drivers/gpu/drm/panel/panel-sitronix-st7701.c	2024-11-09 15:37:39.898556064 +0000
+++ linux/drivers/gpu/drm/panel/panel-sitronix-st7701.c	2024-11-17 14:42:52.356446105 +0000
@@ -1036,17 +1036,17 @@ static const struct st7701_panel_desc rg
 };
 
 static const struct drm_display_mode rg28xx_mode = {
-	.clock		= 22325,
+	.clock		= 24000,
 
 	.hdisplay	= 480,
-	.hsync_start	= 480 + 40,
-	.hsync_end	= 480 + 40 + 4,
-	.htotal		= 480 + 40 + 4 + 20,
+	.hsync_start	= 480 + 50,
+	.hsync_end	= 480 + 50 + 6,
+	.htotal		= 480 + 50 + 6 + 50,
 
 	.vdisplay	= 640,
 	.vsync_start	= 640 + 2,
-	.vsync_end	= 640 + 2 + 40,
-	.vtotal		= 640 + 2 + 40 + 16,
+	.vsync_end	= 640 + 24 + 6,
+	.vtotal		= 640 + 24 + 6 + 16,
 
 	.width_mm	= 44,
 	.height_mm	= 58,
