diff -ur a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg28xx.dts b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg28xx.dts
--- a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg28xx.dts	2024-12-24 12:16:07.434240834 +0000
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg28xx.dts	2024-12-24 12:21:15.784286265 +0000
@@ -10,6 +10,7 @@
 / {
 	model = "Anbernic RG28XX";
 	compatible = "anbernic,rg28xx", "allwinner,sun50i-h700";
+	rocknix-dt-id = "sun50i-h700-anbernic-rg28xx";
 };
 
 &panel {
diff -ur a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg34xx.dts b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg34xx.dts
--- a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg34xx.dts	2024-12-24 12:16:07.634240872 +0000
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg34xx.dts	2024-12-24 12:21:35.804288763 +0000
@@ -9,6 +9,7 @@
 / {
 	model = "Anbernic RG34XX";
 	compatible = "anbernic,rg34xx", "allwinner,sun50i-h700";
+	rocknix-dt-id = "sun50i-h700-anbernic-rg34xx";
 };
 
 &panel {
diff -ur a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024-rev6-panel.dts b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024-rev6-panel.dts
--- a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024-rev6-panel.dts	2024-12-24 12:16:07.534240856 +0000
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024-rev6-panel.dts	2024-12-24 12:25:16.464311447 +0000
@@ -6,6 +6,10 @@
 
 #include "sun50i-h700-anbernic-rg35xx-2024.dts"
 
+/ {
+	rocknix-dt-id = "sun50i-h700-anbernic-rg35xx-2024-rev6-panel";
+};
+
 &panel {
 	compatible = "anbernic,rg35xx-plus-rev6-panel";
 };
diff -ur a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts
--- a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts	2024-12-24 12:16:07.664240881 +0000
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-2024.dts	2024-12-24 12:21:50.854290587 +0000
@@ -16,6 +16,7 @@
 	model = "Anbernic RG35XX 2024";
 	chassis-type = "handset";
 	compatible = "anbernic,rg35xx-2024", "allwinner,sun50i-h700";
+	rocknix-dt-id = "sun50i-h700-anbernic-rg35xx-2024";
 
 	aliases {
 		mmc0 = &mmc0;
diff -ur a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-h-rev6-panel.dts b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-h-rev6-panel.dts
--- a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-h-rev6-panel.dts	2024-12-24 12:16:07.534240856 +0000
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-h-rev6-panel.dts	2024-12-24 12:25:36.484323945 +0000
@@ -6,6 +6,10 @@
 
 #include "sun50i-h700-anbernic-rg35xx-h.dts"
 
+/ {
+	rocknix-dt-id = "sun50i-h700-anbernic-rg35xx-h-rev6-panel";
+};
+
 &panel {
 	compatible = "anbernic,rg35xx-plus-rev6-panel";
 };
diff -ur a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-h.dts b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-h.dts
--- a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-h.dts	2024-12-24 12:16:07.624240873 +0000
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-h.dts	2024-12-24 12:22:18.754293123 +0000
@@ -9,6 +9,7 @@
 / {
 	model = "Anbernic RG35XX H";
 	compatible = "anbernic,rg35xx-h", "allwinner,sun50i-h700";
+	rocknix-dt-id = "sun50i-h700-anbernic-rg35xx-h";
 
 	reg_vcc3v8_usb: regulator-vcc3v8-usb {
 		compatible = "regulator-fixed";
diff -ur a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-plus-rev6-panel.dts b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-plus-rev6-panel.dts
--- a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-plus-rev6-panel.dts	2024-12-24 12:16:07.534240856 +0000
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-plus-rev6-panel.dts	2024-12-24 12:25:53.084329227 +0000
@@ -6,6 +6,10 @@
 
 #include "sun50i-h700-anbernic-rg35xx-plus.dts"
 
+/ {
+	rocknix-dt-id = "sun50i-h700-anbernic-rg35xx-plus-rev6-panel";
+};
+
 &panel {
 	compatible = "anbernic,rg35xx-plus-rev6-panel";
 };
diff -ur a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-plus.dts b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-plus.dts
--- a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-plus.dts	2024-12-24 12:16:07.694240885 +0000
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-plus.dts	2024-12-24 12:22:35.594294024 +0000
@@ -8,6 +8,7 @@
 / {
 	model = "Anbernic RG35XX Plus";
 	compatible = "anbernic,rg35xx-plus", "allwinner,sun50i-h700";
+	rocknix-dt-id = "sun50i-h700-anbernic-rg35xx-plus";
 
 	wifi_pwrseq: pwrseq {
 		compatible = "mmc-pwrseq-simple";
diff -ur a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-sp-rev6-panel.dts b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-sp-rev6-panel.dts
--- a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-sp-rev6-panel.dts	2024-12-24 12:16:07.534240856 +0000
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-sp-rev6-panel.dts	2024-12-24 12:26:06.924337562 +0000
@@ -6,6 +6,10 @@
 
 #include "sun50i-h700-anbernic-rg35xx-sp.dts"
 
+/ {
+	rocknix-dt-id = "sun50i-h700-anbernic-rg35xx-sp-rev6-panel";
+};
+
 &panel {
 	compatible = "anbernic,rg35xx-plus-rev6-panel";
 };
diff -ur a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-sp.dts b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-sp.dts
--- a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-sp.dts	2024-12-19 17:13:24.000000000 +0000
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg35xx-sp.dts	2024-12-24 12:23:03.124296054 +0000
@@ -10,6 +10,7 @@
 / {
 	model = "Anbernic RG35XX SP";
 	compatible = "anbernic,rg35xx-sp", "allwinner,sun50i-h700";
+	rocknix-dt-id = "sun50i-h700-anbernic-rg35xx-sp";
 
 	gpio-keys-lid {
 		compatible = "gpio-keys";
diff -ur a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg40xx-h.dts b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg40xx-h.dts
--- a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg40xx-h.dts	2024-12-24 12:16:07.584240864 +0000
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg40xx-h.dts	2024-12-24 12:23:16.964297942 +0000
@@ -8,6 +8,7 @@
 / {
 	model = "Anbernic RG40XX H";
 	compatible = "anbernic,rg40xx-h", "allwinner,sun50i-h700";
+	rocknix-dt-id = "sun50i-h700-anbernic-rg40xx-h.dts";
 };
 
 &joypad {
diff -ur a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg40xx-v.dts b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg40xx-v.dts
--- a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg40xx-v.dts	2024-12-24 12:16:07.604240868 +0000
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rg40xx-v.dts	2024-12-24 12:23:32.134312689 +0000
@@ -8,6 +8,7 @@
 / {
 	model = "Anbernic RG40XX V";
 	compatible = "anbernic,rg40xx-V", "allwinner,sun50i-h700";
+	rocknix-dt-id = "sun50i-h700-anbernic-rg40xx-v";
 };
 
 &joypad {
diff -ur a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rgcubexx.dts b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rgcubexx.dts
--- a/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rgcubexx.dts	2024-12-24 12:16:07.614240868 +0000
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h700-anbernic-rgcubexx.dts	2024-12-24 12:23:45.914316697 +0000
@@ -9,6 +9,7 @@
 / {
 	model = "Anbernic RG CubeXX";
 	compatible = "anbernic,rgcubexx", "allwinner,sun50i-h700";
+	rocknix-dt-id = "sun50i-h700-anbernic-rgcubexx";
 };
 
 &joypad {
