diff -rupN linux.orig/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi linux/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
--- linux.orig/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2024-08-20 17:25:30.543982337 +0000
+++ linux/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi	2024-08-22 03:08:20.605186251 +0000
@@ -825,20 +825,20 @@
 			status = "disabled";
 		};
 
-		codec: codec@05096000 {
-			#sound-dai-cells = <0>;
-			compatible = "allwinner,sun50i-h616-codec";
-			reg = <0x05096000 0x31c>;
-			interrupts = <GIC_SPI 58 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&ccu CLK_BUS_AUDIO_CODEC>,
-				 <&ccu CLK_AUDIO_CODEC_1X>,
-				 <&ccu CLK_AUDIO_CODEC_4X>;
-			clock-names = "apb", "codec", "audio-codec-4x";
-			resets = <&ccu RST_BUS_AUDIO_CODEC>;
-			dmas = <&dma 6>;
-			dma-names = "tx";
-			status = "disabled";
-		};
+		//codec: codec@05096000 {
+		//	#sound-dai-cells = <0>;
+		//	compatible = "allwinner,sun50i-h616-codec";
+		//	reg = <0x05096000 0x31c>;
+		//	interrupts = <GIC_SPI 58 IRQ_TYPE_LEVEL_HIGH>;
+		//	clocks = <&ccu CLK_BUS_AUDIO_CODEC>,
+		//		 <&ccu CLK_AUDIO_CODEC_1X>,
+		//		 <&ccu CLK_AUDIO_CODEC_4X>;
+		//	clock-names = "apb", "codec", "audio-codec-4x";
+		//	resets = <&ccu RST_BUS_AUDIO_CODEC>;
+		//	dmas = <&dma 6>;
+		//	dma-names = "tx";
+		//	status = "disabled";
+		//};
 
 		usbotg: usb@5100000 {
 			compatible = "allwinner,sun50i-h616-musb",
