diff -rupN geolith-libretro.orig/libretro/Makefile geolith-libretro/libretro/Makefile
--- geolith-libretro.orig/libretro/Makefile	2024-04-18 21:11:00.166799951 +0000
+++ geolith-libretro/libretro/Makefile	2024-04-18 22:08:26.503486311 +0000
@@ -375,6 +375,16 @@ else ifeq ($(platform), rpi4)
 	FLAGS += -mcpu=cortex-a72 -mtune=cortex-a72
 	FLAGS += -fomit-frame-pointer -ffast-math
 
+# @DEVICE@
+else ifeq ($(platform), @DEVICE@)
+        TARGET := $(TARGET_NAME)_libretro.so
+        fpic := -fPIC
+        SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
+        CFLAGS+=-fsigned-char
+        FLAGS += -DARM
+        FLAGS += -mcpu=@TARGET_CPU@ -mtune=@TARGET_CPU@
+        FLAGS += -fomit-frame-pointer -ffast-math
+
 else ifeq ($(platform), emscripten)
 	TARGET := $(TARGET_NAME)_libretro_$(platform).bc
 	STATIC_LINKING = 1
