Subject: [PATCH] Enabled extended namehints in alsa.conf

diff --git a/src/conf/alsa.conf b/src/conf/alsa.conf
--- a/src/conf/alsa.conf
+++ b/src/conf/alsa.conf
@@ -96,11 +96,11 @@
 #
 
 # show all name hints also for definitions without hint {} section
-defaults.namehint.showall off
+defaults.namehint.showall on
 # show just basic name hints
 defaults.namehint.basic on
 # show extended name hints
-defaults.namehint.extended off
+defaults.namehint.extended on
 #
 defaults.ctl.card 0
 defaults.pcm.card 0
