From 191cd427d386323c11a02c186f4dab0dc3a9e957 Mon Sep 17 00:00:00 2001
From: John Williams <porschemad911@gmail.com>
Date: Thu, 14 May 2026 14:01:00 +1000
Subject: [PATCH 2/2] Qualcomm - battery - power_now

---
 src/battery.cpp | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/src/battery.cpp b/src/battery.cpp
index 947f446..3d5239b 100644
--- a/src/battery.cpp
+++ b/src/battery.cpp
@@ -111,16 +111,7 @@ float BatteryStats::getPower() {
             return 0.0f;
         }
 
-        // Prefer power_now (µW) when available.
-        if (fs::exists(power_now)) {
-            std::ifstream input(power_now);
-            std::string line;
-            if (std::getline(input, line)) {
-                power_w += std::fabs(stof(line)) / 1000000.0f;
-            }
-            continue;
-        }
-
+        // Prefer current_now and voltage_now, but fall back to power_now (µW).
         if (fs::exists(current_now) && fs::exists(voltage_now)) {
             float i_ua = 0.0f;
             float v_uv = 0.0f;
@@ -142,6 +133,15 @@ float BatteryStats::getPower() {
 
             power_w += (std::fabs(i_ua) * std::fabs(v_uv)) * 1e-12f;
         }
+        
+        else if (fs::exists(power_now)) {
+            std::ifstream input(power_now);
+            std::string line;
+            if (std::getline(input, line)) {
+                power_w += std::fabs(stof(line)) / 10000000.0f;
+            }
+            continue;
+        }
     }
 
     return power_w;
-- 
2.47.3

