diff -urN a/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c b/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c
--- a/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c	2026-09-02 03:25:02.000000000 +0000
+++ b/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c	2026-09-11 01:23:13.427100529 +0000
@@ -4514,6 +4514,9 @@
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 8, 0)
  struct wireless_dev *wdev,
 #endif
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 17, 0)
+	int radio_idx, unsigned int link_id,
+#endif
 	int *mbm)
 {
     #if LINUX_VERSION_CODE < KERNEL_VERSION(3, 8, 0)
diff -urN a/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_rx.c b/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_rx.c
--- a/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_rx.c	2026-09-02 03:25:02.000000000 +0000
+++ b/src/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_rx.c	2026-09-11 01:23:13.428100558 +0000
@@ -1638,7 +1638,7 @@
             reord_rxframe_free(&rx_priv->freeq_lock, &rx_priv->rxframes_freequeue, &req->rxframe_list);
         }
 
-		AICWFDBG(LOGINFO, "reord dinit in_irq():%d in_atomic:%d in_softirq:%d\r\n", (int)in_irq()
+		AICWFDBG(LOGINFO, "reord dinit in_hardirq():%d in_atomic:%d in_softirq:%d\r\n", (int)in_hardirq()
 			,(int)in_atomic(), (int)in_softirq());
         spin_unlock_bh(&preorder_ctrl->reord_list_lock);
     }
