diff --git a/externals/nx_tzdb/tzdb_to_nx/src/CMakeLists.txt b/externals/nx_tzdb/tzdb_to_nx/src/CMakeLists.txt
--- a/externals/nx_tzdb/tzdb_to_nx/src/CMakeLists.txt
+++ b/externals/nx_tzdb/tzdb_to_nx/src/CMakeLists.txt
@@ -7,5 +7,11 @@ add_compile_options(
 
 include_directories(.)
 
-add_subdirectory(tzdb2nx)
+include(ExternalProject)
+ExternalProject_Add(tzdb2nx
+    SOURCE_DIR ${CMAKE_CURRENT_LIST_DIR}/tzdb2nx/
+    CMAKE_ARGS -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_CXX_FLAGS= -DCMAKE_C_FLAGS=
+    INSTALL_COMMAND cmake -E echo "No install step"
+    BUILD_IN_SOURCE ON)
+set(TZDB2NX_PATH "${CMAKE_CURRENT_LIST_DIR}/tzdb2nx/tzdb2nx" CACHE PATH "Path to tzdb2nx path")
 add_subdirectory(tzdb)
