diff --git a/src/suyu/main.cpp b/src/suyu/main.cpp
index 991ff6776..12c4d1418 100644
--- a/src/suyu/main.cpp
+++ b/src/suyu/main.cpp
@@ -5641,6 +5641,7 @@ int main(int argc, char* argv[]) {
     // Disables the "?" button on all dialogs. Disabled by default on Qt6.
     QCoreApplication::setAttribute(Qt::AA_DisableWindowContextHelpButton);
 #endif
+    QCoreApplication::setAttribute(Qt::AA_DontUseNativeDialogs);
 
     // Enables the core to make the qt created contexts current on std::threads
     QCoreApplication::setAttribute(Qt::AA_DontCheckOpenGLContextThreadAffinity);
