diff --git a/arch/arm64/boot/dts/rockchip/rk3566.dtsi b/arch/arm64/boot/dts/rockchip/rk3566.dtsi
index 6c4b17d27bdc..f4050ae1cbf4 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3566.dtsi
@@ -33,3 +33,11 @@ &usb_host0_xhci {
 &vop {
 	compatible = "rockchip,rk3566-vop";
 };
+
+&cpu0_opp_table {
+	opp-1992000000 {
+		opp-hz = /bits/ 64 <1992000000>;
+		opp-microvolt = <1150000 1150000 1150000>;
+		turbo-mode;
+	};
+};
--- linux-6.9-rc3/arch/arm64/boot/dts/rockchip/rk356x.dtsi.orig	2024-04-19 15:00:45.070637916 +0300
+++ linux-6.9-rc3/arch/arm64/boot/dts/rockchip/rk356x.dtsi	2024-04-19 15:03:01.108366560 +0300
@@ -1507,7 +1507,7 @@
 
 			cooling-maps {
 				map0 {
-					trip = <&cpu_alert0>;
+					trip = <&cpu_alert1>;
 					cooling-device =
 						<&cpu0 THERMAL_NO_LIMIT THERMAL_NO_LIMIT>,
 						<&cpu1 THERMAL_NO_LIMIT THERMAL_NO_LIMIT>,
