diff --git a/arch/arm64/boot/dts/rockchip/Makefile b/arch/arm64/boot/dts/rockchip/Makefile
index f906a868b71a..ddb728e35a4e 100644
--- a/arch/arm64/boot/dts/rockchip/Makefile
+++ b/arch/arm64/boot/dts/rockchip/Makefile
@@ -71,6 +71,8 @@ dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rockpro64.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-sapphire.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-sapphire-excavator.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399pro-rock-pi-n10.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += overlays/rk3566-undervolt-cpu-l1.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += overlays/rk3566-undervolt-cpu-l2.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-anbernic-rg-arc-d.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-anbernic-rg-arc-s.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3566-anbernic-rg353p.dtb
diff --git a/arch/arm64/boot/dts/rockchip/overlays/rk3566-undervolt-cpu-l1.dts b/arch/arm64/boot/dts/rockchip/overlays/rk3566-undervolt-cpu-l1.dts
new file mode 100644
index 000000000000..0c5c504ee570
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/overlays/rk3566-undervolt-cpu-l1.dts
@@ -0,0 +1,35 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+fragment@0 {
+//target = <&cpu0_opp_table>;
+        target-path = "/opp-table-0";
+__overlay__ {
+opp-408000000 {
+opp-microvolt = <800000 800000 1150000>;
+};
+opp-600000000 {
+opp-microvolt = <800000 800000 1150000>;
+};
+opp-816000000 {
+opp-microvolt = <800000 800000 1150000>;
+};
+opp-1104000000 {
+opp-microvolt = <800000 800000 1150000>;
+};
+opp-1416000000 {
+opp-microvolt = <850000 850000 1150000>;
+};
+opp-1608000000 {
+opp-microvolt = <900000 900000 1150000>;
+};
+opp-1800000000 {
+opp-microvolt = <950000 950000 1150000>;
+};
+opp-1992000000 {
+opp-microvolt = <1000000 1000000 1150000>;
+};
+};
+};
+};
diff --git a/arch/arm64/boot/dts/rockchip/overlays/rk3566-undervolt-cpu-l2.dts b/arch/arm64/boot/dts/rockchip/overlays/rk3566-undervolt-cpu-l2.dts
new file mode 100644
index 000000000000..60036cedee2a
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/overlays/rk3566-undervolt-cpu-l2.dts
@@ -0,0 +1,35 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+fragment@0 {
+//target = <&cpu0_opp_table>;
+        target-path = "/opp-table-0";
+__overlay__ {
+opp-408000000 {
+opp-microvolt = <775000 775000 1150000>;
+};
+opp-600000000 {
+opp-microvolt = <800000 800000 1150000>;
+};
+opp-816000000 {
+opp-microvolt = <800000 800000 1150000>;
+};
+opp-1104000000 {
+opp-microvolt = <800000 800000 1150000>;
+};
+opp-1416000000 {
+opp-microvolt = <850000 850000 1150000>;
+};
+opp-1608000000 {
+opp-microvolt = <900000 900000 1150000>;
+};
+opp-1800000000 {
+opp-microvolt = <925000 925000 1150000>;
+};
+opp-1992000000 {
+opp-microvolt = <950000 950000 1150000>;
+};
+};
+};
+};
