diff --git a/drivers/gpu/drm/panel/panel-sitronix-st7703.c b/drivers/gpu/drm/panel/panel-sitronix-st7703.c
index a3e142f156d5..c5dab15ea5cf 100644
--- a/drivers/gpu/drm/panel/panel-sitronix-st7703.c
+++ b/drivers/gpu/drm/panel/panel-sitronix-st7703.c
@@ -507,7 +507,7 @@ static const struct drm_display_mode rgb30panel_mode = {
 	.vsync_start	= 720 + 15,
 	.vsync_end	= 720 + 15 + 3,
 	.vtotal		= 720 + 15 + 3 + 11,
-	.clock		= 36570,
+	.clock		= 37000,
 	.flags		= DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC,
 	.width_mm	= 76,
 	.height_mm	= 76,
