diff -rupN linux-6.9.2.orig/arch/arm64/boot/dts/rockchip/rk3566-powkiddy-rgb30.dts linux-6.9.2/arch/arm64/boot/dts/rockchip/rk3566-powkiddy-rgb30.dts
--- linux-6.9.2.orig/arch/arm64/boot/dts/rockchip/rk3566-powkiddy-rgb30.dts	2024-05-31 04:28:12.285455853 +0000
+++ linux-6.9.2/arch/arm64/boot/dts/rockchip/rk3566-powkiddy-rgb30.dts	2024-05-31 06:02:00.739585316 +0000
@@ -55,3 +55,7 @@
 		};
 	};
 };
+
+&pwm5 {
+	status = "okay";
+};
