diff --git a/arch/arm64/boot/dts/rockchip/rk3566-powkiddy-rk2023.dtsi b/arch/arm64/boot/dts/rockchip/rk3566-powkiddy-rk2023.dtsi
index d5e2a9a4974b..26152f711fb7 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-powkiddy-rk2023.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3566-powkiddy-rk2023.dtsi
@@ -18,6 +18,10 @@ aliases {
 		mmc3 = &sdmmc2;
 	};
 
+	chosen: chosen {
+		stdout-path = "serial2:1500000n8";
+	};
+
 	backlight: backlight {
 		compatible = "pwm-backlight";
 		power-supply = <&vcc_sys>;
@@ -792,6 +797,10 @@ bluetooth: bluetooth {
 	};
 };
 
+&uart2 {
+	status = "okay";
+};
+
 &usb_host0_xhci {
 	dr_mode = "peripheral";
 	phys = <&usb2phy0_otg>;
