diff --git a/src/frontend/qt_sdl/Config.cpp b/src/frontend/qt_sdl/Config.cpp
index f767fcc9..ba3e18a8 100644
--- a/src/frontend/qt_sdl/Config.cpp
+++ b/src/frontend/qt_sdl/Config.cpp
@@ -174,6 +174,8 @@ LegacyEntry LegacyFile[] =
     {"HKKey_GuitarGripRed",       0, "Keyboard.HK_GuitarGripRed", true},
     {"HKKey_GuitarGripYellow",    0, "Keyboard.HK_GuitarGripYellow", true},
     {"HKKey_GuitarGripBlue",      0, "Keyboard.HK_GuitarGripBlue", true},
+    {"HKKey_SaveState",           0, "Keyboard.HK_SaveState", true},
+    {"HKKey_LoadState",           0, "Keyboard.HK_LoadState", true},
 
     {"HKJoy_Lid",                 0, "Joystick.HK_Lid", true},
     {"HKJoy_Mic",                 0, "Joystick.HK_Mic", true},
@@ -194,6 +196,8 @@ LegacyEntry LegacyFile[] =
     {"HKJoy_GuitarGripRed",       0, "Joystick.HK_GuitarGripRed", true},
     {"HKJoy_GuitarGripYellow",    0, "Joystick.HK_GuitarGripYellow", true},
     {"HKJoy_GuitarGripBlue",      0, "Joystick.HK_GuitarGripBlue", true},
+    {"HKJoy_SaveState",           0, "Joystick.HK_SaveState", true},
+    {"HKJoy_LoadState",           0, "Joystick.HK_LoadState", true},
 
     {"JoystickID", 0, "JoystickID", true},
 
diff --git a/src/frontend/qt_sdl/EmuInstance.h b/src/frontend/qt_sdl/EmuInstance.h
index c8e756ea..563d2ec1 100755
--- a/src/frontend/qt_sdl/EmuInstance.h
+++ b/src/frontend/qt_sdl/EmuInstance.h
@@ -55,6 +55,8 @@ enum
     HK_GuitarGripRed,
     HK_GuitarGripYellow,
     HK_GuitarGripBlue,
+    HK_SaveState,
+    HK_LoadState,
     HK_MAX
 };
 
diff --git a/src/frontend/qt_sdl/EmuInstanceInput.cpp b/src/frontend/qt_sdl/EmuInstanceInput.cpp
index 866eb29a..4ac4f1ad 100644
--- a/src/frontend/qt_sdl/EmuInstanceInput.cpp
+++ b/src/frontend/qt_sdl/EmuInstanceInput.cpp
@@ -66,7 +66,9 @@ const char* EmuInstance::hotkeyNames[HK_MAX] =
     "HK_GuitarGripGreen",
     "HK_GuitarGripRed",
     "HK_GuitarGripYellow",
-    "HK_GuitarGripBlue"
+    "HK_GuitarGripBlue",
+    "HK_SaveState",
+    "HK_LoadState"
 };
 
 std::shared_ptr<SDL_mutex> EmuInstance::joyMutexGlobal = nullptr;
diff --git a/src/frontend/qt_sdl/EmuThread.cpp b/src/frontend/qt_sdl/EmuThread.cpp
index e754d982..9a09fae3 100755
--- a/src/frontend/qt_sdl/EmuThread.cpp
+++ b/src/frontend/qt_sdl/EmuThread.cpp
@@ -78,6 +78,8 @@ void EmuThread::attachWindow(MainWindow* window)
     connect(this, SIGNAL(autoScreenSizingChange(int)), window->panel, SLOT(onAutoScreenSizingChanged(int)));
     connect(this, SIGNAL(windowFullscreenToggle()), window, SLOT(onFullscreenToggled()));
     connect(this, SIGNAL(screenEmphasisToggle()), window, SLOT(onScreenEmphasisToggled()));
+    connect(this, SIGNAL(SaveState()), window->actSaveState[1], SLOT(trigger()));
+    connect(this, SIGNAL(LoadState()), window->actLoadState[1], SLOT(trigger()));
 
     if (window->winHasMenu())
     {
@@ -168,6 +170,8 @@ void EmuThread::run()
 
         if (emuInstance->hotkeyPressed(HK_SwapScreens)) emit swapScreensToggle();
         if (emuInstance->hotkeyPressed(HK_SwapScreenEmphasis)) emit screenEmphasisToggle();
+        if (emuInstance->hotkeyPressed(HK_SaveState)) emit SaveState();
+        if (emuInstance->hotkeyPressed(HK_LoadState)) emit LoadState();
 
         if (emuStatus == emuStatus_Running || emuStatus == emuStatus_FrameStep)
         {
diff --git a/src/frontend/qt_sdl/EmuThread.h b/src/frontend/qt_sdl/EmuThread.h
index 7a7f0600..60892b0e 100755
--- a/src/frontend/qt_sdl/EmuThread.h
+++ b/src/frontend/qt_sdl/EmuThread.h
@@ -161,6 +161,9 @@ signals:
     void swapScreensToggle();
     void screenEmphasisToggle();
 
+    void SaveState();
+    void LoadState();
+
     void syncVolumeLevel();
 
 private:
diff --git a/src/frontend/qt_sdl/InputConfig/InputConfigDialog.h b/src/frontend/qt_sdl/InputConfig/InputConfigDialog.h
index 0dd38442..64517daa 100644
--- a/src/frontend/qt_sdl/InputConfig/InputConfigDialog.h
+++ b/src/frontend/qt_sdl/InputConfig/InputConfigDialog.h
@@ -67,7 +67,9 @@ static constexpr std::initializer_list<int> hk_general =
     HK_SwapScreenEmphasis,
     HK_PowerButton,
     HK_VolumeUp,
-    HK_VolumeDown
+    HK_VolumeDown,
+    HK_SaveState,
+    HK_LoadState
 };
 
 static constexpr std::initializer_list<const char*> hk_general_labels =
@@ -87,7 +89,9 @@ static constexpr std::initializer_list<const char*> hk_general_labels =
     "Swap screen emphasis",
     "DSi Power button",
     "DSi Volume up",
-    "DSi Volume down"
+    "DSi Volume down",
+    "Save State",
+    "Load State"
 };
 
 static_assert(hk_general.size() == hk_general_labels.size());
