diff --git a/src/frontend/qt_sdl/Config.cpp b/src/frontend/qt_sdl/Config.cpp
index f767fcc9..86981f1f 100644
--- a/src/frontend/qt_sdl/Config.cpp
+++ b/src/frontend/qt_sdl/Config.cpp
@@ -39,10 +39,9 @@ namespace Config
 using namespace melonDS;
 
 
-const char* kConfigFile = "melonDS.toml";
+const char* kConfigFile = "/storage/.config/melonDS/melonDS.toml";
 
-const char* kLegacyConfigFile = "melonDS.ini";
-const char* kLegacyUniqueConfigFile = "melonDS.%d.ini";
+const char* kLegacyConfigFile = "/storage/.config/melonDS/melonDS.ini";
 
 toml::value RootTable;

