diff --git a/src/frontend/qt_sdl/Config.cpp b/src/frontend/qt_sdl/Config.cpp
index 156d539..67978c0 100644
--- a/src/frontend/qt_sdl/Config.cpp
+++ b/src/frontend/qt_sdl/Config.cpp
@@ -208,6 +208,10 @@ LegacyEntry LegacyFile[] =
     {"IntegerScaling", 1, "Window0.IntegerScaling", true},
     {"ScreenAspectTop",0, "Window0.ScreenAspectTop", true},
     {"ScreenAspectBot",0, "Window0.ScreenAspectBot", true},
+    {"Screen1Layout",   0, "Window1.ScreenLayout", true},
+    {"Screen1Enabled",  1, "Window1.Enabled", true},
+    {"Screen1Sizing",   0, "Window1.ScreenSizing", true},
+
 
     {"ScreenFilter",        1, "Screen.Filter", false},
     {"ScreenUseGL",         1, "Screen.UseGL", false},
diff --git a/src/frontend/qt_sdl/main.cpp b/src/frontend/qt_sdl/main.cpp
index 6514266..cc1b892 100644
--- a/src/frontend/qt_sdl/main.cpp
+++ b/src/frontend/qt_sdl/main.cpp
@@ -380,9 +380,12 @@ int main(int argc, char** argv)
         if (memberSyntaxUsed) printf("Warning: use the a.zip|b.nds format at your own risk!\n");
 
         win->preloadROMs(dsfile, gbafile, options->boot);
-
         if (options->fullscreen)
+         QTimer::singleShot(500, [=]() {
+            emuInstances[0]->doOnAllWindows([](MainWindow* win){
             win->toggleFullscreen();
+            });
+         });
     }
 
     int ret = melon.exec();
