diff --git a/arch/arm/include/asm/arch-meson/boot.h b/arch/arm/include/asm/arch-meson/boot.h
index c67d12d06c..301dc9aded 100644
--- a/arch/arm/include/asm/arch-meson/boot.h
+++ b/arch/arm/include/asm/arch-meson/boot.h
@@ -20,5 +20,6 @@
 int meson_get_boot_device(void);
 
 int meson_get_soc_rev(char *buff, size_t buff_len);
+int meson_get_soc_pack(void);
 
 #endif /* __MESON_BOOT_H__ */
diff --git a/arch/arm/mach-meson/board-info.c b/arch/arm/mach-meson/board-info.c
index b4058f5932..cce9f64c0c 100644
--- a/arch/arm/mach-meson/board-info.c
+++ b/arch/arm/mach-meson/board-info.c
@@ -190,3 +190,15 @@ int meson_get_soc_rev(char *buff, size_t buff_len)
 	/* Write SoC info */
 	return snprintf(buff, buff_len, "%x", socinfo_to_minor(socinfo));
 }
+
+int meson_get_soc_pack(void)
+{
+	unsigned int socinfo;
+
+	socinfo = get_socinfo();
+	if (!socinfo)
+		return -1;
+
+	/* Write SoC info */
+	return socinfo_to_pack(socinfo);
+}
diff --git a/board/amlogic/odroid-go-ultra/odroid-go-ultra.c b/board/amlogic/odroid-go-ultra/odroid-go-ultra.c
index f941207173..b769517385 100644
--- a/board/amlogic/odroid-go-ultra/odroid-go-ultra.c
+++ b/board/amlogic/odroid-go-ultra/odroid-go-ultra.c
@@ -6,6 +6,8 @@
 #include <log.h>
 #include <asm/arch/boot.h>
 #include <power/regulator.h>
+#include <env.h>
+#include <asm/arch/boot.h>
 
 int mmc_get_env_dev(void)
 {
@@ -18,3 +20,13 @@ int board_init(void)
 {
 	return 0;
 }
+
+int misc_init_r(void)
+{
+	if (meson_get_soc_pack() == 64) {
+		env_set("fdtfile", "meson-g12b-odroid-go-ultra.dtb");
+	} else {
+		env_set("fdtfile", "meson-g12b-powkiddy-rgb10-max-3-pro.dtb");
+	}
+	return 0;
+}
diff --git a/configs/odroid-go-ultra_defconfig b/configs/odroid-go-ultra_defconfig
index 4bbda284ef..3698987bc0 100644
--- a/configs/odroid-go-ultra_defconfig
+++ b/configs/odroid-go-ultra_defconfig
@@ -23,6 +23,7 @@ CONFIG_FIT_VERBOSE=y
 CONFIG_LEGACY_IMAGE_FORMAT=y
 CONFIG_OF_BOARD_SETUP=y
 # CONFIG_DISPLAY_CPUINFO is not set
+CONFIG_MISC_INIT_R=y
 CONFIG_SYS_MAXARGS=32
 # CONFIG_CMD_BDI is not set
 # CONFIG_CMD_IMI is not set
