--- a/src/shared/machine-id-setup.c
+++ b/src/shared/machine-id-setup.c
@@ -185,7 +185,7 @@
                                 /* If that didn't work, convert it into a readable file */
                                 fd = xopenat_full(inode_fd, /* path= */ NULL, O_RDONLY|O_CLOEXEC, XO_REGULAR, MODE_INVALID);
                                 if (fd < 0)
-                                        return log_error_errno(fd, "Cannot open '%s' in neither writable nor read-only mode: %m", etc_machine_id);
+                                        return 0;

                                 log_debug("Successfully opened existing '%s' file in read-only mode.", etc_machine_id);
                                 writable = false;
