--- grim-1.4.1/write_ppm.c.orig	2024-12-20 20:21:19.335943763 +0000
+++ grim-1.4.1/write_ppm.c	2024-12-20 20:25:18.726959907 +0000
@@ -30,7 +30,10 @@
 	buffer += header_len;
 
 	pixman_format_code_t format = pixman_image_get_format(image);
-	assert(format == PIXMAN_a8r8g8b8 || format == PIXMAN_x8r8g8b8);
+	if ((format != PIXMAN_a8r8g8b8) && (format != PIXMAN_x8r8g8b8)) {
+		fprintf(stderr, "Unsupported format %x\n", format);
+		return -1;
+	}
 
 	// Both formats are native-endian 32-bit ints
 	uint32_t *pixels = pixman_image_get_data(image);
