diff --git a/data/vulkan/VkLayer_window_system_integration.json.in b/data/vulkan/VkLayer_window_system_integration.json.in
deleted file mode 100755
index a89127960595..000000000000
--- a/data/vulkan/VkLayer_window_system_integration.json.in
+++ /dev/null
@@ -1,62 +0,0 @@
-{
-    "file_format_version": "1.1.2",
-    "layer": {
-        "name": "VK_LAYER_window_system_integration",
-        "type": "GLOBAL",
-        "library_path": "@LIB@",
-        "api_version": "1.4.299",
-        "implementation_version": "1",
-        "description": "Window system integration layer",
-        "functions": {
-            "vkNegotiateLoaderLayerInterfaceVersion": "wsi_layer_vkNegotiateLoaderLayerInterfaceVersion"
-        },
-        "instance_extensions": [
-            {"name" : "VK_KHR_wayland_surface", "spec_version" : "6"},
-            {"name" : "VK_KHR_xcb_surface", "spec_version" : "6"},
-            {"name" : "VK_KHR_xlib_surface", "spec_version" : "6"},
-            {"name" : "VK_KHR_surface", "spec_version" : "25"},
-            {"name" : "VK_KHR_get_surface_capabilities2", "spec_version" : "1"},
-            {"name" : "VK_EXT_surface_maintenance1", "spec_version" : "1"}
-        ],
-        "device_extensions": [
-            {"name": "VK_EXT_image_compression_control_swapchain", "spec_version": "1"},
-            {
-                "name": "VK_KHR_swapchain",
-                "spec_version": "70",
-                "entrypoints": [
-                    "vkAcquireNextImageKHR",
-                    "vkCreateSwapchainKHR",
-                    "vkDestroySwapchainKHR",
-                    "vkGetSwapchainImagesKHR",
-                    "vkQueuePresentKHR",
-                    "vkAcquireNextImage2KHR",
-                    "vkGetDeviceGroupPresentCapabilitiesKHR",
-                    "vkGetDeviceGroupSurfacePresentModesKHR",
-                    "vkGetPhysicalDevicePresentRectanglesKHR"
-                ]
-            },
-            {"name": "VK_KHR_present_id", "spec_version": "1"},
-            {
-                "name": "VK_EXT_swapchain_maintenance1",
-                "spec_version": "1",
-                "entrypoints": [
-                    "vkReleaseSwapchainImagesEXT"
-                ]
-            },
-            {
-                "name": "VK_KHR_present_wait",
-                "spec_version": "1",
-                "entrypoints": [
-                    "vkWaitForPresentKHR"
-                ]
-            },
-            {
-                "name": "VK_EXT_present_mode_fifo_latest_ready",
-                "spec_version": "1"
-            }
-        ],
-        "disable_environment": {
-            "DISABLE_WSI_LAYER": "1"
-        }
-    }
-}
diff --git a/data/vulkan/mali.json.in b/data/vulkan/mali.json.in
index 11e8941bc9c8..ca4d984e9177 100644
--- a/data/vulkan/mali.json.in
+++ b/data/vulkan/mali.json.in
@@ -1,7 +1,7 @@
 {
   "file_format_version" : "1.0.0",
   "ICD" : {
-    "library_path" : "@LIB@",
+    "library_path" : "libMaliVulkan.so.1",
     "api_version" : "1.3.276"
   }
 }
diff --git a/meson.build b/meson.build
index 451e48528c09..83ea2084fa88 100644
--- a/meson.build
+++ b/meson.build
@@ -252,25 +252,6 @@ foreach name, values : map
       install : true)
   endif
 
-  if is_vulkan_icd
-    # Install Vulkan WSI config
-    custom_target(
-      'Vulkan manifest layer file',
-      input : 'data/vulkan/VkLayer_window_system_integration.json.in',
-      output : 'VkLayer_window_system_integration.json',
-      command : ['sed', 's/@LIB@/libVkLayer_window_system_integration.so/', '@INPUT@'],
-      capture : true,
-      install_dir : get_option('datadir') / 'vulkan' / 'implicit_layer.d',
-      install : true)
-
-    # Install Vulkan WSI library
-    lib_source = join_paths('data/vulkan/lib', arch, 'libVkLayer_window_system_integration.so')
-    install_data(
-        lib_source,
-        install_dir: get_option('libdir')
-    )
-  endif
-
   # No {headers, pkgconfig} for OpenCL ICD and vendor packages
   if is_opencl_icd or vendor_package
     continue
