--- a/gptcl.cc
+++ b/gptcl.cc
@@ -155,7 +155,8 @@
    } // while
 
    // Assume first non-option argument is the device filename....
-   device = (char*) poptGetArg(poptCon);
+   const char *deviceArg = poptGetArg(poptCon);
+   device = deviceArg ? strdup(deviceArg) : NULL;
    poptResetContext(poptCon);
 
    if (device != NULL) {
@@ -499,6 +500,7 @@
          retval = 4;
       } // if
    } // if (device != NULL)
+   free(device);
    poptFreeContext(poptCon);
    return retval;
 } // GPTDataCL::DoOptions()
