From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ROCKNIX <rocknix@rocknix.org>
Date: Fri, 14 Mar 2026 00:00:00 +0000
Subject: [PATCH] arm64: dts: rockchip: dts rk3566 powkiddy rk2023 enable dmc devfreq

Signed-off-by: ROCKNIX <rocknix@rocknix.org>
---
--- a/arch/arm64/boot/dts/rockchip/rk3566-powkiddy-rk2023.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3566-powkiddy-rk2023.dtsi
@@ -931,3 +931,8 @@
 		remote-endpoint = <&dsi0_in_vp1>;
 	};
 };
+
+&dmc {
+	center-supply = <&vdd_logic>;
+	status = "okay";
+};
-- 
2.34.1

