From d09878c7dbd638a3c41640214acc6dd490a6617e Mon Sep 17 00:00:00 2001
From: Philippe Simons <simons.philippe@gmail.com>
Date: Fri, 24 Jan 2025 01:15:59 +0100
Subject: [PATCH] fix for kernel 6.13

---
 rocknix-singleadc-joypad.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/rocknix-singleadc-joypad.c b/rocknix-singleadc-joypad.c
index e6b3b29..f55d60d 100644
--- a/rocknix-singleadc-joypad.c
+++ b/rocknix-singleadc-joypad.c
@@ -1912,7 +1912,11 @@ MODULE_DEVICE_TABLE(of, joypad_of_match);
 /*----------------------------------------------------------------------------*/
 static struct platform_driver joypad_driver = {
 	.probe = joypad_probe,
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 13, 0))
 	.remove_new = joypad_remove,
+#else
+	.remove = joypad_remove,
+#endif
 	.driver = {
 		.name = DRV_NAME,
 		.pm = &joypad_pm_ops,
-- 
2.47.1

