From 9f5ab5bf2e81031f9046a5cc5d513283d8d4fcd0 Mon Sep 17 00:00:00 2001
From: John Williams <porschemad911@gmail.com>
Date: Tue, 12 May 2026 10:02:07 +1000
Subject: [PATCH] SM6115 GPU

---
 src/gpu_fdinfo.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/gpu_fdinfo.cpp b/src/gpu_fdinfo.cpp
index c4d63f5..6aae9fe 100644
--- a/src/gpu_fdinfo.cpp
+++ b/src/gpu_fdinfo.cpp
@@ -192,7 +192,7 @@ void GPU_fdinfo::find_hwmon_sensors()
     std::string hwmon;
 
     if (module == "msm_drm" || module == "msm_dpu" )
-        hwmon = find_hwmon_sensor_dir("gpu");
+        hwmon = find_hwmon_sensor_dir("gpu_thermal");
     else if (module == "panfrost" || module == "panthor")
         hwmon = find_hwmon_sensor_dir("gpu_thermal");
     else
@@ -655,7 +655,7 @@ float GPU_fdinfo::amdgpu_helper_get_proc_vram() {
 }
 
 void GPU_fdinfo::init_kgsl() {
-    const std::string sys_path = "/sys/class/kgsl/kgsl-3d0";
+    const std::string sys_path = "/sys/class/devfreq/5900000.gpu";
 
     try {
         if (!fs::exists(sys_path)) {
-- 
2.47.3

