diff --git a/meson.build b/meson.build
index b9fc4cacb6..a931a0af14 100644
--- a/meson.build
+++ b/meson.build
@@ -2326,8 +2326,8 @@ elif host_os == 'linux'
 endif
 
 if vulkan.found()
-  libglslang = dependency('glslang', version: '>=15.0.0', required: false)
-  if not libglslang.found()
+  #libglslang = dependency('glslang', version: '>=15.0.0', required: false)
+  #if not libglslang.found()
     # FIXME: Get spirv-tools to enable opt.
     glslang_opts = cmake.subproject_options()
     glslang_opts.add_cmake_defines({'ENABLE_OPT': false, 'ENABLE_HLSL': false})
@@ -2339,7 +2339,7 @@ if vulkan.found()
           glslang_subpro.target('SPIRV'),
         ], include_directories: glslang_subpro.include_directories('glslang')
       )
-  endif
+  #endif
 
   volk_opts = cmake.subproject_options()
   volk_opts.add_cmake_defines({'VOLK_STATIC_DEFINES': 'VK_NO_PROTOTYPES'})
