diff --git a/xemu-version.c b/xemu-version.c
index f2e7a958e9..8674ca7f48 100644
--- a/xemu-version.c
+++ b/xemu-version.c
@@ -1,9 +1,9 @@
 #include "xemu-version-macro.h"
 
-const int xemu_version_major = XEMU_VERSION_MAJOR;
-const int xemu_version_minor = XEMU_VERSION_MINOR;
-const int xemu_version_patch = XEMU_VERSION_PATCH;
-const char *xemu_version = XEMU_VERSION;
-const char *xemu_branch  = XEMU_BRANCH;;
-const char *xemu_commit  = XEMU_COMMIT;
+const int xemu_version_major = 0;
+const int xemu_version_minor = 8;
+const int xemu_version_patch = 115;
+const char *xemu_version = "0.8.115";
+const char *xemu_branch  = "master";
+const char *xemu_commit  = "e6e828362e2b2e2d29c674fc104d5d3086a76f49";
 const char *xemu_date    = XEMU_DATE;