diff --git a/config_spec.yml b/config_spec.yml
index 3e36d1e5c6..bdcdcf4462 100644
--- a/config_spec.yml
+++ b/config_spec.yml
@@ -272,7 +272,7 @@ display:
       default: true
     fit:
       type: enum
-      values: [center, scale, stretch]
+      values: [center, integer, scale, stretch]
       default: scale
     aspect_ratio:
       type: enum
diff --git a/ui/xui/gl-helpers.cc b/ui/xui/gl-helpers.cc
index cdef85861f..c9ef938c12 100644
--- a/ui/xui/gl-helpers.cc
+++ b/ui/xui/gl-helpers.cc
@@ -884,7 +884,7 @@ void ScaleDimensions(int src_width, int src_height, int max_width, int max_heigh
     }
 }
 
-void RenderFramebuffer(GLint tex, int width, int height, bool flip, float scale[2])
+void RenderFramebuffer(GLint tex, int offset_x, int offset_y, int width, int height, bool flip, float scale[2])
 {
     glActiveTexture(GL_TEXTURE0);
     glBindTexture(GL_TEXTURE_2D, tex);
@@ -902,7 +902,7 @@ void RenderFramebuffer(GLint tex, int width, int height, bool flip, float scale[
 
     DecalShader *s = g_framebuffer_shader;
     s->flip = flip;
-    glViewport(0, 0, width, height);
+    glViewport(offset_x, offset_y, width, height);
     glUseProgram(s->prog);
     glBindVertexArray(s->vao);
     glUniform1i(s->flipy_loc, s->flip);
@@ -944,6 +944,7 @@ void RenderFramebuffer(GLint tex, int width, int height, bool flip)
 {
     int tw, th;
     float scale[2];
+    int offset_x = 0, offset_y = 0;
 
     glActiveTexture(GL_TEXTURE0);
     glBindTexture(GL_TEXTURE_2D, tex);
@@ -958,8 +959,29 @@ void RenderFramebuffer(GLint tex, int width, int height, bool flip)
     } else if (g_config.display.ui.fit == CONFIG_DISPLAY_UI_FIT_CENTER) {
         // Centered
         float t_ratio = GetDisplayAspectRatio(tw, th);
-        scale[0] = t_ratio*(float)th/(float)width;
-        scale[1] = (float)th/(float)height;
+        int fb_height = th;
+        int fb_width = (int)(fb_height * t_ratio);
+        offset_x = (width - fb_width) / 2;
+        offset_y = (height - fb_height) / 2;
+        width = fb_width;
+        height = fb_height;
+        scale[0] = 1.0;
+        scale[1] = 1.0;
+    } else if (g_config.display.ui.fit == CONFIG_DISPLAY_UI_FIT_INTEGER) {
+        // Integer scaling
+        float t_ratio = GetDisplayAspectRatio(tw, th);
+        int factor = width / (th * t_ratio);
+        int factor_h = height / th;
+        factor = (factor_h < factor) ? factor_h : factor;
+        factor = (factor <= 1) ? 1 : factor;
+        int fb_height = th * factor;
+        int fb_width = (int)(fb_height * t_ratio);
+        offset_x = (width - fb_width) / 2;
+        offset_y = (height - fb_height) / 2;
+        width = fb_width;
+        height = fb_height;
+        scale[0] = 1.0;
+        scale[1] = 1.0;
     } else {
         float t_ratio = GetDisplayAspectRatio(tw, th);
         float w_ratio = (float)width/(float)height;
@@ -972,12 +994,13 @@ void RenderFramebuffer(GLint tex, int width, int height, bool flip)
         }
     }
 
-    RenderFramebuffer(tex, width, height, flip, scale);
+    RenderFramebuffer(tex, offset_x, offset_y, width, height, flip, scale);
 }
 
 bool RenderFramebufferToPng(GLuint tex, bool flip, std::vector<uint8_t> &png, int max_width, int max_height)
 {
     int width, height;
+    int offset_x = 0, offset_y = 0;
 
     glActiveTexture(GL_TEXTURE0);
     glBindTexture(GL_TEXTURE_2D, tex);
@@ -998,7 +1021,7 @@ bool RenderFramebufferToPng(GLuint tex, bool flip, std::vector<uint8_t> &png, in
     bool blend = glIsEnabled(GL_BLEND);
     if (blend) glDisable(GL_BLEND);
     float scale[2] = {1.0, 1.0};
-    RenderFramebuffer(tex, width, height, !flip, scale);
+    RenderFramebuffer(tex, offset_x, offset_y, width, height, !flip, scale);
     if (blend) glEnable(GL_BLEND);
     glPixelStorei(GL_PACK_ROW_LENGTH, width);
     glPixelStorei(GL_PACK_IMAGE_HEIGHT, height);
diff --git a/ui/xui/gl-helpers.hh b/ui/xui/gl-helpers.hh
index 82da963e6b..f9e94bff25 100644
--- a/ui/xui/gl-helpers.hh
+++ b/ui/xui/gl-helpers.hh
@@ -50,7 +50,7 @@ void RenderControllerPort(float frame_x, float frame_y, int i,
 void RenderXmu(float frame_x, float frame_y, uint32_t primary_color,
                uint32_t secondary_color);
 void RenderFramebuffer(GLint tex, int width, int height, bool flip);
-void RenderFramebuffer(GLint tex, int width, int height, bool flip, float scale[2]);
+void RenderFramebuffer(GLint tex, int offset_x, int offset_y, int width, int height, bool flip, float scale[2]);
 bool RenderFramebufferToPng(GLuint tex, bool flip, std::vector<uint8_t> &png, int max_width = 0, int max_height = 0);
 void SaveScreenshot(GLuint tex, bool flip);
 void ScaleDimensions(int src_width, int src_height, int max_width, int max_height, int *out_width, int *out_height);
diff --git a/ui/xui/main-menu.cc b/ui/xui/main-menu.cc
index af290093ec..40a807a13c 100644
--- a/ui/xui/main-menu.cc
+++ b/ui/xui/main-menu.cc
@@ -822,6 +822,7 @@ void MainMenuDisplayView::Draw()
            "Enable xemu user interface animations");
     ChevronCombo("Display mode", &g_config.display.ui.fit,
                  "Center\0"
+                 "Integer\0"
                  "Scale\0"
                  "Stretch\0",
                  "Select how the framebuffer should fit or scale into the window");
diff --git a/ui/xui/menubar.cc b/ui/xui/menubar.cc
index 799f0da1c7..c29692e1fd 100644
--- a/ui/xui/menubar.cc
+++ b/ui/xui/menubar.cc
@@ -199,7 +199,7 @@ void ShowMainMenu()
             }
 
             ImGui::Combo("Display Mode", &g_config.display.ui.fit,
-                         "Center\0Scale\0Stretch\0");
+                         "Center\0Integer\0Scale\0Stretch\0");
             ImGui::SameLine();
             HelpMarker("Controls how the rendered content should be scaled "
                        "into the window");
diff --git a/ui/xui/popup-menu.cc b/ui/xui/popup-menu.cc
index eee10893c0..348451c1df 100644
--- a/ui/xui/popup-menu.cc
+++ b/ui/xui/popup-menu.cc
@@ -260,7 +260,7 @@ public:
     bool DrawItems(PopupMenuItemDelegate &nav) override
     {
         const char *values[] = {
-            "Center", "Scale", "Stretch"
+            "Center", "Integer", "Scale", "Stretch"
         };
 
         for (int i = 0; i < CONFIG_DISPLAY_UI_FIT__COUNT; i++) {
