diff --git a/xemu-version.c b/xemu-version.c
index f2e7a958e9..8674ca7f48 100644
--- a/xemu-version.c
+++ b/xemu-version.c
@@ -1,9 +1,9 @@
 #include "xemu-version-macro.h"
 
-const int xemu_version_major = XEMU_VERSION_MAJOR;
-const int xemu_version_minor = XEMU_VERSION_MINOR;
-const int xemu_version_patch = XEMU_VERSION_PATCH;
-const char *xemu_version = XEMU_VERSION;
-const char *xemu_branch  = XEMU_BRANCH;;
-const char *xemu_commit  = XEMU_COMMIT;
+const int xemu_version_major = 0;
+const int xemu_version_minor = 8;
+const int xemu_version_patch = 117;
+const char *xemu_version = "$xemu_version_major.$xemu_version_minor.$xemu_version_patch";
+const char *xemu_branch  = "master";
+const char *xemu_commit  = "aa496f863f174fa1bd488dabf7f4f0f2c3646a30";
 const char *xemu_date    = XEMU_DATE;
