diff --git a/src/gpu_fdinfo.cpp b/src/gpu_fdinfo.cpp
index 84b5451..91ddb89 100644
--- a/src/gpu_fdinfo.cpp
+++ b/src/gpu_fdinfo.cpp
@@ -167,7 +167,7 @@ void GPU_fdinfo::find_hwmon_sensors()
     std::string hwmon;
 
     if (module == "msm")
-        hwmon = find_hwmon_sensor_dir("gpu");
+        hwmon = find_hwmon_sensor_dir("gpu_top_thermal");
     else if (module == "panfrost")
         hwmon = find_hwmon_sensor_dir("gpu_thermal");
     else
@@ -621,7 +621,7 @@ float GPU_fdinfo::amdgpu_helper_get_proc_vram() {
 }
 
 void GPU_fdinfo::init_kgsl() {
-    const std::string sys_path = "/sys/class/kgsl/kgsl-3d0";
+    const std::string sys_path = "/sys/class/devfreq/3d00000.gpu";
 
     try {
         if (!fs::exists(sys_path)) {
@@ -633,7 +633,7 @@ void GPU_fdinfo::init_kgsl() {
         return;
     }
 
-    for (std::string metric : {"gpu_busy_percentage", "temp", "clock_mhz" }) {
+    for (const std::string& metric : {"gpu_busy_percentage", "temp1_input", "cur_freq" }) {
         std::string p = sys_path + "/" + metric;
 
         if (!fs::exists(p)) {
@@ -643,10 +643,11 @@ void GPU_fdinfo::init_kgsl() {
 
         SPDLOG_DEBUG("kgsl: {} found", p);
 
-        if (metric == "clock_mhz")
+        if (metric == "cur_freq") {
             gpu_clock_stream.open(p);
-        else
+        } else {
             kgsl_streams[metric].open(p);
+        }
     }
 }
 
