From 60731bb7fd6002649580ab9c63f7f43159ecac10 Mon Sep 17 00:00:00 2001
From: John Williams <porschemad911@gmail.com>
Date: Wed, 14 Jan 2026 21:18:18 +1100
Subject: [PATCH 2/2] SM8250 big core

---
 src/cpu.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/cpu.cpp b/src/cpu.cpp
index f46faba..5174c75 100644
--- a/src/cpu.cpp
+++ b/src/cpu.cpp
@@ -236,6 +236,11 @@ bool CPUStats::UpdateCPUData()
 
     m_cpuPeriod = (double)m_cpuData[0].totalPeriod / m_cpuData.size();
     m_updatedCPUs = true;
+
+    if (m_cpuData.size() > 7) {
+        m_cpuDataTotal.percent = m_cpuData[7].percent;
+    }
+
     return ret;
 }
 
-- 
2.47.3

