From b60118739e12c84f1e6202f54ede6d6a322b7a02 Mon Sep 17 00:00:00 2001
From: John Williams <porschemad911@gmail.com>
Date: Wed, 12 Nov 2025 22:40:57 +1100
Subject: [PATCH] Fix build

---
 meson.build | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/meson.build b/meson.build
index cc6f97f..d1d97f5 100644
--- a/meson.build
+++ b/meson.build
@@ -6,7 +6,7 @@ if get_option('buildtype') == 'debug'
   add_project_arguments('-DDEBUG', language: ['c', 'cpp'])
 endif
 
-libweston_version = 10
+libweston_version = 14
 
 dir_prefix = get_option('prefix')
 
@@ -16,7 +16,7 @@ dir_lib = dir_prefix / get_option('libdir')
 dir_executable_weston = dir_bin
 dir_module_weston = dir_lib / 'weston'
 
-dir_system_include = '/usr/include'
+dir_system_include = dir_lib / 'include'
 
 weston = dependency('libweston-@0@'.format(libweston_version), method: 'pkg-config')
 wayland_server = dependency('wayland-server', method: 'pkg-config')
@@ -53,7 +53,7 @@ shared_library(
   name_prefix: '',
   install: true,
   install_dir: dir_module_weston,
-  include_directories: ['@0@/libweston-@1@'.format(dir_system_include, libweston_version), '@0@/pixman-1'.format(dir_system_include)]
+  #include_directories: ['@0@/libweston-@1@'.format(dir_system_include, libweston_version), '@0@/pixman-1'.format(dir_system_include)]
 )
 
 executable(
-- 
2.47.3

