Stub PostProcessor::Render
--- a/core/rend/gles/postprocess.cpp
+++ b/core/rend/gles/postprocess.cpp
@@ -319,11 +319,11 @@
 {/*
 	if (framebuffer == 0)
 		Init();
-	glBindFramebuffer(GL_FRAMEBUFFER, framebuffer);
+	glBindFramebuffer(GL_FRAMEBUFFER, framebuffer);*/
 }
 
 void PostProcessor::Render(GLuint output_fbo)
-{
+{/*
 	glcache.Disable(GL_SCISSOR_TEST);
 	glcache.Disable(GL_DEPTH_TEST);
 	glcache.Disable(GL_STENCIL_TEST);
