--- a/common/cmd_pxe.c
+++ b/common/cmd_pxe.c
@@ -682,6 +682,8 @@
 			printf("Failed to apply overlay %s, skipping\n",
 				overlayfile);
 			goto skip_overlay;
+		} else {
+			printf("Overlay %s applied successfully\n", overlayfile);
 		}

 skip_overlay:
--- a/common/cmd_fdt.c
+++ b/common/cmd_fdt.c
@@ -644,6 +644,8 @@
                 unsigned long addr;
                 struct fdt_header *blob;
 
+		printf("Applying an overlay\n");
+
                 if (argc != 3)
                         return CMD_RET_USAGE;
 
@@ -655,7 +657,7 @@
                 if (!fdt_valid(&blob))
                         return CMD_RET_FAILURE;
 
-                if (fdt_overlay_apply(working_fdt, blob))
+                if (fdt_overlay_apply_verbose(working_fdt, blob))
                         return CMD_RET_FAILURE;
         }
 #endif
