--- a/Makefiles/Rules.inc
+++ b/Makefiles/Rules.inc
@@ -276,13 +276,13 @@
 # the source code. This technique is described in the reply by user "rr-" at:
 # https://stackoverflow.com/questions/8025766/makefile-auto-dependency-generation
 #
-$(OBJ_DIR)/%.o:	%.cpp
+$(OBJ_DIR)/%.o: %.cpp | $(OBJ_DIR)
 	$(info Generating dependencies: $< -> $(OBJ_DIR)/$(*F).d)
 	$(SILENT)$(CXX) -MM -MP -MT $(OBJ_DIR)/$(*F).o -MT $(OBJ_DIR)/$(*F).d $(CXXFLAGS) $< > $(OBJ_DIR)/$(*F).d
 	$(info Compiling              : $< -> $@)
 	$(SILENT)$(CXX) $(CXXFLAGS) $< -o $@
 
-$(OBJ_DIR)/%.o:	%.c
+$(OBJ_DIR)/%.o: %.c | $(OBJ_DIR)
 	$(info Generating dependencies: $< -> $(OBJ_DIR)/$(*F).d)
 	$(SILENT)$(CC) -MM -MP -MT $(OBJ_DIR)/$(*F).o -MT $(OBJ_DIR)/$(*F).d $(CFLAGS) $< > $(OBJ_DIR)/$(*F).d
 	$(info Compiling              : $< -> $@)
