--- mali-bifrost-r50p0-00eac0/product/kernel/drivers/gpu/arm/midgard/mali_kbase_mem.h.orig	2024-07-04 20:05:04.046675083 +0000
+++ mali-bifrost-r50p0-00eac0/product/kernel/drivers/gpu/arm/midgard/mali_kbase_mem.h	2024-07-04 20:06:12.857657030 +0000
@@ -43,6 +43,7 @@
 #include <linux/version_compat_defs.h>
 #include <linux/sched/mm.h>
 #include <linux/kref.h>
+#include <linux/vmalloc.h>
 
 static inline void kbase_process_page_usage_inc(struct kbase_context *kctx, int pages);
 
