diff --git a/menu/drivers/ozone.c b/menu/drivers/ozone.c
index 23a2d13245..45db0d5bf8 100644
--- a/menu/drivers/ozone.c
+++ b/menu/drivers/ozone.c
@@ -9103,7 +9103,7 @@ static void *ozone_init(void **userdata, bool video_is_threaded)
    ozone->last_width                            = width;
    ozone->last_height                           = height;
    ozone->last_scale_factor                     = gfx_display_get_dpi_scale(p_disp,
-         settings, width, height, false, false);
+         settings, width, height, false, false)  * 2.000f;
    ozone->last_thumbnail_scale_factor           = settings->floats.ozone_thumbnail_scale_factor;
 
    ozone->selection_buf_old.list                = NULL;
@@ -10256,7 +10256,7 @@ static void ozone_render(void *data,
    /* Check whether screen dimensions or menu scale
     * factor have changed */
    scale_factor               = gfx_display_get_dpi_scale(p_disp, settings,
-            width, height, false, false);
+            width, height, false, false) * 2.000f;
    thumbnail_scale_factor     = settings->floats.ozone_thumbnail_scale_factor;
    padding_factor             = settings->floats.ozone_padding_factor;
    font_scale_factor_global   = (font_scale == 1) ? (settings->floats.ozone_font_scale_factor_global) : 1.0f;
