--- a/command.c
+++ b/command.c
@@ -476,6 +476,16 @@
          strlcpy(value_dynamic, "0 0 0", sizeof(value_dynamic));
    }
    #endif
+#ifdef HAVE_MENU
+   else if (memcmp(arg, "menu_active", sizeof("menu_active")) == 0)
+   {
+      struct menu_state *menu_st = menu_state_get_ptr();
+      if (menu_st && (menu_st->flags & MENU_ST_FLAG_ALIVE))
+         value = "true";
+      else
+         value = "false";
+   }
+#endif
    /* TODO: query any string */
 
    _len  = strlcpy(reply, "GET_CONFIG_PARAM ", sizeof(reply));
