diff --git a/CMake/ScmRevGen.cmake b/CMake/ScmRevGen.cmake
index 12ee7727b2..889cc02b52 100644
--- a/CMake/ScmRevGen.cmake
+++ b/CMake/ScmRevGen.cmake
@@ -37,12 +37,12 @@ set(DOLPHIN_VERSION_PATCH ${DOLPHIN_WC_REVISION})
 
 # If Dolphin is not built from a Git repository, default the version info to
 # reasonable values.
-if(NOT DOLPHIN_WC_REVISION)
+#if(NOT DOLPHIN_WC_REVISION)
   set(DOLPHIN_WC_DESCRIBE "${DOLPHIN_VERSION_MAJOR}.${DOLPHIN_VERSION_MINOR}")
   set(DOLPHIN_WC_REVISION "${DOLPHIN_WC_DESCRIBE} (no further info)")
   set(DOLPHIN_WC_BRANCH "master")
   set(DOLPHIN_WC_COMMITS_AHEAD_MASTER 0)
-endif()
+#endif()
 
 # If this is a tag (i.e. a release), then set the current patch version and
 # the number of commits ahead to zero.
