diff --git a/yabause/src/CMakeLists.txt b/yabause/src/CMakeLists.txt
index 0b0e624..7e4037e 100644
--- a/yabause/src/CMakeLists.txt
+++ b/yabause/src/CMakeLists.txt
@@ -194,7 +194,8 @@ if (YAB_USE_SSF)
   find_package(ZLIB)
 
   if (ZLIB_FOUND)
     add_definitions(-DYAB_WANT_SSF=1)
+    set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-error=incompatible-pointer-types -Wno-int-conversion -Wno-implicit-function-declaration -D_LARGEFILE_SOURCE" )
     include_directories(${ZLIB_INCLUDE_DIRS})
     set(YABAUSE_LIBRARIES ${YABAUSE_LIBRARIES} ${ZLIB_LIBRARIES})
 
