diff -ur build.ROCKNIX-RK3566.aarch64/mali-bifrost-r48p0-01eac0.orig/product/kernel/drivers/gpu/arm/midgard/mali_kbase_core_linux.c build.ROCKNIX-RK3566.aarch64/mali-bifrost-r48p0-01eac0/product/kernel/drivers/gpu/arm/midgard/mali_kbase_core_linux.c
--- mali-bifrost-r48p0-01eac0.orig/product/kernel/drivers/gpu/arm/midgard/mali_kbase_core_linux.c	2024-04-10 00:29:25.679588022 +0300
+++ mali-bifrost-r48p0-01eac0/product/kernel/drivers/gpu/arm/midgard/mali_kbase_core_linux.c	2024-04-10 00:33:32.693922486 +0300
@@ -548,9 +548,9 @@
 static int get_irqs(struct kbase_device *kbdev, struct platform_device *pdev)
 {
 	int i;
-	static const char *const irq_names_caps[] = { "JOB", "MMU", "GPU" };
+	static const char *const irq_names[] = { "job", "mmu", "gpu" };
 
-	for (i = 0; i < ARRAY_SIZE(irq_names_caps); i++) {
+	for (i = 0; i < ARRAY_SIZE(irq_names); i++) {
 		struct irq_data *irqdata;
 		int irq;
 
@@ -560,11 +560,11 @@
 		 * first then try using Lower case names. If both attempts fail then
 		 * we assume there is no IRQ resource specified for the GPU.
 		 */
-		irq = platform_get_irq_byname(pdev, irq_names_caps[i]);
+		irq = platform_get_irq_byname(pdev, irq_names[i]);
 		if (irq < 0) {
-			static const char *const irq_names[] = { "job", "mmu", "gpu" };
+			static const char *const irq_names_caps[] = { "JOB", "MMU", "GPU" };
 
-			irq = platform_get_irq_byname(pdev, irq_names[i]);
+			irq = platform_get_irq_byname(pdev, irq_names_caps[i]);
 		}
 
 		if (irq < 0)
