diff -ruN w2a/include/woff2/decode.h w2b/include/woff2/decode.h
--- a/include/woff2/decode.h	2017-11-13 18:31:28.000000000 +0000
+++ b/include/woff2/decode.h	2026-08-30 01:27:42.519540206 +0000
@@ -9,6 +9,8 @@
 #ifndef WOFF2_WOFF2_DEC_H_
 #define WOFF2_WOFF2_DEC_H_
 
+#include <cstdint>
+
 #include <stddef.h>
 #include <inttypes.h>
 #include <woff2/output.h>
diff -ruN w2a/include/woff2/encode.h w2b/include/woff2/encode.h
--- a/include/woff2/encode.h	2017-11-13 18:31:28.000000000 +0000
+++ b/include/woff2/encode.h	2026-08-30 01:27:42.531233920 +0000
@@ -9,6 +9,8 @@
 #ifndef WOFF2_WOFF2_ENC_H_
 #define WOFF2_WOFF2_ENC_H_
 
+#include <cstdint>
+
 #include <stddef.h>
 #include <inttypes.h>
 #include <string>
diff -ruN w2a/include/woff2/output.h w2b/include/woff2/output.h
--- a/include/woff2/output.h	2017-11-13 18:31:28.000000000 +0000
+++ b/include/woff2/output.h	2026-08-30 01:27:42.507971829 +0000
@@ -9,6 +9,8 @@
 #ifndef WOFF2_WOFF2_OUT_H_
 #define WOFF2_WOFF2_OUT_H_
 
+#include <cstdint>
+
 #include <algorithm>
 #include <cstring>
 #include <memory>
diff -ruN w2a/src/font.h w2b/src/font.h
--- a/src/font.h	2017-11-13 18:31:28.000000000 +0000
+++ b/src/font.h	2026-08-30 01:27:42.588846346 +0000
@@ -10,6 +10,8 @@
 #ifndef WOFF2_FONT_H_
 #define WOFF2_FONT_H_
 
+#include <cstdint>
+
 #include <stddef.h>
 #include <inttypes.h>
 #include <map>
diff -ruN w2a/src/glyph.h w2b/src/glyph.h
--- a/src/glyph.h	2017-11-13 18:31:28.000000000 +0000
+++ b/src/glyph.h	2026-08-30 01:27:42.553874368 +0000
@@ -10,6 +10,8 @@
 #ifndef WOFF2_GLYPH_H_
 #define WOFF2_GLYPH_H_
 
+#include <cstdint>
+
 #include <stddef.h>
 #include <inttypes.h>
 #include <vector>
diff -ruN w2a/src/store_bytes.h w2b/src/store_bytes.h
--- a/src/store_bytes.h	2017-11-13 18:31:28.000000000 +0000
+++ b/src/store_bytes.h	2026-08-30 01:27:42.566334881 +0000
@@ -10,6 +10,8 @@
 #ifndef WOFF2_STORE_BYTES_H_
 #define WOFF2_STORE_BYTES_H_
 
+#include <cstdint>
+
 #include <inttypes.h>
 #include <stddef.h>
 #include <string.h>
diff -ruN w2a/src/table_tags.h w2b/src/table_tags.h
--- a/src/table_tags.h	2017-11-13 18:31:28.000000000 +0000
+++ b/src/table_tags.h	2026-08-30 01:27:42.600721681 +0000
@@ -9,6 +9,8 @@
 #ifndef WOFF2_TABLE_TAGS_H_
 #define WOFF2_TABLE_TAGS_H_
 
+#include <cstdint>
+
 #include <inttypes.h>
 
 namespace woff2 {
diff -ruN w2a/src/variable_length.h w2b/src/variable_length.h
--- a/src/variable_length.h	2017-11-13 18:31:28.000000000 +0000
+++ b/src/variable_length.h	2026-08-30 01:27:42.577678486 +0000
@@ -9,6 +9,8 @@
 #ifndef WOFF2_VARIABLE_LENGTH_H_
 #define WOFF2_VARIABLE_LENGTH_H_
 
+#include <cstdint>
+
 #include <inttypes.h>
 #include <vector>
 #include "./buffer.h"
diff -ruN w2a/src/woff2_common.h w2b/src/woff2_common.h
--- a/src/woff2_common.h	2017-11-13 18:31:28.000000000 +0000
+++ b/src/woff2_common.h	2026-08-30 01:27:42.542165860 +0000
@@ -9,6 +9,8 @@
 #ifndef WOFF2_WOFF2_COMMON_H_
 #define WOFF2_WOFF2_COMMON_H_
 
+#include <cstdint>
+
 #include <stddef.h>
 #include <inttypes.h>
 
