--- a/wscript_configure_embedded_heimdal
+++ b/wscript_configure_embedded_heimdal
@@ -4,5 +4,5 @@
 if not conf.env['BISON']:
     conf.fatal("Embedded Heimdal build requires bison but it was not found.  Install bison or use --with-system-mitkrb5 or --with-system-heimdalkrb5")
 
-conf.define('USING_EMBEDDED_HEIMDAL', 1)
+conf.define('USING_EMBEDDED_HEIMDAL', 0)
 conf.RECURSE('third_party/heimdal_build')

--- a/wscript_configure_embedded_heimdal
+++ b/wscript_configure_embedded_heimdal
@@ -17,2 +17,13 @@
 conf.define('HAVE_KRB5_INIT_CREDS_STEP', 1)
 conf.define('HAVE_GSS_KEY_VALUE_SET_DESC', 1)
+
+def check_system_heimdal_binary(name):
+    if conf.LIB_MAY_BE_BUNDLED(name):
+        return False
+    if not conf.find_program(name, var=name.upper()):
+        return False
+    conf.define('USING_SYSTEM_%s' % name.upper(), 1)
+    return True
+
+check_system_heimdal_binary("compile_et")
+check_system_heimdal_binary("asn1_compile")
