From 91f2ec3c86569788031022a620329ca5902af823 Mon Sep 17 00:00:00 2001
From: spycat88 <spycat88@users.noreply.github.com>
Date: Sun, 16 Nov 2025 15:08:34 +0000
Subject: [PATCH] Fix building with gcc15

---
 src/core.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/core.c b/src/core.c
index b14a1fd..aeefaa0 100644
--- a/src/core.c
+++ b/src/core.c
@@ -663,10 +663,10 @@ EventQueue *gmu_core_get_event_queue(void)
 	return &event_queue;
 }
 
-static void set_gmu_running(int bool)
+static void set_gmu_running(int state)
 {
 	pthread_mutex_lock(&gmu_running_mutex);
-	gmu_running = bool;
+	gmu_running = state;
 	pthread_mutex_unlock(&gmu_running_mutex);
 }
 
-- 
2.34.1

