diff --git a/src/TxFilterStub.cpp b/src/TxFilterStub.cpp
index fea57bc..fa33356 100644
--- a/src/TxFilterStub.cpp
+++ b/src/TxFilterStub.cpp
@@ -31,12 +31,25 @@ txfilter_checksum(uint8 *src, int width, int height, int size, int rowStride, ui
 	return 0U;
 }
 
+
+TAPI uint64 TAPIENTRY
+txfilter_checksum_strong(uint8 *src, int width, int height, int size, int rowStride, uint8 *palette)
+{
+    return 0;
+}
+
 TAPI boolean TAPIENTRY
 txfilter_dmptx(uint8 *src, int width, int height, int rowStridePixel, uint16 gfmt, N64FormatSize n64FmtSz, Checksum r_crc64)
 {
 	return 0;
 }
 
+TAPI boolean TAPIENTRY
+txfilter_dmptx_strong(uint8 *src, int width, int height, int rowStridePixel, uint16 gfmt, N64FormatSize n64FmtSz, Checksum r_crc64)
+{
+    return 0;
+}
+
 TAPI boolean TAPIENTRY
 txfilter_reloadhirestex()
 {
