diff -rupbN linux.orig/drivers/gpu/drm/msm/disp/dpu1/dpu_writeback.c linux/drivers/gpu/drm/msm/disp/dpu1/dpu_writeback.c
--- linux.orig/drivers/gpu/drm/msm/disp/dpu1/dpu_writeback.c	2024-11-29 14:02:16.647083955 +0000
+++ linux/drivers/gpu/drm/msm/disp/dpu1/dpu_writeback.c	2024-11-29 14:45:48.714009101 +0000
@@ -42,9 +42,6 @@ static int dpu_wb_conn_atomic_check(stru
 	if (!conn_state || !conn_state->connector) {
 		DPU_ERROR("invalid connector state\n");
 		return -EINVAL;
-	} else if (conn_state->connector->status != connector_status_connected) {
-		DPU_ERROR("connector not connected %d\n", conn_state->connector->status);
-		return -EINVAL;
 	}
 
 	crtc = conn_state->crtc;
