diff -rupN linux.orig/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi linux/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi
--- linux.orig/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi	2024-04-12 20:11:08.857321667 +0000
+++ linux/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi	2024-05-01 18:44:49.576985506 +0000
@@ -9,30 +9,35 @@
 		opp-shared;
 
 		opp00 {
-			opp-hz = /bits/ 64 <408000000>;
+			opp-hz = /bits/ 64 <600000000>;
 			opp-microvolt = <825000 825000 1250000>;
 			clock-latency-ns = <40000>;
 		};
 		opp01 {
-			opp-hz = /bits/ 64 <600000000>;
-			opp-microvolt = <825000 825000 1250000>;
-		};
-		opp02 {
 			opp-hz = /bits/ 64 <816000000>;
 			opp-microvolt = <850000 850000 1250000>;
 		};
-		opp03 {
+		opp02 {
 			opp-hz = /bits/ 64 <1008000000>;
 			opp-microvolt = <925000 925000 1250000>;
 		};
-		opp04 {
+		opp03 {
 			opp-hz = /bits/ 64 <1200000000>;
 			opp-microvolt = <1000000 1000000 1250000>;
 		};
-		opp05 {
+		opp04 {
 			opp-hz = /bits/ 64 <1416000000>;
 			opp-microvolt = <1125000 1125000 1250000>;
 		};
+                opp05 {
+                        opp-hz = /bits/ 64 <1608000000>;
+                        opp-microvolt = <1250000 1275000 1300000>;
+                };
+                opp06 {
+                        opp-hz = /bits/ 64 <1704000000>;
+                        opp-microvolt = <1300000 1300000 1300000>;
+                        turbo-mode;
+                };
 	};
 
 	cluster1_opp: opp-table-1 {
@@ -40,38 +45,43 @@
 		opp-shared;
 
 		opp00 {
-			opp-hz = /bits/ 64 <408000000>;
-			opp-microvolt = <825000 825000 1250000>;
-			clock-latency-ns = <40000>;
-		};
-		opp01 {
 			opp-hz = /bits/ 64 <600000000>;
 			opp-microvolt = <825000 825000 1250000>;
+                        clock-latency-ns = <40000>;
 		};
-		opp02 {
+		opp01 {
 			opp-hz = /bits/ 64 <816000000>;
 			opp-microvolt = <825000 825000 1250000>;
 		};
-		opp03 {
+		opp02 {
 			opp-hz = /bits/ 64 <1008000000>;
 			opp-microvolt = <875000 875000 1250000>;
 		};
-		opp04 {
+		opp03 {
 			opp-hz = /bits/ 64 <1200000000>;
 			opp-microvolt = <950000 950000 1250000>;
 		};
-		opp05 {
+		opp04 {
 			opp-hz = /bits/ 64 <1416000000>;
 			opp-microvolt = <1025000 1025000 1250000>;
 		};
-		opp06 {
+		opp05 {
 			opp-hz = /bits/ 64 <1608000000>;
 			opp-microvolt = <1100000 1100000 1250000>;
 		};
-		opp07 {
+		opp06 {
 			opp-hz = /bits/ 64 <1800000000>;
 			opp-microvolt = <1200000 1200000 1250000>;
 		};
+                opp07 {
+                        opp-hz = /bits/ 64 <2088000000>;
+                        opp-microvolt = <1275000 1300000 1300000>;
+                };
+                opp08 {
+                        opp-hz = /bits/ 64 <2208000000>;
+                        opp-microvolt = <1300000 1325000 1350000>;
+                        turbo-mode;
+                };
 	};
 
 	gpu_opp_table: opp-table-2 {
@@ -82,25 +92,30 @@
 			opp-microvolt = <825000 825000 1150000>;
 		};
 		opp01 {
-			opp-hz = /bits/ 64 <297000000>;
-			opp-microvolt = <825000 825000 1150000>;
-		};
-		opp02 {
-			opp-hz = /bits/ 64 <400000000>;
-			opp-microvolt = <825000 825000 1150000>;
-		};
-		opp03 {
-			opp-hz = /bits/ 64 <500000000>;
-			opp-microvolt = <875000 875000 1150000>;
-		};
-		opp04 {
 			opp-hz = /bits/ 64 <600000000>;
 			opp-microvolt = <925000 925000 1150000>;
 		};
-		opp05 {
+		opp02 {
 			opp-hz = /bits/ 64 <800000000>;
 			opp-microvolt = <1100000 1100000 1150000>;
 		};
+                opp03 {
+                        opp-hz = /bits/ 64 <900000000>;
+                        opp-microvolt = <1150000 1150000 1150000>;
+                };
+	};
+
+	dmc_opp_table: opp-table-3 {
+		compatible = "operating-points-v2";
+
+		opp00 {
+			opp-hz = /bits/ 64 <666000000>;
+			opp-microvolt = <900000>;
+		};
+		opp01 {
+			opp-hz = /bits/ 64 <856000000>;
+			opp-microvolt = <900000>;
+		};
 	};
 };
 
