diff -rupN linux.orig/arch/arm64/boot/dts/rockchip/rk3399.dtsi linux/arch/arm64/boot/dts/rockchip/rk3399.dtsi
--- linux.orig/arch/arm64/boot/dts/rockchip/rk3399.dtsi	2024-04-12 20:11:08.857321667 +0000
+++ linux/arch/arm64/boot/dts/rockchip/rk3399.dtsi	2024-04-29 22:14:18.655706833 +0000
@@ -2110,13 +2110,15 @@
 	};
 
 	gpu: gpu@ff9a0000 {
-		compatible = "rockchip,rk3399-mali", "arm,mali-t860";
+		compatible = "rockchip,rk3399-mali", "arm,mali-t860", "arm,mali-midgard";
 		reg = <0x0 0xff9a0000 0x0 0x10000>;
 		interrupts = <GIC_SPI 20 IRQ_TYPE_LEVEL_HIGH 0>,
 			     <GIC_SPI 21 IRQ_TYPE_LEVEL_HIGH 0>,
 			     <GIC_SPI 19 IRQ_TYPE_LEVEL_HIGH 0>;
 		interrupt-names = "job", "mmu", "gpu";
 		clocks = <&cru ACLK_GPU>;
+		clock-names = "bus";
+		resets = <&cru SRST_A_GPU>;
 		#cooling-cells = <2>;
 		dynamic-power-coefficient = <2640>;
 		power-domains = <&power RK3399_PD_GPU>;
