diff --git a/src/common/common_paths.h b/src/common/common_paths.h
index f12eedb..6f37046 100644
--- a/src/common/common_paths.h
+++ b/src/common/common_paths.h
@@ -29,7 +29,7 @@
 // For compatibility with XDG paths.
 #define EMU_DATA_DIR "citra-emu"
 #else
-#define EMU_DATA_DIR "citra-emu"
+#define EMU_DATA_DIR "lime-emu"
 #endif
 #endif
 
@@ -55,7 +55,7 @@
 
 // Filenames
 // Files in the directory returned by GetUserPath(UserPath::LogDir)
-#define LOG_FILE "citra_log.txt"
+#define LOG_FILE "lime_log.txt"
 
 // Files in the directory returned by GetUserPath(UserPath::ConfigDir)
 #define EMU_CONFIG "emu.ini"
