diff --git a/CMakeLists.txt b/CMakeLists.txt
index 263d19c8..d809c727 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -61,7 +61,7 @@ set_property(GLOBAL PROPERTY PREDEFINED_TARGETS_FOLDER "CMakeTargets")
 # ----------------------------------------------------------------------------
 # enable test in this project
 # ----------------------------------------------------------------------------
-option(BUILD_TEST "enable test binary building)" ON)
+option(BUILD_TEST "enable test binary building)" OFF)
 # ----------------------------------------------------------------------------
 # export json compile commands
 # ----------------------------------------------------------------------------
@@ -332,7 +332,7 @@ add_subdirectory(osal)
 # ----------------------------------------------------------------------------
 #  utils for test case
 # ----------------------------------------------------------------------------
-add_subdirectory(utils)
+#add_subdirectory(utils)
 
 # ----------------------------------------------------------------------------
 #  Media Process Platform library
@@ -344,7 +344,7 @@ add_subdirectory(mpp)
 # ----------------------------------------------------------------------------
 #  test / demo
 # ----------------------------------------------------------------------------
-add_subdirectory(test)
+#add_subdirectory(test)
 
 # ----------------------------------------------------------------------------
 #  install headers
diff --git a/mpp/vproc/iep/test/CMakeLists.txt b/mpp/vproc/iep/test/CMakeLists.txt
index 5f3069f5..40233f8e 100644
--- a/mpp/vproc/iep/test/CMakeLists.txt
+++ b/mpp/vproc/iep/test/CMakeLists.txt
@@ -3,8 +3,8 @@
 # mpp/vproc/iep built-in unit test case
 # ----------------------------------------------------------------------------
 # iep unit test
-option(IEP_TEST "Build base iep unit test" ${BUILD_TEST})
-add_executable(iep_test iep_test.cpp)
-target_link_libraries(iep_test ${MPP_SHARED} utils)
-set_target_properties(iep_test PROPERTIES FOLDER "mpp/vproc/iep")
-add_test(NAME iep_test COMMAND iep_test)
+#option(IEP_TEST "Build base iep unit test" ${BUILD_TEST})
+#add_executable(iep_test iep_test.cpp)
+#target_link_libraries(iep_test ${MPP_SHARED} utils)
+#set_target_properties(iep_test PROPERTIES FOLDER "mpp/vproc/iep")
+#add_test(NAME iep_test COMMAND iep_test)
diff --git a/mpp/vproc/iep2/test/CMakeLists.txt b/mpp/vproc/iep2/test/CMakeLists.txt
index e54298bc..15df89fb 100644
--- a/mpp/vproc/iep2/test/CMakeLists.txt
+++ b/mpp/vproc/iep2/test/CMakeLists.txt
@@ -3,8 +3,8 @@
 # mpp/vproc/iep2 built-in unit test case
 # ----------------------------------------------------------------------------
 # iep2 unit test
-option(IEP2_TEST "Build base iep2 unit test" ON)
-add_executable(iep2_test iep2_test.c)
-target_link_libraries(iep2_test ${MPP_SHARED} utils)
-set_target_properties(iep2_test PROPERTIES FOLDER "mpp/vproc/iep2")
-add_test(NAME iep2_test COMMAND iep2_test)
+#option(IEP2_TEST "Build base iep2 unit test" ON)
+#add_executable(iep2_test iep2_test.c)
+#target_link_libraries(iep2_test ${MPP_SHARED} utils)
+#set_target_properties(iep2_test PROPERTIES FOLDER "mpp/vproc/iep2")
+#add_test(NAME iep2_test COMMAND iep2_test)
diff --git a/mpp/vproc/rga/test/CMakeLists.txt b/mpp/vproc/rga/test/CMakeLists.txt
index af0ee436..f7bb5988 100644
--- a/mpp/vproc/rga/test/CMakeLists.txt
+++ b/mpp/vproc/rga/test/CMakeLists.txt
@@ -3,8 +3,8 @@
 # mpp/vproc/rga built-in unit test case
 # ----------------------------------------------------------------------------
 # rga unit test
-option(RGA_TEST "Build base rga unit test" ${BUILD_TEST})
-add_executable(rga_test rga_test.cpp)
-target_link_libraries(rga_test ${MPP_SHARED} utils)
-set_target_properties(rga_test PROPERTIES FOLDER "mpp/vproc/rga")
-add_test(NAME rga_test COMMAND rga_test)
+#option(RGA_TEST "Build base rga unit test" ${BUILD_TEST})
+#add_executable(rga_test rga_test.cpp)
+#target_link_libraries(rga_test ${MPP_SHARED} utils)
+#set_target_properties(rga_test PROPERTIES FOLDER "mpp/vproc/rga")
+#add_test(NAME rga_test COMMAND rga_test)
diff --git a/mpp/vproc/vdpp/test/CMakeLists.txt b/mpp/vproc/vdpp/test/CMakeLists.txt
index a88935d6..1bc4934d 100644
--- a/mpp/vproc/vdpp/test/CMakeLists.txt
+++ b/mpp/vproc/vdpp/test/CMakeLists.txt
@@ -3,14 +3,14 @@
 # mpp/vproc/vdpp built-in unit test case
 # ----------------------------------------------------------------------------
 # vdpp unit test
-option(VDPP_TEST "Build base vdpp unit test" ON)
-add_executable(vdpp_test vdpp_test.c)
-target_link_libraries(vdpp_test ${MPP_SHARED} utils vproc_vdpp)
-set_target_properties(vdpp_test PROPERTIES FOLDER "mpp/vproc/vdpp")
-add_test(NAME vdpp_test COMMAND vdpp_test)
+#option(VDPP_TEST "Build base vdpp unit test" ON)
+#add_executable(vdpp_test vdpp_test.c)
+#target_link_libraries(vdpp_test ${MPP_SHARED} utils vproc_vdpp)
+#set_target_properties(vdpp_test PROPERTIES FOLDER "mpp/vproc/vdpp")
+#add_test(NAME vdpp_test COMMAND vdpp_test)
 
 # hwpq test (call libvdpp.so)
-add_executable(hwpq_test hwpq_test.cpp)
-target_link_libraries(hwpq_test vdpp)
-set_target_properties(hwpq_test PROPERTIES FOLDER "mpp/vproc/vdpp")
-add_test(NAME hwpq_test COMMAND hwpq_test)
+#add_executable(hwpq_test hwpq_test.cpp)
+#target_link_libraries(hwpq_test vdpp)
+#set_target_properties(hwpq_test PROPERTIES FOLDER "mpp/vproc/vdpp")
+#add_test(NAME hwpq_test COMMAND hwpq_test)
