From 5911b0ea29e64c289cf8582c55d8892687149602 Mon Sep 17 00:00:00 2001
From: John Williams <porschemad911@gmail.com>
Date: Wed, 14 Jan 2026 22:18:16 +1100
Subject: [PATCH 1/3] No static lib

---
 src/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/meson.build b/src/meson.build
index e500c0c..9ca3f5b 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -176,7 +176,7 @@ if is_unixy
   endif
 endif
 
-link_args = cc.get_supported_link_arguments(['-Wl,-Bsymbolic-functions', '-Wl,-z,relro', '-Wl,--exclude-libs,ALL', '-lGL', '-static-libstdc++'])
+link_args = cc.get_supported_link_arguments(['-Wl,-Bsymbolic-functions', '-Wl,-z,relro', '-Wl,--exclude-libs,ALL', '-lGL'])
 # meson fails to check version-script so just force add
 link_args += '-Wl,--version-script,@0@'.format(join_paths(meson.current_source_dir(), 'mangohud.version'))
 
-- 
2.47.3

