From fc30aa5437bee410ada66244f8e2a163eb58a488 Mon Sep 17 00:00:00 2001
From: John Williams <porschemad911@gmail.com>
Date: Sun, 24 May 2026 22:28:30 +1000
Subject: [PATCH 2/3] Battery name

---
 src/battery.cpp | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/src/battery.cpp b/src/battery.cpp
index 1612bff..16b0355 100644
--- a/src/battery.cpp
+++ b/src/battery.cpp
@@ -10,14 +10,16 @@ void BatteryStats::numBattery() {
     if (!fs::exists("/sys/class/power_supply/")) {
          batteryCount = 0;
     }
-    fs::path path("/sys/class/power_supply/");
-    for (auto& p : fs::directory_iterator(path)) {
-        string fileName = p.path().filename();
-        if (fileName.find("BAT") != std::string::npos) {
-            battPath[batteryCount] = p.path();
+    
+    else {
+        string battery = "/sys/class/power_supply/battery";
+        
+        if (fs::exists(battery)) {
+            battPath[batteryCount] = battery;
             batteryCount += 1;
         }
     }
+
     batt_count = batteryCount;
     batt_check = true;
 }
-- 
2.47.3

