From d0ce0d7ed5d7b6122057976627f6761d70f1c59c Mon Sep 17 00:00:00 2001
From: John Williams <porschemad911@gmail.com>
Date: Wed, 14 Jan 2026 22:19:09 +1100
Subject: [PATCH 3/3] RAM name

---
 src/hud_elements.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/hud_elements.cpp b/src/hud_elements.cpp
index 3cac5af..86d5002 100644
--- a/src/hud_elements.cpp
+++ b/src/hud_elements.cpp
@@ -804,7 +804,7 @@ void HudElements::procmem()
         return;
 
     ImguiNextColumnFirstItem();
-    HUDElements.TextColored(HUDElements.colors.ram, "PMEM");
+    HUDElements.TextColored(HUDElements.colors.ram, "RAM");
     ImguiNextColumnOrNewRow();
     right_aligned_text(HUDElements.colors.text, HUDElements.ralign_width, "%.1f", format_units(proc_mem_resident, unit));
     ImGui::SameLine(0, 1.0f);
-- 
2.47.3

