--- avfs-1.1.5/modules/extfs.c.orig	2024-11-23 10:31:14.685469575 +0000
+++ avfs-1.1.5/modules/extfs.c	2024-11-23 10:31:56.944259056 +0000
@@ -464,8 +464,6 @@
     fd = open(extfs_conf, O_RDONLY);
     if(fd == -1) {
         res = -errno;
-        av_log(AVLOG_WARNING, "Could not open extfs config file %s: %s", 
-                 extfs_conf, strerror(errno));
         av_free(extfs_conf);
         av_free(extfs_dir);
         return res;
