Subject: [PATCH] arm64: dts: qcom: sm8750: add GPU thermal cooling

---
--- a/arch/arm64/boot/dts/qcom/sm8750.dtsi
+++ b/arch/arm64/boot/dts/qcom/sm8750.dtsi
@@ -6360,9 +6360,24 @@
 		gpuss0-thermal {
+			polling-delay-passive = <250>;
+
 			thermal-sensors = <&tsens2 1>;
 
+			cooling-maps {
+				gpu0_cdev {
+					trip = <&gpuss0_tj>;
+					cooling-device = <&gpu THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+				};
+			};
+
 			trips {
+				gpuss0_tj: tj-cfg {
+					temperature = <95000>;
+					hysteresis = <5000>;
+					type = "passive";
+				};
+
 				trip-point0 {
 					temperature = <120000>;
 					hysteresis = <5000>;
 					type = "hot";
 				};
@@ -6378,9 +6393,24 @@
 		gpuss1-thermal {
+			polling-delay-passive = <250>;
+
 			thermal-sensors = <&tsens2 2>;
 
+			cooling-maps {
+				gpu1_cdev {
+					trip = <&gpuss1_tj>;
+					cooling-device = <&gpu THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+				};
+			};
+
 			trips {
+				gpuss1_tj: tj-cfg {
+					temperature = <95000>;
+					hysteresis = <5000>;
+					type = "passive";
+				};
+
 				trip-point0 {
 					temperature = <120000>;
 					hysteresis = <5000>;
 					type = "hot";
 				};
@@ -6396,9 +6426,24 @@
 		gpuss2-thermal {
+			polling-delay-passive = <250>;
+
 			thermal-sensors = <&tsens2 3>;
 
+			cooling-maps {
+				gpu2_cdev {
+					trip = <&gpuss2_tj>;
+					cooling-device = <&gpu THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+				};
+			};
+
 			trips {
+				gpuss2_tj: tj-cfg {
+					temperature = <95000>;
+					hysteresis = <5000>;
+					type = "passive";
+				};
+
 				trip-point0 {
 					temperature = <120000>;
 					hysteresis = <5000>;
 					type = "hot";
 				};
@@ -6414,9 +6459,24 @@
 		gpuss3-thermal {
+			polling-delay-passive = <250>;
+
 			thermal-sensors = <&tsens2 4>;
 
+			cooling-maps {
+				gpu3_cdev {
+					trip = <&gpuss3_tj>;
+					cooling-device = <&gpu THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+				};
+			};
+
 			trips {
+				gpuss3_tj: tj-cfg {
+					temperature = <95000>;
+					hysteresis = <5000>;
+					type = "passive";
+				};
+
 				trip-point0 {
 					temperature = <120000>;
 					hysteresis = <5000>;
 					type = "hot";
 				};
@@ -6432,9 +6492,24 @@
 		gpuss4-thermal {
+			polling-delay-passive = <250>;
+
 			thermal-sensors = <&tsens2 5>;
 
+			cooling-maps {
+				gpu4_cdev {
+					trip = <&gpuss4_tj>;
+					cooling-device = <&gpu THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+				};
+			};
+
 			trips {
+				gpuss4_tj: tj-cfg {
+					temperature = <95000>;
+					hysteresis = <5000>;
+					type = "passive";
+				};
+
 				trip-point0 {
 					temperature = <120000>;
 					hysteresis = <5000>;
 					type = "hot";
 				};
@@ -6450,9 +6525,24 @@
 		gpuss5-thermal {
+			polling-delay-passive = <250>;
+
 			thermal-sensors = <&tsens2 6>;
 
+			cooling-maps {
+				gpu5_cdev {
+					trip = <&gpuss5_tj>;
+					cooling-device = <&gpu THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+				};
+			};
+
 			trips {
+				gpuss5_tj: tj-cfg {
+					temperature = <95000>;
+					hysteresis = <5000>;
+					type = "passive";
+				};
+
 				trip-point0 {
 					temperature = <120000>;
 					hysteresis = <5000>;
 					type = "hot";
 				};
@@ -6468,9 +6558,24 @@
 		gpuss6-thermal {
+			polling-delay-passive = <250>;
+
 			thermal-sensors = <&tsens2 7>;
 
+			cooling-maps {
+				gpu6_cdev {
+					trip = <&gpuss6_tj>;
+					cooling-device = <&gpu THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+				};
+			};
+
 			trips {
+				gpuss6_tj: tj-cfg {
+					temperature = <95000>;
+					hysteresis = <5000>;
+					type = "passive";
+				};
+
 				trip-point0 {
 					temperature = <120000>;
 					hysteresis = <5000>;
 					type = "hot";
 				};
@@ -6486,9 +6591,24 @@
 		gpuss7-thermal {
+			polling-delay-passive = <250>;
+
 			thermal-sensors = <&tsens2 8>;
 
+			cooling-maps {
+				gpu7_cdev {
+					trip = <&gpuss7_tj>;
+					cooling-device = <&gpu THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+				};
+			};
+
 			trips {
+				gpuss7_tj: tj-cfg {
+					temperature = <95000>;
+					hysteresis = <5000>;
+					type = "passive";
+				};
+
 				trip-point0 {
 					temperature = <120000>;
 					hysteresis = <5000>;
 					type = "hot";
 				};
