diff --git a/src/shared/h/shared/system.h b/src/shared/h/shared/system.h
index f5f38b34..2d765a07 100644
--- a/src/shared/h/shared/system.h
+++ b/src/shared/h/shared/system.h
@@ -289,7 +289,11 @@ int vasprintf(char **buf, const char *fmt, va_list v);
 //
 // See https://lwn.net/Articles/612244/
 #ifndef HAVE_STRLCPY
+#if __GNUC__ >= 14
+extern "C" size_t strlcpy(char *dest, const char *src, size_t size) noexcept;
+#else
 size_t strlcpy(char *dest, const char *src, size_t size);
+#endif
 #endif // HAVE_STRLCPY
 #endif
 

