From 06e7c06d3be130ad43732fe31e208a1021af61a8 Mon Sep 17 00:00:00 2001
From: Teguh Sobirin <teguh@sobir.in>
Date: Mon, 24 Feb 2025 17:18:33 +0800
Subject: [PATCH 2/4] Don't verbose scsi_scan

Signed-off-by: Teguh Sobirin <teguh@sobir.in>
---
 cmd/scsi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmd/scsi.c b/cmd/scsi.c
index c286bdc072..b8a6f3434e 100644
--- a/cmd/scsi.c
+++ b/cmd/scsi.c
@@ -39,7 +39,7 @@ static int do_scsi(struct cmd_tbl *cmdtp, int flag, int argc,
 			return ret;
 		}
 		if (strncmp(argv[1], "scan", 4) == 0) {
-			ret = scsi_scan(true);
+			ret = scsi_scan(false);
 			if (ret)
 				return CMD_RET_FAILURE;
 			return ret;
-- 
2.43.0

