From ac20d9cbe2f8cfa220960507bd2f0ab7fd018223 Mon Sep 17 00:00:00 2001
From: Teguh Sobirin <teguh@sobir.in>
Date: Mon, 24 Feb 2025 17:18:45 +0800
Subject: [PATCH 3/4] Hide logo since it's not rotated when use
 CONFIG_CONSOLE_ROTATION

Signed-off-by: Teguh Sobirin <teguh@sobir.in>
---
 drivers/video/simplefb.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/video/simplefb.c b/drivers/video/simplefb.c
index 239b006744..89eb391348 100644
--- a/drivers/video/simplefb.c
+++ b/drivers/video/simplefb.c
@@ -36,6 +36,7 @@ static int simple_video_probe(struct udevice *dev)
 	 */
 	plat->base = base;
 	plat->size = size;
+	plat->hide_logo = true;
 
 	video_set_flush_dcache(dev, true);
 
-- 
2.43.0

