--- a/src/pcm/pcm.c	2026-09-23 13:36:00.001935496 -0400
+++ b/src/pcm/pcm.c	2026-09-23 13:36:00.021935545 -0400
@@ -8783,8 +8783,8 @@
 			s = "underrun";
 		else
 			s = "overrun";
-		if (!silent)
-			snd_error(PCM, "%s occurred", s);
+		//if (!silent)
+		//	snd_error(PCM, "%s occurred", s);
 		err = snd_pcm_prepare(pcm);
 		if (err < 0) {
 			snd_error(PCM, "cannot recovery from %s, prepare failed: %s", s, snd_strerror(err));
