diff --git a/arch/arm64/boot/dts/qcom/qcs8550-ayaneo-pocketevo.dts b/arch/arm64/boot/dts/qcom/qcs8550-ayaneo-pocketevo.dts
new file mode 100644
index 000000000..2dd554820
--- /dev/null
+++ b/arch/arm64/boot/dts/qcom/qcs8550-ayaneo-pocketevo.dts
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright (c) 2025, Teguh Sobirin.
+ * Copyright (c) 2025, ROCKNIX (https://github.com/ROCKNIX)
+ */
+
+/dts-v1/;
+
+#include "qcs8550-ayaneo-pocket-common.dtsi"
+
+/ {
+	model = "AYANEO Pocket EVO";
+	qcom,board-id = <0x1000b 0x01>;
+	qcom,msm-id = <0x258 0x20000>, <0x259 0x20000>;
+	compatible = "ayaneo,pocketevo", "qcom,qcs8550", "qcom,sm8550";
+};
+
+&tlmm {
+	panel_pwr_active: panel-pwr-active-state {
+		pins = "gpio152", "gpio153";
+		function = "gpio";
+		drive-strength = <8>;
+		bias-disable;
+		output-high;
+	};
+};
+
+&i2c0 {
+   clock-frequency = <400000>;
+	status = "okay";
+
+};
+
+&display_panel {
+	compatible = "chipone,icna3512";
+
+	vci-supply = <&vreg_l13b_3p0>;
+	vddio-supply = <&vreg_l12b_1p8>;
+
+	reset-gpios = <&tlmm 133 GPIO_ACTIVE_LOW>;
+
+	rotation = <270>;
+};
