diff -rupN rocknix-hotkey.orig/gptokeyb.cpp rocknix-hotkey/gptokeyb.cpp
--- rocknix-hotkey.orig/gptokeyb.cpp	2024-06-27 17:37:46.031358768 +0000
+++ rocknix-hotkey/gptokeyb.cpp	2024-06-27 18:38:19.195976235 +0000
@@ -1609,11 +1609,11 @@ bool handleEvent(const SDL_Event& event)
         // Fake Xbox360 mode
         switch (event.cbutton.button) {
           case SDL_CONTROLLER_BUTTON_A:
-            emitKey(BTN_A, is_pressed);
+            emitKey(BTN_B, is_pressed);
             break;
 
           case SDL_CONTROLLER_BUTTON_B:
-            emitKey(BTN_B, is_pressed);
+            emitKey(BTN_A, is_pressed);
             break;
 
           case SDL_CONTROLLER_BUTTON_X:
