Add the YMZ ADPCM sources to the libretro build
--- a/sdl/Makefile.common
+++ b/sdl/Makefile.common
@@ -104,6 +104,8 @@
 	$(NP2_PATH)/sound/soundrom.c \
 	$(NP2_PATH)/sound/tms3631c.c \
 	$(NP2_PATH)/sound/tms3631g.c \
+	$(NP2_PATH)/sound/ymzadpcmc.c \
+	$(NP2_PATH)/sound/ymzadpcmg.c \
 	$(wildcard $(NP2_PATH)/sound/getsnd/*.c) \
 	$(wildcard $(NP2_PATH)/sound/mame/*.c) \
 	$(wildcard $(NP2_PATH)/sound/vermouth/*.c) \
