diff --git a/src/Cafe/Filesystem/FST/KeyCache.cpp b/src/Cafe/Filesystem/FST/KeyCache.cpp
index 5d8d51c..92678e4 100644
--- a/src/Cafe/Filesystem/FST/KeyCache.cpp
+++ b/src/Cafe/Filesystem/FST/KeyCache.cpp
@@ -59,7 +59,16 @@ void KeyCache_Prepare()
 	sKeyCachePrepared = true;
 	g_keyCache.clear();
 	// load keys
-	auto keysPath = ActiveSettings::GetUserDataPath("keys.txt");
+
+	std::string keysDir = "/storage/roms/bios/cemu";
+	// Let's first make sure the parent directory exists.
+	std::error_code err;
+	fs::create_directories(keysDir, err);
+	if (err) {
+		wxMessageBox("Unable to create /storage/roms/bios/cemu directory, to store the keys.txt file\nThis can happen if Cemu does not have write permission to /storage/roms/bios/ directory, the disk is full or if anti-virus software is blocking Cemu.", "Error", wxOK | wxCENTRE | wxICON_ERROR);
+	}
+
+	auto keysPath = keysDir + "/keys.txt";
 	FileStream* fs_keys = FileStream::openFile2(keysPath);
 	if( !fs_keys )
 	{
