diff --git a/src/meson.build b/src/meson.build
index 5d04334..59af71a 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -65,7 +65,7 @@ if get_option('scanner')
 		version: meson.project_version(),
 		variables: [
 			'datarootdir=' + join_paths('${prefix}', get_option('datadir')),
-			'pkgdatadir=' + join_paths('${pc_sysrootdir}${datarootdir}', meson.project_name()),
+			'pkgdatadir=' + join_paths('${datarootdir}', meson.project_name()),
 			'bindir=' + join_paths('${prefix}', get_option('bindir')),
 			'wayland_scanner=${bindir}/wayland-scanner'
 		],
