diff --git a/hw/xbox/xid.c b/hw/xbox/xid.c
index 86f34f7de2..935c809cfd 100644
--- a/hw/xbox/xid.c
+++ b/hw/xbox/xid.c
@@ -171,10 +171,10 @@ static const XIDDesc desc_xid_xbox_gamepad = {
     .wAlternateProductIds = { 0xFFFF, 0xFFFF, 0xFFFF, 0xFFFF },
 };
 
-#define GAMEPAD_A                0
-#define GAMEPAD_B                1
-#define GAMEPAD_X                2
-#define GAMEPAD_Y                3
+#define GAMEPAD_B                0
+#define GAMEPAD_A                1
+#define GAMEPAD_Y                2
+#define GAMEPAD_X                3
 #define GAMEPAD_BLACK            4
 #define GAMEPAD_WHITE            5
 #define GAMEPAD_LEFT_TRIGGER     6
diff --git a/ui/xui/gl-helpers.cc b/ui/xui/gl-helpers.cc
index 6c07d087b4..448b5d51af 100644
--- a/ui/xui/gl-helpers.cc
+++ b/ui/xui/gl-helpers.cc
@@ -473,10 +473,10 @@ void RenderController(float frame_x, float frame_y, uint32_t primary_color,
     const struct rect lstick_ctr = {  93, 246,   0,   0 };
     const struct rect rstick_ctr = { 342, 148,   0,   0 };
     const struct rect buttons[12] = {
-        { 367, 187, 30, 38 }, // A
         { 368, 229, 30, 38 }, // B
-        { 330, 204, 30, 38 }, // X
+        { 367, 187, 30, 38 }, // A
         { 331, 247, 30, 38 }, // Y
+        { 330, 204, 30, 38 }, // X
         {  82, 121, 31, 47 }, // D-Left
         { 104, 160, 44, 25 }, // D-Up
         { 141, 121, 31, 47 }, // D-Right
diff --git a/ui/xui/input-manager.cc b/ui/xui/input-manager.cc
index 786b54b176..43736e32f8 100644
--- a/ui/xui/input-manager.cc
+++ b/ui/xui/input-manager.cc
@@ -60,10 +60,10 @@ void InputManager::Update()
     const int thumb_dead_zone = 8000;           // SDL_gamecontroller.h suggests using this value.
     MAP_BUTTON(ImGuiKey_GamepadStart,           CONTROLLER_BUTTON_START);
     MAP_BUTTON(ImGuiKey_GamepadBack,            CONTROLLER_BUTTON_BACK);
-    MAP_BUTTON(ImGuiKey_GamepadFaceDown,        CONTROLLER_BUTTON_A);              // Xbox A, PS Cross
-    MAP_BUTTON(ImGuiKey_GamepadFaceRight,       CONTROLLER_BUTTON_B);              // Xbox B, PS Circle
-    MAP_BUTTON(ImGuiKey_GamepadFaceLeft,        CONTROLLER_BUTTON_X);              // Xbox X, PS Square
-    MAP_BUTTON(ImGuiKey_GamepadFaceUp,          CONTROLLER_BUTTON_Y);              // Xbox Y, PS Triangle
+    MAP_BUTTON(ImGuiKey_GamepadFaceDown,        CONTROLLER_BUTTON_B);              // Xbox A, PS Cross
+    MAP_BUTTON(ImGuiKey_GamepadFaceRight,       CONTROLLER_BUTTON_A);              // Xbox B, PS Circle
+    MAP_BUTTON(ImGuiKey_GamepadFaceLeft,        CONTROLLER_BUTTON_Y);              // Xbox X, PS Square
+    MAP_BUTTON(ImGuiKey_GamepadFaceUp,          CONTROLLER_BUTTON_X);              // Xbox Y, PS Triangle
     MAP_BUTTON(ImGuiKey_GamepadDpadLeft,        CONTROLLER_BUTTON_DPAD_LEFT);
     MAP_BUTTON(ImGuiKey_GamepadDpadRight,       CONTROLLER_BUTTON_DPAD_RIGHT);
     MAP_BUTTON(ImGuiKey_GamepadDpadUp,          CONTROLLER_BUTTON_DPAD_UP);
