diff --git a/Core/Config.cpp b/Core/Config.cpp
index 4f6ee43dce..e61e3eda4c 100644
--- a/Core/Config.cpp
+++ b/Core/Config.cpp
@@ -1616,12 +1616,6 @@ void Config::PostLoadCleanup() {
 		g_Config.bSkipBufferEffects = false;
 	}
 
-	// Automatically silence secondary instances. Could be an option I guess, but meh.
-	if (PPSSPP_ID > 1) {
-		NOTICE_LOG(Log::Audio, "Secondary instance %d - silencing audio", (int)PPSSPP_ID);
-		g_Config.iGameVolume = 0;
-	}
-
 	// Automatically switch away from deprecated setting value.
 	if (iTexScalingLevel <= 0) {
 		iTexScalingLevel = 1;
