index a8300543d..37c12caa4 100644
--- a/Core/SaveState.cpp
+++ b/Core/SaveState.cpp
@@ -333,7 +333,7 @@ int g_screenshotFailures;
 	}
 
 	std::string GetGamePrefix(const ParamSFOData &paramSFO) {
-		return GenerateFullDiscId(paramSFO);
+		return PSP_CoreParameter().fileToStart.WithReplacedExtension("").GetFilename();
 	}
 
 	// The prefix is always based on GenerateFullDiscId. So we can find these by scanning, too.
