--- a/product/kernel/include/linux/version_compat_defs.h
+++ b/product/kernel/include/linux/version_compat_defs.h
@@ -418,7 +418,9 @@
 kbase_mm_get_unmapped_area_helper(struct mm_struct *mm, struct file *filp, unsigned long addr,
 				  unsigned long len, unsigned long pgoff, unsigned long flags)
 {
-#if (KERNEL_VERSION(6, 10, 0) <= LINUX_VERSION_CODE)
+#if (KERNEL_VERSION(6, 19, 0) <= LINUX_VERSION_CODE)
+	return mm_get_unmapped_area(filp, addr, len, pgoff, flags);
+#elif (KERNEL_VERSION(6, 10, 0) <= LINUX_VERSION_CODE)
 	return mm_get_unmapped_area(mm, filp, addr, len, pgoff, flags);
 #else
 	return mm->get_unmapped_area(filp, addr, len, pgoff, flags);
