diff --git a/product/kernel/drivers/gpu/arm/midgard/mali_kbase_fence.h b/product/kernel/drivers/gpu/arm/midgard/mali_kbase_fence.h
--- a/product/kernel/drivers/gpu/arm/midgard/mali_kbase_fence.h
+++ b/product/kernel/drivers/gpu/arm/midgard/mali_kbase_fence.h
@@ -176,7 +176,8 @@ static inline int kbase_fence_out_signal(struct kbase_jd_atom *katom, int status
 {
 	if (status)
 		dma_fence_set_error_helper(katom->dma_fence.fence, status);
-	return dma_fence_signal(katom->dma_fence.fence);
+	dma_fence_signal(katom->dma_fence.fence);
+	return 0;
 }
 
 #if IS_ENABLED(CONFIG_SYNC_FILE)
