diff -rupbN pipewire.orig/src/daemon/pipewire-pulse.conf.in pipewire/src/daemon/pipewire-pulse.conf.in
--- pipewire.orig/src/daemon/pipewire-pulse.conf.in	2024-10-08 04:01:35.119453036 +0000
+++ pipewire/src/daemon/pipewire-pulse.conf.in	2024-10-08 04:03:35.898636383 +0000
@@ -27,7 +27,7 @@ context.modules = [
     { name = libpipewire-module-rt
         args = {
             nice.level   = -11
-            #rt.prio      = @rtprio_client@
+            rt.prio      = 99
             #rt.time.soft = -1
             #rt.time.hard = -1
             #uclamp.min = 0
@@ -102,12 +102,12 @@ pulse.properties = {
     ]
     #server.dbus-name       = "org.pulseaudio.Server"
     #pulse.allow-module-loading = true
-    #pulse.min.req          = 128/48000     # 2.7ms
+    pulse.min.req          = 1024/48000     # 2.7ms
     #pulse.default.req      = 960/48000     # 20 milliseconds
-    #pulse.min.frag         = 128/48000     # 2.7ms
+    pulse.min.frag         = 1024/48000     # 2.7ms
     #pulse.default.frag     = 96000/48000   # 2 seconds
     #pulse.default.tlength  = 96000/48000   # 2 seconds
-    #pulse.min.quantum      = 128/48000     # 2.7ms
+    pulse.min.quantum      = 1024/48000     # 2.7ms
     #pulse.idle.timeout     = 0             # don't pause after underruns
     #pulse.default.format   = F32
     #pulse.default.position = [ FL FR ]
