From 6617d9359d54b3858bf7276ed2a6ae41d441fabd Mon Sep 17 00:00:00 2001
From: Philippe Simons <simons.philippe@gmail.com>
Date: Wed, 18 Dec 2024 10:20:18 +0100
Subject: [PATCH] allow amux_count to be 0

---
 rocknix-singleadc-joypad.c | 20 ++++++++++++--------
 1 file changed, 12 insertions(+), 8 deletions(-)

diff --git a/rocknix-singleadc-joypad.c b/rocknix-singleadc-joypad.c
index 2848103..7646899 100644
--- a/rocknix-singleadc-joypad.c
+++ b/rocknix-singleadc-joypad.c
@@ -1149,7 +1149,9 @@ static int joypad_input_setup(struct device *dev, struct joypad *joypad)
 	input->id.version = (u16)joypad_revision;
 
 	/* IIO ADC key setup (0 mv ~ 1800 mv) * adc->scale */
-	__set_bit(EV_ABS, input->evbit);
+	if (joypad->amux_count > 0) {
+		__set_bit(EV_ABS, input->evbit);
+	}
 
 	// Set mapped ones on dt
 	for(nbtn = 0; nbtn < joypad->amux_count; nbtn++) {
@@ -1244,19 +1246,21 @@ static int joypad_dt_parse(struct device *dev, struct joypad *joypad)
 
 	joypad->bt_gpio_count = device_get_child_node_count(dev);
 
-	if ((joypad->amux_count == 0) || (joypad->bt_gpio_count == 0)) {
+	if ((joypad->amux_count == 0) && (joypad->bt_gpio_count == 0)) {
 		dev_err(dev, "adc key = %d, gpio key = %d error!",
 			joypad->amux_count, joypad->bt_gpio_count);
 		return -EINVAL;
 	}
 
-	error = joypad_adc_setup(dev, joypad);
-	if (error)
-		return error;
+	if (joypad->amux_count > 0) {
+		error = joypad_adc_setup(dev, joypad);
+		if (error)
+			return error;
 
-	error = joypad_amux_setup(dev, joypad);
-	if (error)
-		return error;
+		error = joypad_amux_setup(dev, joypad);
+		if (error)
+			return error;
+	}
 
 	error = joypad_gpio_setup(dev, joypad);
 	if (error)
-- 
2.47.1

