--- util-linux-2.39.2/configure.ac.orig	2024-11-23 11:15:21.120872078 +0000
+++ util-linux-2.39.2/configure.ac	2024-11-23 11:17:32.011211100 +0000
@@ -1737,6 +1737,14 @@
 UL_REQUIRES_BUILD([fstrim], [libmount])
 AM_CONDITIONAL([BUILD_FSTRIM], [test "x$build_fstrim" = xyes])
 
+AC_ARG_ENABLE([blkdiscard],
+  AS_HELP_STRING([--disable-blkdiscard], [do not build blkdiscard(8)]),
+  [], [UL_DEFAULT_ENABLE([blkdiscard], [check])]
+)
+UL_BUILD_INIT([blkdiscard])
+UL_REQUIRES_LINUX([blkdiscard])
+AM_CONDITIONAL([BUILD_BLKDISCARD], [test "x$build_blkdiscard" = xyes])
+
 
 AC_ARG_ENABLE([swapon],
   AS_HELP_STRING([--disable-swapon], [do not build swapon(8) and swapoff(8)]),
@@ -2075,10 +2083,6 @@
 UL_REQUIRES_LINUX([fsfreeze])
 AM_CONDITIONAL([BUILD_FSFREEZE], [test "x$build_fsfreeze" = xyes])
 
-UL_BUILD_INIT([blkdiscard], [check])
-UL_REQUIRES_LINUX([blkdiscard])
-AM_CONDITIONAL([BUILD_BLKDISCARD], [test "x$build_blkdiscard" = xyes])
-
 UL_BUILD_INIT([blkzone], [check])
 UL_REQUIRES_LINUX([blkzone])
 UL_REQUIRES_HAVE([blkzone], [linux_blkzoned_h], [linux/blkzoned.h header])
