diff --git a/es-app/src/ApiSystem.cpp b/es-app/src/ApiSystem.cpp
index 19acd7ded..721347be9 100644
--- a/es-app/src/ApiSystem.cpp
+++ b/es-app/src/ApiSystem.cpp
@@ -745,12 +745,20 @@ std::vector<std::string> ApiSystem::getAvailableChannels()
 
 std::vector<std::string> ApiSystem::getAvailableCpuGovernors()
 {
-	return executeEnumerationScript("/usr/bin/sh -lc \"echo \\\"default\\\"; tr \\\" \\\" \\\"\\n\\\" < /sys/devices/system/cpu/cpufreq/policy0/scaling_available_governors\" | grep \[a-z\]");
+#ifdef __x86_64__
+        return executeEnumerationScript("/usr/bin/sh -lc \". /etc/profile.d/100-amd64-freqfunctions; get_available_cpu_governors\"");
+#else
+        return executeEnumerationScript("/usr/bin/sh -lc \". /etc/profile.d/099-freqfunctions; get_available_cpu_governors\"");
+#endif
 }
 
 std::vector<std::string> ApiSystem::getAvailableGpuGovernors()
 {
-	return executeEnumerationScript("/usr/bin/sh -lc \". /etc/profile.d/099-freqfunctions; get_available_gpu_governors\"");
+#ifdef __x86_64__
+        return executeEnumerationScript("/usr/bin/sh -lc \". /etc/profile.d/100-amd64-freqfunctions; get_available_cpu_governors\"");
+#else
+        return executeEnumerationScript("/usr/bin/sh -lc \". /etc/profile.d/099-freqfunctions; get_available_cpu_governors\"");
+#endif
 }
 
 std::vector<std::string> ApiSystem::getAvailableDisplayModes()
