Add missing libretro-common sources to CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -82,10 +82,13 @@
         ${CMAKE_CURRENT_SOURCE_DIR}/extern/libretro-common/net/net_socket.c
         ${CMAKE_CURRENT_SOURCE_DIR}/extern/libretro-common/net/net_compat.c
         ${CMAKE_CURRENT_SOURCE_DIR}/extern/libretro-common/net/net_http.c
+        ${CMAKE_CURRENT_SOURCE_DIR}/extern/libretro-common/encodings/encoding_utf.c
         ${CMAKE_CURRENT_SOURCE_DIR}/extern/libretro-common/features/features_cpu.c
         ${CMAKE_CURRENT_SOURCE_DIR}/extern/libretro-common/time/rtime.c
         ${CMAKE_CURRENT_SOURCE_DIR}/extern/libretro-common/file/config_file.c
         ${CMAKE_CURRENT_SOURCE_DIR}/extern/libretro-common/file/file_path.c
+        ${CMAKE_CURRENT_SOURCE_DIR}/extern/libretro-common/file/file_path_io.c
+        ${CMAKE_CURRENT_SOURCE_DIR}/extern/libretro-common/file/retro_dirent.c
         ${CMAKE_CURRENT_SOURCE_DIR}/extern/libretro-common/lists/dir_list.c
         ${CMAKE_CURRENT_SOURCE_DIR}/extern/libretro-common/lists/string_list.c
         ${CMAKE_CURRENT_SOURCE_DIR}/extern/libretro-common/string/stdstring.c
