diff --git a/external/CMakeLists.txt.orig b/external/CMakeLists.txt
index 5c09aab150..dae9c30c15 100644
--- a/external/CMakeLists.txt
+++ b/external/CMakeLists.txt
@@ -395,9 +395,9 @@ target_include_directories(libatrac9 PUBLIC LibAtrac9/C/src)
 option(BUILD_SHARED_LIBS "Build shared library" OFF)
 option(XXHASH_BUILD_XXHSUM "Build the xxhsum binary" OFF)
 option(XXH_X86DISPATCH_ALLOW_AVX "Allow building XXH3 with AVX even if it crashes on SSE2-Only CPUs" OFF)
-if((NOT APPLE) AND (ARCHITECTURE STREQUAL "x86_64"))
-	set(DISPATCH 1)
-endif()
+#if((NOT APPLE) AND (ARCHITECTURE STREQUAL "x86_64"))
+#	set(DISPATCH 1)
+#endif()
 add_subdirectory(xxHash/cmake_unofficial EXCLUDE_FROM_ALL)
 if(XXH_X86DISPATCH_ALLOW_AVX)
 	target_compile_definitions(xxhash PRIVATE XXH_X86DISPATCH_ALLOW_AVX)
