diff -rupN GamepadTester.orig/src/main.cpp GamepadTester/src/main.cpp
--- GamepadTester.orig/src/main.cpp	2024-04-26 21:31:53.731460755 +0000
+++ GamepadTester/src/main.cpp	2024-04-26 22:29:53.239175112 +0000
@@ -5,8 +5,8 @@
 #include <SDL2/SDL2_gfxPrimitives_font.h>
 
 #define STICK_MAX_VAL (float(0x7fff))
-#define WIN_W 1100
-#define WIN_H 800
+#define WIN_W 1920
+#define WIN_H 1152
 
 std::string cStringToStdString(const char* str, const std::string& or_="N/A")
 {
