--- a/main.cpp
+++ b/main.cpp
@@ -258,8 +258,13 @@
 
 	unzipFiles(fname, ".");
 	if (zipOpen(fname, &b, fsize, ftype)) {
-		const char* tmpName = (ftype == 1) ? "unz.d64" : "unz.tap";
+		const char* tmpName = (ftype == 1) ? "/tmp/unz.d64" : "/tmp/unz.tap";
 		FILE *tmp = fopen(tmpName, "wb");
+		if (!tmp) {
+			fprintf(stderr, "ERROR: Cannot write to %s\n", tmpName);
+			perror("fopen");
+			return false;
+		}
 
 		if (b && tmp) {
 			fwrite(b, sizeof(char), fsize, tmp);
@@ -1076,15 +1081,32 @@
 				break;
 
 			case SDL_JOYBUTTONUP:
+				// Anbernic RG ARC buttons
+				// 0 -  B
+				// 1 -  A
+				// 2 -  C
+				// 3 -  X
+				// 4 -  Y
+				// 5 -  Z
+				// 6 -  L1
+				// 7 -  R1
+				// 8 -  L2
+				// 9 -  R2
+				// 10 - SELECT
+				// 11 - START
+				// 12 - DPAD UP
+				// 13 - DPAD DOWN
+				// 14 - DPAD LEFT
+				// 15 - DPAD RIGHT
 				//printf("joy: %u, button: %u was pressed!\n", event.jbutton.which, event.jbutton.button);
-				if (event.jbutton.button == 5 || event.jbutton.button == 14) { // START button
+				if (event.jbutton.button == 5) {
 					enterMenu();
 				} else if (event.jbutton.button == 4) {
-					ted8360->copyToKbBuffer("RUN:\r",5);
-				} else if (event.jbutton.button == 11 || event.jbutton.button == 1) {
+					ted8360->getKeys()->simulateKeyPress(SDL_SCANCODE_F1, 100); // F1
+				} else if (event.jbutton.button == 3) {
+					ted8360->getKeys()->simulateKeyPress(SDL_SCANCODE_SPACE, 100); // SPACE
+				} else if (event.jbutton.button == 1) {
 					doSwapJoy();
-				} else if (event.jbutton.button == 12) {
-					toggleFullThrottle(NULL);
 				}
 				break;
 
