--- a/keyboard.cpp
+++ b/keyboard.cpp
@@ -13,6 +13,8 @@
 #include <stdio.h>
 #include "keyboard.h"
 
+KEYS* KEYS::instance = nullptr;
+
 static SDL_Scancode kbmatrix[] = {
 	SDL_SCANCODE_BACKSPACE, SDL_SCANCODE_RETURN, SDL_SCANCODE_END, SDL_SCANCODE_F4, SDL_SCANCODE_F1, SDL_SCANCODE_F2, SDL_SCANCODE_F3, SDL_SCANCODE_LEFTBRACKET,
 	SDL_SCANCODE_3, SDL_SCANCODE_W, SDL_SCANCODE_A, SDL_SCANCODE_4, SDL_SCANCODE_Z, SDL_SCANCODE_S, SDL_SCANCODE_E, SDL_SCANCODE_LSHIFT,
@@ -48,6 +50,7 @@
 
 KEYS::KEYS() 
 {
+	KEYS::instance = this;
 	empty();
 	block(false);
 }
@@ -84,7 +87,13 @@
 
 unsigned char KEYS::keyReadMatrixRow(unsigned int r)
 {
-	const Uint8 *kbstate = SDL_GetKeyboardState(NULL);
+	//const Uint8 *kbstate = SDL_GetKeyboardState(NULL);
+	Uint8 mergedState[SDL_NUM_SCANCODES];
+	memcpy(mergedState, SDL_GetKeyboardState(NULL), SDL_NUM_SCANCODES);
+	for (int i = 0; i < SDL_NUM_SCANCODES; ++i)
+		mergedState[i] |= fakeKbState[i];
+	const Uint8 *kbstate = mergedState;
+
 	unsigned char tmp;
 
 	//for(int i = 0; i < 256; i++)
@@ -353,6 +362,18 @@
 	return tmp;
 }
 
+void KEYS::simulateKeyPress(SDL_Scancode scan, Uint32 durationMs) {
+    fakeKbState[scan] = 1;
+
+    SDL_AddTimer(durationMs, [](Uint32, void* param) -> Uint32 {
+        SDL_Scancode* sc = static_cast<SDL_Scancode*>(param);
+        KEYS::getInstance()->fakeKbState[*sc] = 0;
+        delete sc;
+        return 0;
+    }, new SDL_Scancode(scan));
+}
+
+
 KEYS::~KEYS()
 {
 
