--- a/keyboard.h
+++ b/keyboard.h
@@ -26,6 +26,7 @@
 		unsigned char latched;
 		unsigned char keyReadMatrixRow(unsigned int r);
 		unsigned char blockMask;
+		Uint8 fakeKbState[SDL_NUM_SCANCODES] = {0};
 		virtual unsigned int fireButtonIndex(unsigned int a) const { return 6 + a; };
 
 	public:
@@ -51,9 +52,12 @@
 		}
 		unsigned char readLatch() { return latched | blockMask; };
 		void block(bool isBlocked) { blockMask = isBlocked ? 0xFF : 0x00; };
+		void simulateKeyPress(SDL_Scancode scan, Uint32 durationMs = 100);
 		static unsigned char readPaddleAxis(unsigned short axis);
 		static unsigned char readPaddleFireButton(unsigned int paddleID);
 		unsigned char readSidcardJoyport();
+		static KEYS* instance;
+		static KEYS* getInstance() { return instance; }
 };
 
 extern rvar_t inputSettings[];
