diff --git a/FEXCore/Source/CMakeLists.txt b/FEXCore/Source/CMakeLists.txt
index 8150844..ed0aae4 100644
--- a/FEXCore/Source/CMakeLists.txt
+++ b/FEXCore/Source/CMakeLists.txt
@@ -251,6 +251,10 @@ endfunction()
 
 # Build FEXCore_Base static library
 add_library(FEXCore_Base STATIC ${FEXCORE_BASE_SRCS})
+set_source_files_properties(
+    Interface/Core/Interpreter/Fallbacks/InterpreterFallbacks.cpp
+    PROPERTIES COMPILE_FLAGS "-O0 -fno-lto"
+)
 target_link_libraries(FEXCore_Base PUBLIC ${LIBS})
 AddDefaultOptionsToTarget(FEXCore_Base)
 
