diff --git a/ThunkLibs/HostLibs/CMakeLists.txt b/ThunkLibs/HostLibs/CMakeLists.txt
index a526c72ff..346eb71d4 100644
--- a/ThunkLibs/HostLibs/CMakeLists.txt
+++ b/ThunkLibs/HostLibs/CMakeLists.txt
@@ -127,6 +127,7 @@ foreach(GUEST_BITNESS IN LISTS BITNESS_LIST)
 
   generate(libasound ${CMAKE_CURRENT_SOURCE_DIR}/../libasound/libasound_interface.cpp ${GUEST_BITNESS})
   target_include_directories_from_pkgconfig(libasound-${GUEST_BITNESS}-deps alsa)
+  target_include_directories(libasound-${GUEST_BITNESS}-deps INTERFACE ${CMAKE_SYSROOT}/usr/include)
   add_host_lib(asound ${GUEST_BITNESS})
 
   # disabled for now, headers are platform specific
@@ -137,6 +138,11 @@ foreach(GUEST_BITNESS IN LISTS BITNESS_LIST)
 
   generate(libdrm ${CMAKE_CURRENT_SOURCE_DIR}/../libdrm/libdrm_interface.cpp ${GUEST_BITNESS})
   target_include_directories_from_pkgconfig(libdrm-${GUEST_BITNESS}-deps libdrm /usr/include/drm /usr/include/libdrm)
+  target_include_directories(libdrm-${GUEST_BITNESS}-deps INTERFACE
+    ${CMAKE_SYSROOT}/usr/include
+    ${CMAKE_SYSROOT}/usr/include/drm
+    ${CMAKE_SYSROOT}/usr/include/libdrm
+  )
   add_host_lib(drm ${GUEST_BITNESS})
 endforeach()
 
@@ -148,21 +154,30 @@ foreach(GUEST_BITNESS IN LISTS BITNESS_LIST)
   endif()
 
   generate(libvulkan ${CMAKE_CURRENT_SOURCE_DIR}/../libvulkan/libvulkan_interface.cpp ${GUEST_BITNESS})
-  target_include_directories(libvulkan-${GUEST_BITNESS}-deps INTERFACE ${FEX_PROJECT_SOURCE_DIR}/External/Vulkan-Headers/include/)
+  target_include_directories(libvulkan-${GUEST_BITNESS}-deps INTERFACE
+    ${FEX_PROJECT_SOURCE_DIR}/External/Vulkan-Headers/include/
+    ${CMAKE_SYSROOT}/usr/include
+  )
   target_include_directories_from_pkgconfig(libvulkan-${GUEST_BITNESS}-deps "xcb;x11;xrandr;xrender")
   add_host_lib(vulkan ${GUEST_BITNESS})
 
   generate(libwayland-client ${CMAKE_CURRENT_SOURCE_DIR}/../libwayland-client/libwayland-client_interface.cpp ${GUEST_BITNESS})
   add_host_lib(wayland-client ${GUEST_BITNESS})
   target_include_directories_from_pkgconfig(libwayland-client-${GUEST_BITNESS}-deps wayland-client /usr/include/wayland)
+  target_include_directories(libwayland-client-${GUEST_BITNESS}-deps INTERFACE
+    ${CMAKE_SYSROOT}/usr/include
+    ${CMAKE_SYSROOT}/usr/include/wayland
+  )
 
   generate(libEGL ${CMAKE_CURRENT_SOURCE_DIR}/../libEGL/libEGL_interface.cpp ${GUEST_BITNESS})
   target_include_directories_from_pkgconfig(libEGL-${GUEST_BITNESS}-deps egl)
+  target_include_directories(libEGL-${GUEST_BITNESS}-deps INTERFACE ${CMAKE_SYSROOT}/usr/include)
   add_host_lib(EGL ${GUEST_BITNESS})
 
   generate(libGL ${CMAKE_CURRENT_SOURCE_DIR}/../libGL/libGL_interface.cpp ${GUEST_BITNESS})
   target_include_directories_from_pkgconfig(libGL-${GUEST_BITNESS}-deps gl)
   target_include_directories_from_pkgconfig(libGL-${GUEST_BITNESS}-deps "xcb;x11;xrandr;xrender")
+  target_include_directories(libGL-${GUEST_BITNESS}-deps INTERFACE ${CMAKE_SYSROOT}/usr/include)
   add_host_lib(GL ${GUEST_BITNESS})
 
   find_package(OpenGL REQUIRED)
