diff --git a/ThunkLibs/HostLibs/CMakeLists.txt b/ThunkLibs/HostLibs/CMakeLists.txt
--- a/ThunkLibs/HostLibs/CMakeLists.txt
+++ b/ThunkLibs/HostLibs/CMakeLists.txt
@@ -105,6 +105,7 @@
   #add_host_lib(fex_malloc ${GUEST_BITNESS})
 
   generate(libasound ${CMAKE_CURRENT_SOURCE_DIR}/../libasound/libasound_interface.cpp ${GUEST_BITNESS})
+  target_include_directories(libasound-${GUEST_BITNESS}-deps INTERFACE ${CMAKE_SYSROOT}/usr/include)
   add_host_lib(asound ${GUEST_BITNESS})
 
   # disabled for now, headers are platform specific
@@ -114,8 +115,11 @@
   # target_include_directories(SDL2-host PRIVATE ${SDL2_INCLUDE_DIRS})
 
   generate(libdrm ${CMAKE_CURRENT_SOURCE_DIR}/../libdrm/libdrm_interface.cpp ${GUEST_BITNESS})
-  target_include_directories(libdrm-${GUEST_BITNESS}-deps INTERFACE /usr/include/drm/)
-  target_include_directories(libdrm-${GUEST_BITNESS}-deps INTERFACE /usr/include/libdrm/)
+  target_include_directories(libdrm-${GUEST_BITNESS}-deps INTERFACE
+    ${CMAKE_SYSROOT}/usr/include
+    ${CMAKE_SYSROOT}/usr/include/drm
+    ${CMAKE_SYSROOT}/usr/include/libdrm
+  )
   add_host_lib(drm ${GUEST_BITNESS})
 endforeach()
 
@@ -127,17 +131,25 @@
   endif()
 
   generate(libvulkan ${CMAKE_CURRENT_SOURCE_DIR}/../libvulkan/libvulkan_interface.cpp ${GUEST_BITNESS})
-  target_include_directories(libvulkan-${GUEST_BITNESS}-deps INTERFACE ${FEX_PROJECT_SOURCE_DIR}/External/Vulkan-Headers/include/)
+  target_include_directories(libvulkan-${GUEST_BITNESS}-deps INTERFACE
+    ${FEX_PROJECT_SOURCE_DIR}/External/Vulkan-Headers/include/
+    ${CMAKE_SYSROOT}/usr/include
+  )
   add_host_lib(vulkan ${GUEST_BITNESS})
 
   generate(libwayland-client ${CMAKE_CURRENT_SOURCE_DIR}/../libwayland-client/libwayland-client_interface.cpp ${GUEST_BITNESS})
   add_host_lib(wayland-client ${GUEST_BITNESS})
-  target_include_directories(libwayland-client-${GUEST_BITNESS}-deps INTERFACE /usr/include/wayland)
+  target_include_directories(libwayland-client-${GUEST_BITNESS}-deps INTERFACE
+    ${CMAKE_SYSROOT}/usr/include
+    ${CMAKE_SYSROOT}/usr/include/wayland
+  )
 
   generate(libEGL ${CMAKE_CURRENT_SOURCE_DIR}/../libEGL/libEGL_interface.cpp ${GUEST_BITNESS})
+  target_include_directories(libEGL-${GUEST_BITNESS}-deps INTERFACE ${CMAKE_SYSROOT}/usr/include)
   add_host_lib(EGL ${GUEST_BITNESS})
 
   generate(libGL ${CMAKE_CURRENT_SOURCE_DIR}/../libGL/libGL_interface.cpp ${GUEST_BITNESS})
+  target_include_directories(libGL-${GUEST_BITNESS}-deps INTERFACE ${CMAKE_SYSROOT}/usr/include)
   add_host_lib(GL ${GUEST_BITNESS})
 
   find_package(OpenGL REQUIRED)
