diff --git a/ThunkLibs/include/common/Host.h b/ThunkLibs/include/common/Host.h
--- a/ThunkLibs/include/common/Host.h
+++ b/ThunkLibs/include/common/Host.h
@@ -446,8 +446,22 @@ struct host_to_guest_convertible {
   operator guest_layout<uint8_t*>() const requires (std::is_same_v<T, char*>)
   {
     return (guest_layout<uint8_t*>)reinterpret_cast<const host_to_guest_convertible<uint8_t*>&>(*this);
   }
 
+  operator guest_layout<const signed char*>() const requires (std::is_same_v<T, const char*>)
+  {
+    guest_layout<const signed char*> ret;
+    ret.data = reinterpret_cast<uintptr_t>(from.data);
+    return ret;
+  }
+
+  operator guest_layout<signed char*>() const requires (std::is_same_v<T, char*>)
+  {
+    guest_layout<signed char*> ret;
+    ret.data = reinterpret_cast<uintptr_t>(from.data);
+    return ret;
+  }
+
   // Make guest_layout of "wchar_t" and "uint32_t" interoperable
   operator guest_layout<uint32_t*>() const requires (std::is_same_v<T, wchar_t*>)
   {
