--- a/ThunkLibs/HostLibs/CMakeLists.txt	2026-05-17 16:07:31.481665429 +1000
+++ b/ThunkLibs/HostLibs/CMakeLists.txt	2026-05-17 16:09:47.417483655 +1000
@@ -123,7 +123,7 @@
   add_host_lib(drm ${GUEST_BITNESS})
 endforeach()
 
-set(BITNESS_LIST "32;64")
+set(BITNESS_LIST "64")
 foreach(GUEST_BITNESS IN LISTS BITNESS_LIST)
   if (BUILD_FEX_LINUX_TESTS)
     generate(libfex_thunk_test ${CMAKE_CURRENT_SOURCE_DIR}/../libfex_thunk_test/libfex_thunk_test_interface.cpp ${GUEST_BITNESS})
