diff -rupN linux.orig/arch/arm64/boot/dts/amlogic/meson-g12-common.dtsi linux/arch/arm64/boot/dts/amlogic/meson-g12-common.dtsi
--- linux.orig/arch/arm64/boot/dts/amlogic/meson-g12-common.dtsi	2024-09-23 08:43:22.120928646 +1000
+++ linux/arch/arm64/boot/dts/amlogic/meson-g12-common.dtsi	2024-09-23 10:52:16.716701961 +1000
@@ -2405,6 +2405,7 @@
 			};
 		};
 
+		// Mali GPU - Panfrost
 		mali: gpu@ffe40000 {
 			compatible = "amlogic,meson-g12a-mali", "arm,mali-bifrost";
 			reg = <0x0 0xffe40000 0x0 0x40000>;
@@ -2417,6 +2418,30 @@
 			resets = <&reset RESET_DVALIN_CAPB3>, <&reset RESET_DVALIN>;
 			operating-points-v2 = <&gpu_opp_table>;
 			#cooling-cells = <2>;
+			status = "disabled";
+		};
+
+		// Mali GPU - Vulkan
+		mali_vulkan: gpu_mali_vulkan@ffe40000 {
+			compatible = "arm,mali-midgard";
+			reg = <0x0 0xffe40000 0x0 0x40000>,
+                    <0 0xFFD01000 0 0x01000>,
+                    <0 0xFF800000 0 0x01000>,
+                    <0 0xFF63c000 0 0x01000>,
+                    <0 0xFFD01000 0 0x01000>;
+ 			interrupt-parent = <&gic>;
+			interrupts = <GIC_SPI 160 IRQ_TYPE_LEVEL_HIGH>,
+ 				     <GIC_SPI 161 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 162 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "GPU", "MMU", "JOB";
+ 			clocks = <&clkc CLKID_MALI>;
+			clock-names = "clk_mali";
+ 			resets = <&reset RESET_DVALIN_CAPB3>, <&reset RESET_DVALIN>;
+ 			operating-points-v2 = <&gpu_opp_table>;
+ 			#cooling-cells = <2>;
+			system-coherency = <0>;
+			power_policy = "always_on";
+			status = "okay";
 		};
 	};
 
